\section*{Summary}

Using genome scale models to estimate cost has precedent in the estimates of oxygen related shadow prices in the work of Varma \emph{et al.} \cite{varma1993}. In chapter I have illustrated how similar methods can be used to estimate amino acid costs and how these costs show a high degree of similarity with previous measures of energetic cost, or atomic content. Furthermore new measures of amino acid cost have been estimated taking into consideration the estimated quantity of the amino acid present in biomass.

\clearpage

\section{Introduction}

\subsection{Understanding biological systems}

Early enzymology predicted rate limiting steps in biological pathways. The slowest step in the pathway will limit the rate at which the pathway acts. If the rate of the slowest step is increased, then intuition suggests that the rate of the entire pathway will increase proportionally. When attempting to modulate the flux through the tryptophan biosynthetic pathway in yeast Niederberger \emph{et. al} \cite{niederberger1992} found that individual up or down regulation of specific enzymes had little overall effect on tryptophan biosynthesis. Instead the rate of the biosynthetic pathway could instead be increased by increasing the quantity of five related enzymes in tandem. This research is an example control in a biological system being distributed over the system as a whole rather than focused at individual points.

The theory of metabolic control analysis \cite{fell} states that there are no rate limiting steps in biological systems, but instead that each part of the systems shares a measure of control, with some parts having more control than others. The control an enzyme has on the rest of the system can determined by measuring the effect of its up or down modulation on the reaction in question. For example in metabolic control analysis the control of enzyme X on reaction Y can be described mathematically as follows.

\begin{equation}\label{fcc}
C_{X}^{Y} = \frac{dX}{X}\div\frac{dY}{Y}
\end{equation}

The function $\frac{dX}{X}$ represents the effect of a small change in the reaction flux through reaction X, given the original flux. The function $\div\frac{dY}{Y}$ represents the proportional change in Y resulting from the change in X. Metabolic control analysis defines the ratio of these two functions as the flux control coefficient of reaction X on reaction Y using the nomenclature $C_{X}^{Y}$. Larger values of $C_{X}^{Y}$ indicate that reaction X has a high degree of control on reaction Y. Smaller values indicate that reaction Y experiences only small change in flux in response to changes in X.

Determining a numeric value for the relationships between reactions in a metabolic network allows properties of biological systems to be explored. For example Westerhoff \emph{et al.} \cite{hierarchical_control} predicted with an \emph{in silico} model of yeast glycolysis \cite{teusink} which reaction were controlled via changes in enzyme quantity and which were controlled by increasing in metabolite concentrations.

WAS TEUSINK MODEL USED?
INSERT EXAMPLES OF MCA USE

Other aspects of metabolic control theory is the prediction of metabolite control coefficients, whereby metabolite concentrations will have a degree of control on the rate of reactions.

INSERT EXAMPLES OF MCC use.

\subsection{Stoichiometric approaches}

\emph{In silico} kinetic models require specific information about enzyme activity for each reaction. In the case of an enzyme that follows simple Michalis-Menten kinetics parameters such as Km and Vmax are sufficient. To model more complex enzymatic reactions requires larger numbers of parameter sets which require more experimental effort to derive. This means that kinetic model of biological processes are often small with a limited number of reaction sets, due to the difficulty in finding sufficient enzymatic data for the reactions in the system. Stoichiometric models do not require kinetic data but instead only the connectivity between reactions, the participating metabolites and the \emph{in vivo} direction of the reaction.

Requiring less data for construction, stoichiometric models are much larger than kinetic models. Genome scale stoichiometric models can be constructed from the genome sequence of the organism through identifying the presence of enzymes, and connecting the encoded reactions based on metabolic maps. An iterative process of simulation can identify possible missing reactions required for viable growth \cite{genome_model_construction}. The growth rate of the model is simulated through of linear optimisation problem to maximise a given objective reaction such as the production of biomass defined in the combination of internal reaction and the available nutrients.

Genome scale models can be used to predict which \emph{in silico} single \cite{single_knockout} or double \cite{double_knockout} gene deletions lead to an \emph{in vivo} lethal or sick phenotype. Further \emph{in silico} analysis predicts that additional genes making a significant contribution to growth could be identified through multiple gene knockout \cite{deutscher2008}.

INSERT FURTHER FBA EXAMPLES HERE

Similar to the calculation of metabolite control coefficients described in the previous section, stoichiometric models can be used to calculate the shadow price of a molecule. In economics the shadow price, for example of a sold product, represents the effect in changes of prices on the market. Varma \emph{et al.} \cite{varma1993} transfered the use of shadow prices to biology and used this in the estimation of the shadow price of oxygen on \emph{E. coli} energy metabolism. Through simulating a decrease in the availability of oxygen the researchers identified how the shadow prices of the redox carriers changed with the availability of oxygen. The research showed that with decreasing availability of oxygen, the shadow price of oxygen use rises, this leads to a decrease in the shadow price of ethanol for energy production decreases as it becomes instead used to eliminate redox and secreted.

\subsection{Estimating the cost of amino acids}

\subsubsection{Energetic cost}

The article by Varma \emph{et al.} highlights how systems biology models can be used to estimate the importance of a set of metabolites to cell growth. Related to this is the question of how much it costs the cell to synthesise and use amino acids. Craig \& Weber \cite{craig1998} estimated the cost of the twenty amino acids as the sum of how many high energy phosphate bonds (e.g. ATP) and reducing molecules (e.g. NADPH) are diverted from the available energy pool in the synthesis of each amino acid from glucose. The aim of deriving these costs was to estimate the importance of amino acid cost the evolution of a set of \emph{E. coli} proteins.

Akashi \& Gojobori \cite{akashi2002} explored a similar approach calculating cost on a range different energy sources to estimate the importance of amino acid cost in predicted \emph{E. coli} and \emph{B. subtilis} gene expression. Heizer \emph{et al.} \cite{heizer} expanded this estimating amino acid cost for inorganic sources in their analysis of the genomes of four prokaryotic species including photoautotrophs. These estimates of cost are based on the curation of the metabolic network to account for the expenditure of high energy molecules during synthesis. Seligmann argued \cite{seligmann2004} that the cost of amino acid biosynthesis must take into account more than just the energy required for synthesis. Seligmann used molecular weight as a proxy amino acid cost arguing that this may take into account investments in synthesising larger more complex amino acids. Using molecular weight also has the advantage where the metabolic network in not required to curate the number of high energy molecules used during synthesis.

DOUBLE CHECK AKASHI PAPER
HOW IS WAGNER'S COST DIFFERENT

\subsubsection{Atomic content}

Amino acid synthesis will require energy to power each step in the metabolic network, in addition to the energetic cost the molecules part of the amino acid structure may also be considered a cost of synthesis. Mazel \& Marliere \cite{mazel1989} showed that \emph{Calothrix} proteins expressed in sulphur limiting conditions are depleted for sulphur containing cysteine and methionine amino acids. This work indicating the atomic content of an amino acid should also be taken into consideration when attempting to define a measure of cost. Work by both Baudoin-Cornu \emph{et al.} \cite{baudoin2001} and Bragg \emph{et al.}	\cite{bragg2006} used atomic content as a measure of amino acid cost when examining protein content.

\subsection{Summary of results}

The results presented in this chapter are split into two sections. The first describes an approach to estimating amino acid cost focusing on the \emph{S. cerevisiae} genome scale metabolic model. The approach used to estimate amino acid cost is similar to that described by Varma \emph{et al.} in their estimation of the shadow prices in \emph{E.coli} energy metabolism. The estimated costs are compared to the other estimates of amino acid cost that appear in the literature described above.

The second part of results section describes an experimental approach to estimate the cost of a gene. This is again in \emph{S. cerevisiae} using the genome scale metabolic model. This approach performs a type of robustness analysis on the gene encoded metabolic function. This analysis produces a robustness distribution describing how increasing and decreasing metabolic rate for an function affect the \emph{in silico} model fitness.

\clearpage

\section{Results}

The biomass equation in a flux balance analysis model describes the estimated requirements and stoichiometry of metabolites required growth. In the biomass reaction in the \emph{S. cerevisiae} model, the requirements include fats such as triglyceride and ergosterol, sugars such as glucose and mannose, as well as smaller metabolites sulphate and water. The stoichiometry of the biomass equation is derived from the dry weight biomass of the \emph{S. cerevisiae} cell \cite{duarte2004}. The units of each metabolite requirement are mmols per gram of dry weight biomass (mmol $^-1$ gDW $^-1$). In addition to the previous described metabolites, the yeast biomass reaction includes the twenty proteogenic amino acids. Each amino acid stoichiometric requirement can be manipulated to estimate the effect on biomass production. Figure~\vref{figure:curve_estimation} outlines this process.

\subsection{Amino acid cost estimation}\label{section:amino_acid_cost_estimation}

As discussed, metabolic control analysis can be used to estimate the importance of an object in a kinetic model. In this section I will show similar principles of can be used to estimate the sensitivity of growth rate to the changes in amino acid requirements in biomass. The biomass equation in a flux balance analysis model describes the estimated requirements and stoichiometry of metabolites required for the production of a biomass, which is equivalent to growth. For instance in the biomass reaction in the \emph{S. cerevisiae} model, the requirements include fats such as triglyceride and ergosterol, sugars such as glucose and mannose, as well as smaller metabolites such as sulphate and water. The stoichiometry of the biomass equation is derived from the dry weight biomass of the \emph{S. cerevisiae} cell \cite{duarte2004a}. The protein content of a yeast cell is also included in the biomass equation as the twenty individual amino acids that make up protein, with the inclusion of theses amino acids in the biomass equation, the reaction stoichiometries of each can be manipulated to estimate the effect on biomass production.

\begin{figure}
  \subfloat[Cost estimation in glucose limited conditions]{
    \label{figure:curve_estimation:schematic}
    \includegraphics*[width=6cm]{cost_schematic.eps}
  }
  \hfill
  \subfloat[Structure of Tryptophan]{
    \label{figure:curve_estimation:acids}
    \includegraphics*[width=5cm]{tryptophan.eps}
  }
  \hfill
  \subfloat[Supply-demand elasticity for tryptophan]{
    \label{figure:curve_estimation:curve}
    \includegraphics*[width=7cm]{curve_estimation.eps}
  }
  \hfill
  \subfloat[Comparison of costs for glycine, histidine, and tryptophan]{
    \label{figure:curve_estimation:costs}
    \includegraphics*[width=7cm]{comparison_of_costs.eps}
  }
  \hfill
  \label{figure:curve_estimation}
  \caption[Methods used for estimation of amino acid cost]{Outline of the methods used for the estimation of amino acid cost. Part~\ref{figure:curve_estimation:schematic} shows a schematic representation of estimating amino cost in glucose limited conditions using a genome scale model. The growth rate of the model is fixed at a constant value so that costs can be compared  between different environments and models. The objective function of the model is then set to be the nutrient in question, which in this example is glucose. The glucose flux entering the model is therefore minimised when optimising the model. All other nutrient entering the model, for example ammonia are then unbounded. Part~\ref{figure:curve_estimation:acids} illustrates the structure of tryptophan which contains a relative complex double ring side chain which also includes two nitrogen atoms. Part~\ref{figure:curve_estimation:curve} the glucose and ammonium slopes for tryptophan when the requirement for tryptophan is altered in the biomass reaction. Part~\ref{figure:curve_estimation:costs} compares the glucose and ammonium costs for tryptophan, glycine and histidine. }
\end{figure}

Figure~\vref{figure:curve_estimation} illustrates the method used to estimate the cost of an amino acid with a genome scale metabolic model. Figure~\ref{figure:curve_estimation:a} illustrates the genome scale model usage to estimate amino acid cost in a glucose limited environment. The minimal and maximal boundary on the production of biomass is fixed at a constant value, for example 0.3 hr$^{-1}$. Using a constant growth rate value for amino acid estimation means that all amino acid cost estimations are scaled by the same growth rate. The objective function of the model is then set to be the minimisation of cellular entry of the nutrient for which the amino acid costs are being estimated, which in Figure~\ref{figure:curve_estimation:a} is glucose. All all other nutrient exchange fluxes were set to have a however boundary of -10000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$, the aim of this is to set all other nutrients to be non limiting on simulated growth. For the \emph{S. cerevisiae} iND750 \cite{duarte2004a} genome scale model the default exchange reactions when simulating growth are glucose, ammonium, sulphate, phosphate, oxygen and water. Each exchange flux represented the only source for an essential nutrient, and therefore when estimating the glucose cost of an amino acid there were no other high energy sugars available, or when estimating the ammonium cost there were no other nitrogen sources entering the cell. Model optimisation then found what the minimum required amount of nutrient intake flux was required to maintain the growth rate at which the model was fixed. Four nutrients were considered in the analysis: glucose, ammonium, sulphate, and phosphate. 

Figure\ref{figure:curve_estimation:acids} illustrates the structure of the amino acid tryptophan. Tryptophan has a double ring side chain which is large and complex relative to other amino acids. This side chain also has two nitrogen atoms. Figure\ref{figure:curve_estimation:curve} shows the slope for supply and demand elasticity for tryptophan in glucose and ammonium limiting conditions. The slope for each environment is produced by altering the requirement of tryptophan in model biomass reaction. The stoichiometry of tryptophan is changed with a slight increase or small decrease in requirement of the range of $\pm$0.0002\%. The result of this change is then measured in the effect on the uptake flux in question. Figure\ref{figure:curve_estimation:curve} shows two different response slopes for changes in tryptophan requirement on the ammonium and glucose uptake fluxes. The slope of the relationship of tryptophan on the respective uptake flux was used as the cost of that amino acid in the given nutrient limitation. This cost is the value of $m$ in the linear relationship $y = mx + c$ where $x$ is tryptophan uptake requirement and $y$ is resulting nutrient uptake flux. This cost was is based on relative changes to amino acid requirement, where the  amino acid stoichiometry is multiplied by a small percentage shown on the x-axis of Figures~\vref{figure:curve_estimation:curve}. This relative estimation of amino cost can be rescaled to estimate a cost based on absolute changes in amino acid requirement by dividing the estimated relative cost by the original requirement of the amino acid in the biomass reaction. The rescaling provides an estimate of amino acid cost based on absolute changes in amino acid requirement. One further consideration is the estimated amino acid cost are a function of the growth rate at which they were estimated. For instance the absolute cost of tryptophan estimated at 0.2 hr$^{-1}$ is twice the cost of tryptophan estimated at 0.1 hr$^{-1}$. To control for growth rate when estimating amino acid cost, the costs were estimated at a three levels of feasible yeast growth rates 0.1 hr$^{-1}$, 0.2 hr$^{-1}$ and 0.3 hr$^{-1}$. Each amino acid cost is then rescaled to be independent of the growth rate at which is was estimated. The amino acid costs presented in this work are those estimated at the 0.3 hr$^{-1}$ growth rate, a level of growth where \emph{S. cerevisiae} uses repiratory energy production \cite{famili2003}.

Figure\ref{figure:curve_estimation:costs} compares the estimated absolute costs for tryptophan, histidine and glycine in two of the examined conditions, glucose and ammonium limitation. Tryptophan which is an expensive amino acid in glucose limitation, is slightly cheaper in nitrogen limitation. In the reverse of this, histidine which is a relatively cheap amino acid in glucose limitation is correspondingly more expensive, containing three nitrogen atoms, in ammonium limitation. Glycine, a small amino acid with a single hydrogen side chain is relatively cheap in both nutrient limitations.

The relative costs are derived from the slope in the unitless percentage change in the amino acid requirement, and the corresponding mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ response in nutrient intake flux. When divided by the hr$^{-1}$ growth rate at which the cost was estimated, the units are mmol$^{-1}$ gDW$^{-1}$. This represents the change in nutrient uptake flux given the factional change in amino acid requirement in biomass. The absolute cost is derived from the relative cost by division by the original mmol$^{-1}$ gDW$^{-1}$ requirement of the amino acid. The absolute amino acid is therefore unitless and represents what a mmol$^{-1}$ gDW$^{-1}$ increase in amino acid requirement has the mmol$^{-1}$ gDW$^{-1}$ uptake flux of the given nutrient.

Relative and absolute amino acid costs where estimated for all twenty amino acids in four simulated nutrient limiting conditions: glucose, ammonium, sulphate, and phosphate. For clarity costs are described as either $R_{nutrient}$ or $A_{nutrient}$ for relative and absolute costs respectively. The relative and absolute estimated costs for each of these amino acids is illustrated in Figure~\vref{figure:costs_dendrogram_dotplot} along with amino acid costs described in the literature. The estimated amino acid cost are provided in Table~\vref{appendix:table:estimated_costs}, the literature costs are provided in Table\vref{appendix:table:literature_costs}.

\begin{sidewaysfigure}
\centering
\includegraphics*[angle=0,height=13cm]{costs_dendrogram_dotplot.eps}
\caption[Comparison of amino acid cost estimates]{Amino acid cost estimates are shown as bar charts on the left hand side. Each bar chart axis shows the minimum and maximum value of each cost type, rounded to three significant figures. The correlations between costs are compared in a dendrogram on the right hand side computed by complete agglomerative clustering using Spearman's Rank correlation distance between data sets. The illustrated data is shown in Appendix Table~\vref{appendix:table:literature_costs}}
\label{figure:costs_dendrogram_dotplot}
\end{sidewaysfigure}

The $A_{glucose}$ costs is highly correlated with previous measures of amino acid cost. The energetic cost derived by Akashi and Gojobori \cite{akashi2002} is the most correlated with $A_{glucose}$ (Spearman R = 0.94), but $A_{glucose}$ is also correlated (Spearman R > 0.7) with Craig and Weber's energetic cost \cite{craig1998}, Wagner's respiratory cost \cite{wagner2005} and molecular weight \cite{seligmann2004}. This indicates that the $A_{glucose}$ cost captures the energetic cost of synthesising an amino acid. The $A_{ammonium}$ and $A_{sulphate}$ costs are proportional to the nutrient content of the amino acid. The $A_{ammonium}$ costs show a weak correlation (Spearman R 0.5 - 0.6) with the energetic cost measures: Akashi \& Gojobori, Wagner fermentative, and molecular weight.

The $R_{glucose}$ cost shows little correlation with previous measures of amino acid cost. The cost that it is most correlated with is the Wagner fermentative growth (Spearman R = -0.16). The $R_{glucose}$ and $R_{ammonium}$ costs are however correlated with each other (Spearman R = 0.63). Comparing the derived relative and absolute costs, the absolute costs reflect the per molecule biosynthetic, while the relative costs reflect the absolute cost scaled by the use of the amino acid in the genome. For example, the $A_{sulphate}$ costs of methionine and cysteine are both 1, which reflects the single sulphur atom in each amino acid. The $R_{sulphate}$ cost for methionine is however much greater than that of cysteine, as methionine is used proportionally more in the proteome.

$A_{glucose}$ and $R_{glucose}$ were calculated for comparison using the \emph{Escherichia coli iJR904} genome scale model. Figure~\vref{figure:amino_acid_costs_scatterplot} compares both relative and absolute costs for \emph{E. coli} and \emph{S. cerevisiae} models with two costs described in the literature: molecular weight \cite{seligmann2004} and number of ATP and NAPH molecules \cite{akashi2002}.

\begin{figure}
\centering
\includegraphics*[width=13cm]{amino_acid_costs_scatterplot.eps}
\caption[Comparison of the genome scale model derived cost data sets.]{Comparison of estimated amino acid cost with number of ATP and NADPH molecules used in synthesis (left), and molecular weight (right). On the y axis are the amino acid costs estimated using flux balance analysis. Both \emph{S. cerevisiae} and \emph{E. coli} measures are included to illustrate correlation of cost estimates between species. Estimated cost value
es have been been rescaled around their mean value to allow comparisons across species. The trends in each plot are drawn using `loess' smoothing.}
\label{figure:amino_acid_costs_scatterplot}
\end{figure}

Figure~\ref{figure:amino_acid_costs_scatterplot} shows that the both the absolute costs show a small amount of variance in their correlation the example literature described cost measures. Both \emph{E. coli} and \emph{S. cerevisiae} absolute costs measures are particularly well correlated with the Akashi and Gojobori described measure of cost \cite{akashi2002}, however the relationship shows a parabolic relationship. In comparison molecular weight shows a linear relationship with the absolute measures of costs, but shows a greater degree of variation.

The comparison of relative measures of cost illustrates the small degree of correlation with previous estimates of cost. For the both the molecular weight and ATP+NADH cost estimates, neither \emph{E. coli} or \emph{S. cerevisiae} relative cost estimates show any relationship, and furthermore the figure illustrates the disparity in estimated costs between the species models.

\subsection{Reaction elasticity estimation}

\section{Discussion}

This chapter illustrated the methods used to estimate amino acid cost, and further showed the correspondence of these amino acid costs with previously described measures of amino acid cost.

\subsection{Amino acid cost estimation}

The absolute amino acid costs estimated using using the genome scale model are as expected. The $A_{ammonium}$ and $A_{sulphate}$ costs are directly proportional to the content nitrogen and sulphur in the molecule. This indicates, as expected, when either of these two nutrients are limiting the cost of the amino acid is the quantity of the limiting nutrient. The estimated $A_{glucose}$ cost is slightly different as glucose is the source of both carbon atoms and energy. When compared with literature reported cost measures the $A_{glucose}$ cost is most correlated with the Akashi and Gojobori energetic cost. Similar to both these measures of amino acid cost, small less complex amino acids such as glycine and alanine are cheaper, which larger more complex amino acids such as tryptophan and tyrosine are energetically more expensive.

The relative amino acid cost estimations represent the absolute cost of an amino acid scaled by the quantity measured in $S. cerevisiae$ biomass. The relative measures of amino acid cost show little correlation with previous measures of amino acid cost. Furthermore amino acids that are expensive based on their absolute cost, are cheaper for a relative cost. For example tryptophan has a high $A_{glucose}$ cost, but is instead much cheaper when the $R_{glucose}$ cost is considered. This reflects that expensive amino acids may have their use in the genome minimised. On the other hand lysine has one of the most expensive $R_{glucose}$ costs amino acids, but is relative inexpensive in it's $A_{glucose}$ cost. This means that a proportional increase in the requirement of lysine would have a greater effect than of tryptophan, since lysine is used more in the genome.

\subsubsection{Correlation between relative costs}

An interesting point is that the $R_{glucose}$ and $R_{ammonium}$ costs have a high correlation (Spearman R = 0.63). This leads to the possibility that optimisation of energy/carbon based costs in the proteome also results in optimisation of the nitrogen cost. The benefits of this are intuitive where selective advantages for the optimisation of a given amino acid in carbon limitation will have advantages for nitrogen limitation. If there was little or negative correlation between carbon and nitrogen based costs, then optimisation of a protein for carbon, could result in the unoptimisation for nitrogen.

\subsection{Reaction elasticity estimation}

The number of enzyme parameter needed to produce a kinetic model mean that the number of reactions included can be a magnitude smaller than that of genome scale models. The small number of reactions therefore means that the estimation of  control coefficients are limited to the small number of available reactions. For instance, at the time of writing this thesis, there exists no model where it would be possible to derive control coefficients for reactions across different parts of the cell metabolism such as both energy metabolism and amino acid. Existing genome scale models contain reactions contain the majority of the central metabolism for the organism they describe, which solves the problem of limited model size. The restriction of genome scale models is that they do not contain enzyme kinetic data, and so it is not possible to estimate metabolic control coefficients in the same way as kinetic models.

\section{Materials and Methods}

\subsubsection{Genome scale metabolic models}

The genome scale models used in this work were \emph{S.cerevisiae} iND750 \cite{duarte2004a} and \emph{E.coli} iJR904 \cite{reed2003}. Flux balance analysis was performed using the COBRA toolbox \cite{becker2007} using the lpsolve optimisation library \cite{lpsolve}.

Each flux balance analysis model is represented by an $m \times n$ sized matrix names $S$. The matrix $S$ represents the reactions in the metabolism and the metabolites that participate in the reactions. The size of of $m$ is the total number of metabolites in the model, and $n$ is the total number of reactions. The position $S_{ij}$ is the participation of metabolic $i$ in reaction $j$. Positive values of $S_{ij}$ indicate the metabolite is produced by the reaction, negative values mean the metabolite is consumed by the reaction. A zero value indicates the metabolite does not participate in the reaction.

Flux balance analysis is performed by solving the value of $v$ for the equation $S \bullet v = 0$ where $v$ represents the a vector of flux values for each reaction in $S$ and is of the size $n$. Using flux balance analysis, a value $v_{i}$ is optimised for the solution of $S \bullet v = 0$, where $v_{i}$ represents a reaction of interest. For example if $v_{i}$ represents the conversion of metabolites to the production of a new unit of biomass, then solving $v$ for the maximisation of $v_{i}$ represents maximum growth rate. Additional constraints can be placed on the optimisation of the flux balance analysis model, such as upper an lower boundaries at for reaction fluxes. By constraining a reaction to be $\geq 0$ the reaction may only proceed in the forward direction in the optimised solution.

For this analysis the biomass reaction was fixed at a constant value through constraining the upper and lower boundaries to the same value. When optimising the model the objective function was the minimisation of the flux entering the cell of a given nutrient. Performing all flux balance analysis with a fixed growth rate allows are results to be scaled to the same growth rate. Minimisation of a given nutrient flux can be used to simulate starvation of the nutrient, where the nutrient is the limiting factor for growth.

\subsubsection{Calculation of amino acid cost}

Estimation of amino acid cost was performed by changing the of $S_{ij}$ where $i$ is the amino acid and $j$ is the biomass reaction in the model. After flux balance analysis each change in $S_{ij}$ results in a corresponding change in the nutrient uptake flux for which the model is being optimised.
