\section*{Chapter summary}

In this chapter I illustrate how genome scale metabolic models can be used to estimate the importance of objects in a biological system, in this instance \emph{Saccharomyces cerevisiae}. The background describes metabolic control analysis and how this has been used to study metabolism. This is followed by a description of genome scale models and an example of their use. The results and discussion of this chapter illustrate how principle of metabolic control analysis can be applied to genome scale metabolic models to estimate the importance of individual metabolites and gene-encoded reactions.

\clearpage

\section{Introduction}

\subsection{Determing control in biological systems}

Early enzymology assumed the existence of rate limiting steps in biological pathways. The intuition is that the rate of a pathway will be the speed of the slowest step. When the rate of the slowest step increases, the rate of the whole pathway will also increase until another step becomes rate limiting. Testing this assumption Niederberger \emph{et. al} \cite{niederberger1992} found that individual up or down regulation of specific enzymes had little overall effect on tryptophan biosynthesis in \emph{Saccharomyces cerevisiae}. The rate of the pathway could instead be accelerated by increasing the quantity of five related enzymes in tandem. This research is an example control in a biological system being distributed over the system as a whole rather than focused at individual steps.

The theory of metabolic control analysis \cite{fell1992,fell1997} states that there are no rate limiting steps in biological systems, but instead that each part of the system shares a measure of more or less control. The control reaction $x$ has on any other reaction in the system can determined by measuring the effect of an up or down in reaction $x$ has on reaction $y$. In metabolic control analysis this is described as the flux control of reaction $x$ on reaction $y$. The coefficient of this control uses the symbol $C_{x}^{y}$, is described in Equation~\vref{fcc}.

\begin{equation}\label{fcc}
C_{x}^{y} = \frac{ \delta \ln x / \ln x}{\delta \ln y / \ln y}
\end{equation}

The function $\delta \ln x / \ln x$ represents the unitless effect of a small change in the reaction flux through $x$, given the original flux. The function $\delta \ln y / \ln y$ represents the corresponding unitless response in $y$ resulting from the change in $x$, given the original flux of $y$. Larger values of $C_{x}^{y}$ indicate that reaction $x$ has a high degree of control on reaction $y$. Smaller values indicate that reaction $y$ experiences only small change in flux in response to changes in $x$.

\nomenclature{$C_{x}^{y}$}{Flux control coefficient of reaction $x$ on reaction $y$}

Determining a numeric value for the relationships between reactions in a metabolic network allows properties of biological systems to be explored. In addition to increasing the secretion of commercial by products \cite{niederberger1992}, understanding the quantitative control at points in a metabolic network can lead to the development of drugs targeting the combinations of reactions with a high degree of control in the network \cite{lehar2008,hopkins2008}. Furthermore understanding the control exerted by corresponding orthologs in both the host-pathogen can allow the difference in control coefficients to be compared so that specific reactions may be targeted in host with minimal impact in the host \cite{hornberg2007}. Outside the development of commercial products or identification of drug targets, the determination of which steps are controlled through substrate concentration or changes in enzyme activity predictions can be made of the the role of gene expression in the control metabolic systems \cite{rossell2006,daran-lapujade2007}.

%TODO: read Hopkins, Lehar papers

\subsection{Metabolic modelling}

\subsubsection{Genome scale models}

Derivation of a flux control coefficient \emph{in silico} requires a kinetic model with specific information about enzyme activity for each reaction. In the case of an enzyme that follows simple Michalis-Menten kinetics parameters such as substrate affinity (K$_{m}$) and maximum enzyme reaction rate (V$_{max}$) may be sufficient. To model more complex enzymatic reactions requires larger numbers of parameter sets which require more experimental effort to derive. Furthermore even with experimentally derived kinetic parameters, the model must be tested to determination matches expected \emph{in vivo} behaviour \cite{teusink2000}. This means that kinetic model of biological processes are often small with a limited number of reaction sets, due to the difficulty in finding sufficient enzymatic data for the reactions in the system \cite{steuer2007}.
\nomenclature{K$_{m}$}{Substrate concentation where enzyme rate is at half maximum ($\frac{1}{2}$V$_{max}$)}
\nomenclature{V$_{max}$}{Maximum reactions per second per mole of enzyme}

%TODO: check Steuer paper

A stoichiometric model, such as a genome scale model, is represented by an $m \times n$ sized matrix $S$. The matrix $S$ represents the reactions in the metabolism and the metabolites that participate in the reactions. The size of of $m$ is the total number of metabolites in the model, and $n$ is the total number of reactions. The position $S_{ij}$ is the participation of metabolic $i$ in reaction $j$. Positive values of $S_{ij}$ indicate the metabolite is produced by the reaction, negative values mean the metabolite is consumed by the reaction. A zero value indicates the metabolite does not participate in the reaction.

Stoichiometric models on the other hand do not require kinetic data but instead only the connectivity between reactions, the participating metabolites and the \emph{in vivo} direction of the reaction. The growth rate of the model is simulated through of linear optimisation problem to maximise a given objective reaction such as the production of biomass defined in the combination of internal reaction and the available nutrients. Stoichiometric models are much larger than kinetic models as they require less data for construction. Large representations of an organisms metabolism have be created, described as genome scale models. The construction of genome scale metabolic model is reviewed in detail by Feist \emph{et al.} \cite{feist2009} but the process is outlined in brief here. Construction of an genome scale metabolic model begins with existing annotations of enzymes available in databases such as BRENDA \cite{chang2009} and KEGG \cite{okuda2008}, or those specific to an individual organism such as the Saccharomyces Genome Database \cite{cherry1997}. A mathematical representation of the organism is created from the derived enzyme lists and their connectivity. A biomass objective function is required to simulate the model, determined from the quantities of lipids, proteins, nucleic acids and other cofactors required to create a new cell.

The initial stages of a genome scale model construction can be automated computationally but simulation and validation of the model will require manually curation to identify gaps in the model network. Domain specific knowledge of the organism in question aid questions of the specific existence of metabolic reactions that are expected in metabolism, but may not appear in an enzyme database. This particularly the case for organisms with extreme growth environments \cite{sun2009}. The final step in the construction of a metabolic model requires validation against \emph{in vivo} phenotypes, where \emph{in silico} predictions should mimic that of \emph{in vivo} observations as closely as possible \cite{famili2003}.

The \emph{Esherichia coli} construction is almost 20 years old \cite{feist2008}, a simplified model of acetate production was described by Majewski \emph{et al.} in 1990, and the 2007 model accounts for 1260 open reading frames \cite{feist2007}. The range of genome scale models continues to expand and includes attempts to create a human cell reconstruction \cite{duarte2007}. The field of genome scale model construction unfortunately suffers from the lack of a single standard for construction and format of models. This can make model interchange and comparison difficult. There are however attempts to create a unified \emph{S. cerevisiae} model \cite{herrgard2008} using existing standards such are MIRIAM \cite{lenovere2005}, where computationally readable identifiers such a InChi codes for the representation of chemical entities \cite{coles2005}. Genome scale modelling however will not be the last computation field to suffer from a lack of agreed standards.

\subsubsection{Flux balance analysis}

Using just stoichiometric itself it is not possible to make predictions about growth rate or flux distributions, to simulate a model flux balance analysis (FBA) is performed to solve the model so that $S v = 0$ where $v$ represents the a vector of flux values for each reaction in $S$ and is of the size $n$. Using flux balance analysis, a value $v_{i}$ is optimised for the solution of $S v = 0$, where $v_{i}$ represents a reaction of interest. For example if $v_{i}$ represents the conversion of metabolites to the production of a new unit of biomass, then solving $v$ for the maximisation of $v_{i}$ represents maximum growth rate. Additional constraints can be placed on the optimisation of the flux balance analysis model, such as upper an lower boundaries at for reaction fluxes. By constraining a reaction to be $\geq 0$ the reaction may only proceed in the forward direction in the optimised solution.
\nomenclature{FBA}{Flux Balance Analysis}

Simulation of a genome scale model for a given objective function, the solution of a genome scale model is determining the combination of network fluxes that maximise of minimise the objective. This is a linear optimisation problem and requires the use of a specific linear programming library. In addition to a mathematical library tools exist that provide specific help for the use and manipulation of genome scale metabolic models \cite{becker2007}. Optimisation of a metabolic can be carried out by linear programming of the objective function, additional methods may be used when making perturbations to metabolic models. Minimisation of metabolic adjustment \cite{segre2002,burgard2003} optimises for the objective function, while minimising the sum change in flux adjustment between the unperturbed and perturbed models. A related approach minimises the number of on-off fluxes changes, whereby the number of reactions that switch between active or inactive between solutions is minimised \cite{shlomi2005}.

%TODO: read MOMA, MOMA update, and ROOM

\subsubsection{Predicting essential genes using genome scale models and flux balance analysis}

The first section of this introduction described the importance of quantitative metabolic control analysis in determining the points of control in biological systems. The larger size of genome scale models would mean that quantitative analysis of this type would yield much greater datasets than similar analysis in smaller kinetic models.

The estimation of control coefficients using stoichiometric models is difficult as the flux the through the reaction is determined by linear optimisation and any constraints defined in the model, rather than enzyme kinetic parameters. The classic approach to estimating the importance of a reaction in a genome scale model is to remove the reaction from the model, and compared the resulting solution with the original. \emph{in silico} single gene deletions were calculated \emph{S.cerevisiae} and compared with \emph{in vivo} estimations, where the correspondence was \~85\% depending on the simulated media \cite{famili2003,forster2003}. Reactions with buffer the effects of each other can be determined by comparing all possible double deletions and finding the pairs with no phenotypic effect when deleted individually, but predict a sick or lethal phenotype when deleted simultaneously \cite{harrison2007}. This can be further extended with multiple reaction knockouts \cite{deutscher2008}. Reactions which may be active in a specific range of conditions can be assessed through deletion of reactions in a wide range of simulated conditions, where a deletion may result in a fitness effect in only a small range of biologically feasible conditions \cite{papp2004}.

Using gene deletion as an estimate of gene control depends on the interpretation of resulting phenotypic effect. For reaction deletion may be considered to confer a fitness effect if there is a reduction below a certain threshold \cite{pal2006}, while a reaction may be considered essential removal results in no possible solution for the given objective \cite{becker2008}, removal results in a fitness effect of less than half of the original objective flux \cite{papp2004}, or from interpretation of a ratio of knockout to wild type growth rates \cite{wang2009}.

The work described above focused on estimating the importance of a reaction to a biological objective function, such as growth. The use of genome scale metabolic models can also be used to estimate the importance of metabolites to an objective function based on the use of shadow prices. In economics the shadow price, for example of a sold product, characterises the size of the  effect of a change product price on the market in which it is sold. Varma \emph{et al.} \cite{varma1993} simulation this approach in the estimation of the shadow price of oxygen on a stoichiometric model of \emph{E. coli} energy metabolism. Through simulating a decrease in the availability of oxygen the researchers identified how the shadow prices of the redox carriers changed with the availability of oxygen. With the decreasing availability of oxygen and the corresponding rise in its price, the price of ethanol for energy production decreases as it becomes the cheapest electron acceptor for eliminating redox.

\subsection{Estimating amino acid biosynthetic cost}

The article by Varma \emph{et al.} highlights how systems biology models can be used to estimate the importance of a set of metabolites to cell growth. Related to this is the question of how much it costs the cell to synthesise and use amino acids. Craig \& Weber \cite{craig1998} estimated the cost of the twenty amino acids as the sum of how many high energy phosphate bonds (e.g. ATP) and reducing molecules (e.g. NADPH) are diverted from the available energy pool in the synthesis of each amino acid from glucose. The aim of deriving these costs was to estimate the importance of amino acid cost the evolution of a set of \emph{E. coli} proteins.

Akashi \& Gojobori \cite{akashi2002} explored a similar approach calculating cost on a range different energy sources to estimate the importance of amino acid cost in predicted \emph{E. coli} and \emph{B. subtilis} gene expression. Heizer \emph{et al.} \cite{heizer2006} expanded this estimating amino acid cost for inorganic sources in their analysis of the genomes of four prokaryotic species including photoautotrophs. These estimates of cost are based on the curation of the metabolic network to account for the expenditure of high energy molecules during synthesis. Seligmann argued \cite{seligmann2003} that the cost of amino acid biosynthesis must take into account more than just the energy required for synthesis. Seligmann used molecular weight as a proxy amino acid cost arguing that this may take into account investments in synthesising larger more complex amino acids. Using molecular weight also has the advantage where the metabolic network in not required to curate the number of high energy molecules used during synthesis. Table~\ref{table:literature_costs} highlights the various measures of amino acid cost.

%TODO: Check Akashi paper re energy sources

%TODO: Add wager costs to intro

%TODO: How is Wagner's cost different

\begin{table}
\begin{footnotesize}
  \begin{tabular}{ p{1cm} *{6}{p{1.9cm}} }
                                                                                \toprule
          & A\&G   & C\&W   & C\&W  & Wager        & Wagner      & Seligmann \\
          & Energy & Energy & Steps & Fermentative & Respiratory & Weight    \\ \midrule
      ala & 11.7   & 12.5   & 1     & 2            & 14.5        & 89.1      \\
      arg & 27.3   & 18.5   & 10    & 13           & 20.5        & 174.2     \\
      asn & 14.7   & 4      & 1     & 6            & 18.5        & 132.1     \\
      asp & 12.7   & 1      & 1     & 3            & 15.5        & 133.1     \\
      cys & 24.7   & 24.5   & 9     & 13           & 26.5        & 121.2     \\
      gln & 16.3   & 9.5    & 2     & 3            & 10.5        & 146.2     \\
      glu & 15.3   & 8.5    & 1     & 2            & 9.5         & 147.1     \\
      gly & 11.7   & 14.5   & 4     & 1            & 14.5        & 75.1      \\
      his & 38.3   & 33     & 1     & 5            & 29          & 155.2     \\
      ile & 32.3   & 20     & 11    & 14           & 38          & 131.2     \\
      leu & 27.3   & 33     & 7     & 4            & 37          & 131.2     \\
      lys & 30.3   & 18.5   & 10    & 12           & 36          & 146.2     \\
      met & 34.3   & 18.5   & 9     & 24           & 36.5        & 149.2     \\
      phe & 52     & 63     & 9     & 10           & 61          & 165.2     \\
      pro & 20.3   & 12.5   & 4     & 7            & 14.5        & 115.1     \\
      ser & 11.7   & 15     & 3     & 1            & 14.5        & 105.1     \\
      thr & 18.7   & 6      & 6     & 9            & 21.5        & 119.1     \\
      trp & 74.3   & 78.5   & 12    & 14           & 75.5        & 204.2     \\
      tyr & 50     & 56.5   & 9     & 8            & 59          & 181.2     \\
      val & 23.3   & 25     & 4     & 4            & 29          & 117.2     \\ \bottomrule
  \end{tabular}
\end{footnotesize}
\caption[Amino acid costs described in the literature]{Amino acid costs described in the literature. The Akashi \& Gojobori \cite{akashi2002}, Craig \& Weber energy \cite{craig1998}, and the two Wagner \cite{wagner2005} data sets are based on the curation of the number of high-energy molecules used during synthesis, where a defined ratio is used to convert them into a single measures: usually ATP. The Craig \& Weber `steps' measure \cite{craig1998} is based on the number of the number of biosynthetic steps between central metabolism and the produced amino acid. Molecular weight is in Daltons.}
\label{table:literature_costs}
\end{table}

\subsection{Results summary}

The results presented in this chapter are split into two sections. The first describes an approach to estimating amino acid cost focusing on the \emph{S. cerevisiae} genome scale metabolic model. The approach used to estimate amino acid cost is similar to that described by Varma \emph{et al.} in their estimation of the shadow prices in \emph{E.coli} energy metabolism. The estimated costs are compared to the other estimates of amino acid cost that appear in the literature described above.

The second part of results section describes an experimental approach to estimate the cost of a gene. This is again in \emph{S. cerevisiae} using the genome scale metabolic model. This approach performs a type of robustness analysis on the gene encoded metabolic function. This analysis produces a robustness distribution describing how increasing and decreasing metabolic rate for an function affect the \emph{in silico} model fitness.

\clearpage

\section{Materials and Methods}

\subsection{Flux balance analysis}

The genome scale models used in this work were \emph{S.cerevisiae} iND750 \cite{duarte2004a} and \emph{E.coli} iJR904 \cite{reed2003}. Flux balance analysis was performed using the COBRA toolbox \cite{becker2007} using the lpsolve optimisation library \cite{lpsolve}.
\nomenclature{COBRA}{Flux balance analysis toolbox for Matlab (see \cite{becker2007})}
\nomenclature{iND750}{\emph{Saccharomyces cerevisiae} genome scale model (see \cite{duarte2004a})}
\nomenclature{iJR904}{\emph{Esherichia coli} genome scale model (see \cite{reed2003})}

The biomass reaction was fixed at a constant value through constraining the upper and lower boundaries to the same value. When optimising the model the objective function was the minimisation of the given nutrient flux entering the cell. Performing all flux balance analysis with a fixed growth rate allows are results to be scaled to the same growth rate and simulates experimental chemostat conditons where growth is fixed and the entry of a given nutrient is restricted (see Castrillo \emph{et al.} \cite{castrillo2007}). The minimisation of a given nutrient as the \emph{in silico} objective mimics this condition where the nutrient is the limiting factor for growth.

\subsection{Flux balance analysis estimation of amino acid cost}

Estimation of amino acid cost was performed by making a percentage change to the requirement of amino acid at position $S_{ij}$ in the model matrix, where $i$ is the amino acid and $j$ is the biomass reaction. When the model is optimised each change in $S_{ij}$ results in a corresponding change in the nutrient uptake flux for which the model is being optimised. The slope between the change in amino acid requirements and the corresponding change nutrient uptake flux was taken as the relative cost of the amino acid. Division of the absolute cost by the original requirement results in an absolute estimate of amino acid cost.

\subsection{Flux balance analysis estimation of gene cost}

\subsubsection{Determination of gene to reaction association}

A subset of the reactions in the \emph{S. cerevisiae} metabolic model have a gene association. This annotation defines if there is one or genes known to encode an enzyme which catalyses the reaction. This association can include multimeric enzymes where multiple gene products form an enzyme complex. The relationship of gene to reaction is not always 1-to-1 where multiple genes products catalyse the same reaction, or a single gene product catalyses multiple reactions. 

The COBRA toolbox used in the analysis defines a matrix specifying which genes are associated with which reactions. The matrix is size 750 by 1266, where there are 750 annotated genes and 1266 reactions. The matrix is sparse as not all reactions have an associated gene. This matrix was parsed to determine rows which contain only a single 1 value indicating that only a single gene catalyses the reaction. The matrix was also parsed to determine if this value was also the only value appearing in the corresponding column, indicating the gene catalyses no other reactions. This method was used to determine a list of reactions that have a 1:1 gene annotation.

\subsubsection{Analysis of reaction variability}

The yeast model was then optimised for the given set of conditions. Each condition solution was examined to determine if the reaction participated in the solution, i.e. a non zero flux. Reactions at the maximum allowable flux in solution i.e. -1000 or 1000 were also determined. The cost of either of these types of reactions was not determined as they do not participate, or are at the edge of the solution space.

The flux variability of each of the remaining reactions was assessed. The aim was to determine if the reaction could vary in the solution space give the fixed growth rate requirement. To determine the variability of each of the remaining reactions, in each case the model was re-optimised with the objective function to minimise the absolute reaction flux. If the reaction could not be minimised, this indicated the reaction was constrained in the solution space, and no cost measure could be estimated.

\subsubsection{Reaction cost estimation}

To estimate the gene cost of the remaining reactions, the flux through each reaction was fixed at the original flux value in the initial optimisation but reduced by a small constant. The model was then optimised for a given nutrient limitation. Optimisation was performed using minimisation of metabolic adjustment (MOMA). The log change in the objective flux was divided by the log change in reaction flux to give the cost for the reaction in question.

\nomenclature{MOMA}{Minimisation of Metabolic Adjustment}

%TODO: is division correct?
%TODO: what units?

\clearpage

\section{Results}

\subsection{The \emph{S. cerevisiae} iND750 genome scale model}

\subsubsection{Numbers of reactions and genes}

The \emph{S.cerevisiae} iND750 model contains 1266 reactions and is annotated with 750 genes. Table~\vref{figure:gene_associations} illustrates the number and types of associations of genes to reactions in the model. Each reaction in the model can be categorised based on the type gene associations.

\begin{table}
  \centering
  \begin{tabular}{l r}
                                                                   \toprule
    Description                                      & Number   \\ \midrule
    Genes                                            & 750      \\
    Reactions                                        & 1266     \\
    Reactions with at least one associated gene      & 810      \\
    Reactions with a single gene associated          & 579      \\
    One-to-one association between reaction and gene & 262      \\ \bottomrule
  \end{tabular}
  \caption[Gene associations in the \emph{S. cerevisiae} iND750 model]{Comparison of numbers and types of associations between genes and reactions in the \emph{S. cerevisiae} iND750 model. Each association category is a subset of the above category.}
  \label{figure:gene_associations}
\end{table}

Category A describes reactions with many-to-many gene associations, where one gene may may be associated with multiple reactions, and one reaction may be associated with multiple genes. This category includes reactions that may be catalysed by paralogs and/or multimeric enzymes, and genes that are associated with multiple reactions. Category B one-to-many gene associations where one one may associated with many reactions, but each reaction has only one gene association. This category excludes paralogs and multimeric enzymes, but still includes enzymes associated with multiple reactions. Category C represents one-to-one gene associations, where each gene is associated with a single reaction, and each reaction has only a single gene association. This category represents 262 genes which have only a single point of the effect defined in the model.

\subsubsection{Compostion of biomass requirement}

The biomass equation is estimated requirement of metabolites for growth. In flux balance analysis the biomass equation can be used objective function to simulate the level of growth given the available basic input nutrients. The quantities of nutrients in the biomass reaction are organism specific and determined from millimole quantities of each quantities per gram of dry weight biomass (mmol $^-1$ gDW $^-1$).
 
Table \vref{table:biomass_requirements} outlines the required molecule for the \emph{S. cerevisiae} iND750 model. The requirements include fats such as triglyceride and ergosterol, sugars such as glycogen and mannan, as well as smaller metabolites sulphate and water. The genome, transcriptome, and proteome content of the cell is included as the quantities of nucleotides and amino acids.

\begin{table}
  \centering
  \begin{footnotesize}
  \begin{tabular}{l l r}
                                                                     \toprule
    Type           & Molecule                       & Requirement \\ \midrule
    Nucleic Acid   & dAMP                           &  0.00360    \\
                   & dCMP                           &  0.00240    \\
                   & dGMP                           &  0.00240    \\
                   & dTMP                           &  0.00360    \\
                   & AMP                            &  0.04600    \\
                   & CMP                            &  0.04470    \\
                   & GMP                            &  0.04600    \\
                   & UMP                            &  0.05990    \\ \midrule
      Amino Acid   & Alanine                        &  0.45880    \\
                   & Arginine                       &  0.16070    \\
                   & Asparagine                     &  0.10170    \\
                   & Aspartate                      &  0.29750    \\
                   & Cysteine                       &  0.00660    \\
                   & Glutamate                      &  0.30180    \\
                   & Glutamine                      &  0.10540    \\
                   & Glycine                        &  0.29040    \\
                   & Histidine                      &  0.06630    \\
                   & Isoleucine                     &  0.19270    \\
                   & Leucine                        &  0.29640    \\
                   & Lysine                         &  0.28620    \\
                   & Methionine                     &  0.05070    \\
                   & Phenylalanine                  &  0.13390    \\
                   & Proline                        &  0.16470    \\
                   & Serine                         &  0.18540    \\
                   & Threonine                      &  0.19140    \\
                   & Tryptophan                     &  0.02840    \\
                   & Tyrosine                       &  0.10200    \\
                   & Valine                         &  0.26460    \\ \midrule
      Carbohydrate & 1 3 beta D Glucan              &  1.13480    \\
                   & Mannan                         &  0.80790    \\
                   & Glycogen                       &  0.51850    \\
                   & Trehalose                      &  0.02340    \\ \midrule
      Lipid        & Ergosterol                     &  0.00070    \\
                   & Triglyceride                   &  0.00007    \\
                   & Zymosterol                     &  0.00150    \\
      Phospholipid & Phosphatidyl 1D Myo Inositol   &  0.00005    \\
                   & Phosphatidate                  &  0.00001    \\
                   & Phosphatidylcholine            &  0.00006    \\
                   & Phosphatidylethanolamine       &  0.00005    \\
                   & Phosphatidylserine             &  0.00002    \\ \midrule
      Small        & ATP                            & 59.27600    \\
                   & H$_2$O                         & 59.27600    \\
                   & Sulphate                       &  0.02000    \\ \bottomrule
  \end{tabular}
  \end{footnotesize}
  \caption[Biomass requirements of \emph{S. cerevisiae} iND750 model]{Biomass requirements of \emph{S. cerevisiae} iND750 model. Each molecule is categorised by type. The requirement of each molecule is in millimoles per gram of dry weight biomass (mmol$^{-1}$ gDW$^{-1}$). The calculation of the a biomass equation is outlined by Feist \emph{et al.} \cite{feist2009}, the \emph{S. cerevisiae} model construction is described by Duarte \emph{et al.} \cite{duarte2004a}.}
  \label{table:biomass_requirements}
\end{table}

\subsection{A novel approach using genome scale models to estimating amino acid cost}\label{section:amino_acid_cost_estimation}

As discussed, metabolic control analysis can be used to estimate the importance of an object in a kinetic model. In this section I will show similar principles of can be used to estimate the sensitivity of growth rate to the changes in amino acid requirements in biomass. The biomass equation in a flux balance analysis model describes the estimated requirements and stoichiometry of metabolites required for the production of a biomass, which is equivalent to growth. For instance in the biomass reaction in the \emph{S. cerevisiae} model, the requirements include fats such as triglyceride and ergosterol, sugars such as glucose and mannose, as well as smaller metabolites such as sulphate and water. The stoichiometry of the biomass equation is derived from the dry weight biomass of the \emph{S. cerevisiae} cell \cite{duarte2004a}. The protein content of a yeast cell is also included in the biomass equation as the twenty individual amino acids that make up protein, with the inclusion of theses amino acids in the biomass equation, the reaction stoichiometries of each can be manipulated to estimate the effect on biomass production.

\begin{figure}
  \centering
  \subfloat[Cost estimation in glucose limited conditions]{
    \label{figure:curve_estimation:schematic}
    \includegraphics*[width=6cm]{cost_schematic.eps}
  }
  \hfill
  \subfloat[Supply-demand elasticity for tryptophan]{
    \label{figure:curve_estimation:curve}
    \includegraphics*[width=6cm]{curve_estimation.eps}
  }
  \hfill
  \subfloat[Comparison of costs for glycine, histidine, and tryptophan]{
    \label{figure:curve_estimation:costs}
    \includegraphics*[width=7cm]{comparison_of_costs.eps}
  }
  \hfill
  \caption[Methods used for estimation of amino acid cost]{Outline of the methods used for the estimation of amino acid cost. Figure~\ref{figure:curve_estimation:schematic} shows a schematic representation of estimating amino cost in glucose limited conditions using a genome scale model. The growth rate of the model is fixed at a constant value so that costs can be compared  between different environments and models. The objective function of the model is then set to be the nutrient in question, which in this example is glucose. The glucose flux entering the model is therefore minimised when optimising the model. All other nutrient entering the model, for example ammonia are then unbounded. Figure~\ref{figure:curve_estimation:curve} the glucose and ammonium slopes for tryptophan when the requirement for tryptophan is altered in the biomass reaction. Figure~\ref{figure:curve_estimation:costs} compares the glucose and ammonium costs for tryptophan, glycine and histidine. }
  \label{figure:curve_estimation}
\end{figure}

Figure~\vref{figure:curve_estimation} illustrates the method used to estimate the cost of an amino acid with a genome scale metabolic model. Figure~\ref{figure:curve_estimation:schematic} illustrates the genome scale model usage to estimate amino acid cost using the model to simulate a glucose limited environment. The minimal and maximal boundary on the production of biomass is fixed at a constant value, for example 0.3 hr$^{-1}$. Using a constant growth rate value for amino acid estimation means that all amino acid cost estimations are scaled by the same growth rate. The objective function of the model is then set to be the minimisation of cellular entry of the nutrient for which the amino acid costs are being estimated, which in Figure~\ref{figure:curve_estimation:schematic} is glucose. All all other nutrient exchange fluxes were set to have a however boundary of -10000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$, the aim of this is to set all other nutrients to be non limiting on simulated growth. For the \emph{S. cerevisiae} iND750 \cite{duarte2004a} genome scale model the default exchange reactions when simulating growth are glucose, ammonium, sulphate, phosphate, oxygen and water. Each exchange flux represented the only source for an essential nutrient, and therefore when estimating the glucose cost of an amino acid there were no other high energy sugars available, or when estimating the ammonium cost there were no other nitrogen sources entering the cell. Model optimisation then found what the minimum required amount of nutrient intake flux was required to maintain the growth rate at which the model was fixed. Four nutrients were considered in the analysis: glucose, ammonium, sulphate, and phosphate.

Tryptophan has a double ring side chain which is large and complex relative to other amino acids. This side chain also has two nitrogen atoms. Figure~\ref{figure:curve_estimation:curve} shows the slope for supply and demand elasticity for tryptophan in glucose and ammonium limiting conditions. The slope for each environment is produced by altering the requirement of tryptophan in model biomass reaction. The stoichiometry of tryptophan is changed with a slight increase or small decrease in requirement of the range of $\pm$0.0002\%. The result of this change is then measured in the effect on the uptake flux in question. 

Figure~\ref{figure:curve_estimation:curve} shows two different response slopes for changes in tryptophan requirement on the ammonium and glucose uptake fluxes. The slope of the relationship of tryptophan on the respective uptake flux was used as the cost of that amino acid in the given nutrient limitation. This cost is the value of $m$ in the linear relationship $y = mx + c$ where $x$ is tryptophan uptake requirement and $y$ is resulting nutrient uptake flux. This cost was is based on relative changes to amino acid requirement, where the  amino acid stoichiometry is multiplied by a small percentage shown on the x-axis of Figure~\vref{figure:curve_estimation:curve}. This relative estimation of amino cost can be rescaled to estimate a cost based on absolute changes in amino acid requirement by dividing the estimated relative cost by the original requirement of the amino acid in the biomass reaction. The rescaling provides an estimate of amino acid cost based on absolute changes in amino acid requirement.

One further consideration is the estimated amino acid cost are a function of the growth rate at which they were estimated. For instance the absolute cost of tryptophan estimated at 0.2 hr$^{-1}$ is twice the cost of tryptophan estimated at 0.1 hr$^{-1}$. To control for growth rate when estimating amino acid cost, the costs were estimated at a three levels of feasible yeast growth rates 0.1 hr$^{-1}$, 0.2 hr$^{-1}$ and 0.3 hr$^{-1}$. Each amino acid cost is then rescaled to be independent of the growth rate at which is was estimated. The amino acid costs presented in this work are those estimated at the 0.3 hr$^{-1}$ growth rate, a level of growth where \emph{S. cerevisiae} uses respiratory energy production \cite{famili2003}.

Figure~\ref{figure:curve_estimation:costs} compares the estimated absolute costs for tryptophan, histidine and glycine in two of the examined conditions, glucose and ammonium limitation. Tryptophan which is an expensive amino acid in glucose limitation, is slightly cheaper in nitrogen limitation. In the reverse of this, histidine which is a relatively cheap amino acid in glucose limitation is correspondingly more expensive, containing three nitrogen atoms, in ammonium limitation. Glycine, a small amino acid with a single hydrogen side chain is relatively cheap in both nutrient limitations.

The relative costs are derived from the slope in the unitless percentage change in the amino acid requirement, and the corresponding mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ response in nutrient intake flux. When divided by the hr$^{-1}$ growth rate at which the cost was estimated, the units are mmol$^{-1}$ gDW$^{-1}$. This represents the change in nutrient uptake flux given the factional change in amino acid requirement in biomass. The absolute cost is derived from the relative cost by division by the original mmol$^{-1}$ gDW$^{-1}$ requirement of the amino acid. The absolute amino acid is therefore unitless and represents what a mmol$^{-1}$ gDW$^{-1}$ increase in amino acid requirement has the mmol$^{-1}$ gDW$^{-1}$ uptake flux of the given nutrient.

\subsubsection{Estimated amino acid costs in glucose, ammonium, and sulphate limitation}

Relative and absolute amino acid costs where estimated for all twenty amino acids in four simulated nutrient limiting conditions: glucose, ammonium, sulphate, and phosphate. For clarity costs are described as either $R_{nutrient}$ or $A_{nutrient}$ for relative and absolute costs respectively. The relative and absolute estimated costs for each of these amino acids is illustrated in Figure~\vref{figure:costs_dendrogram_dotplot} along with amino acid costs described in the literature. The estimated amino acid cost are provided in Table~\vref{table:estimated_costs}.

\begin{table}
\begin{footnotesize}
  \begin{tabular}{ p{1cm} *{6}{p{1.9cm}} }
  \toprule
      & $A_{glucose}$ & $A_{nitrogen}$ & $A_{sulphur}$ & $R_{glucose} $ & $R_{nitrogen}$ & $R_{sulphur}$ \\ \midrule
      ala & 11.7 & 12.5 & 1  & 0 & 0 & 0 \\
      arg & 27.3 & 18.5 & 10 & 0 & 0 & 0 \\
      asn & 14.7 & 4    & 1  & 0 & 0 & 0 \\
      asp & 12.7 & 1    & 1  & 0 & 0 & 0 \\
      cys & 24.7 & 24.5 & 9  & 0 & 0 & 0 \\
      gln & 16.3 & 9.5  & 2  & 0 & 0 & 0 \\
      glu & 15.3 & 8.5  & 1  & 0 & 0 & 0 \\
      gly & 11.7 & 14.5 & 4  & 0 & 0 & 0 \\
      his & 38.3 & 33   & 1  & 0 & 0 & 0 \\
      ile & 32.3 & 20   & 11 & 0 & 0 & 0 \\
      leu & 27.3 & 33   & 7  & 0 & 0 & 0 \\
      lys & 30.3 & 18.5 & 10 & 0 & 0 & 0 \\
      met & 34.3 & 18.5 & 9  & 0 & 0 & 0 \\
      phe & 52   & 63   & 9  & 0 & 0 & 0 \\
      pro & 20.3 & 12.5 & 4  & 0 & 0 & 0 \\
      ser & 11.7 & 15   & 3  & 0 & 0 & 0 \\
      thr & 18.7 & 6    & 6  & 0 & 0 & 0 \\
      trp & 74.3 & 78.5 & 12 & 0 & 0 & 0 \\
      tyr & 50   & 56.5 & 9  & 0 & 0 & 0 \\
      val & 0    & 0    & 0  & 0 & 0 & 0 \\ \bottomrule
  \end{tabular}
\end{footnotesize}
\caption[\emph{S. cerevisiae} estimated absolute and relative amino acid costs]{The units of the absolute costs ($A_{uptake}$) are mmol of uptake nutrient per gram dry biomass per hour. The relative costs ($R_{uptake}$) are unitless. All costs are measured in the iND750 \emph{S. cerevisiae} model using the COBRA toolbox.} 
\label{table:estimated_costs}
\end{table}

\begin{sidewaysfigure}
\centering
\includegraphics*[angle=0,height=13cm]{costs_dendrogram_dotplot.eps}
\caption[Comparison of amino acid cost estimates]{Amino acid cost estimates are shown as bar charts on the left hand side. Each bar chart axis shows the minimum and maximum value of each cost type, rounded to three significant figures. The correlations between costs are compared in a dendrogram on the right hand side computed by complete agglomerative clustering using Spearman's Rank correlation distance between data sets. The illustrated data is shown in Table~\vref{table:literature_costs}}
\label{figure:costs_dendrogram_dotplot}
\end{sidewaysfigure}

The $A_{glucose}$ costs is highly correlated with previous measures of amino acid cost. The energetic cost derived by Akashi and Gojobori \cite{akashi2002} is the most correlated with $A_{glucose}$ (Spearman R = 0.94), but $A_{glucose}$ is also correlated (Spearman R > 0.7) with Craig and Weber's energetic cost \cite{craig1998}, Wagner's respiratory cost \cite{wagner2005} and molecular weight \cite{seligmann2003}. This indicates that the $A_{glucose}$ cost captures the energetic cost of synthesising an amino acid. The $A_{ammonium}$ and $A_{sulphate}$ costs are proportional to the nutrient content of the amino acid. The $A_{ammonium}$ costs show a weak correlation (Spearman R 0.5 - 0.6) with the energetic cost measures: Akashi \& Gojobori, Wagner fermentative, and molecular weight.

The $R_{glucose}$ cost shows little correlation with previous measures of amino acid cost. The cost that it is most correlated with is the Wagner fermentative growth (Spearman R = -0.16). The $R_{glucose}$ and $R_{ammonium}$ costs are however correlated with each other (Spearman R = 0.63). Comparing the derived relative and absolute costs, the absolute costs reflect the per molecule biosynthetic, while the relative costs reflect the absolute cost scaled by the use of the amino acid in the genome. For example, the $A_{sulphate}$ costs of methionine and cysteine are both 1, which reflects the single sulphur atom in each amino acid. The $R_{sulphate}$ cost for methionine is however much greater than that of cysteine, as methionine is used proportionally more in the proteome.

\subsubsection{Comparison of amino acid costs in \emph{S. cerevisiae} and \emph{E. coli}}

$A_{glucose}$ and $R_{glucose}$ were calculated for comparison using the \emph{Escherichia coli iJR904} genome scale model. Figure~\vref{figure:amino_acid_costs_scatterplot} compares both relative and absolute costs for \emph{E. coli} and \emph{S. cerevisiae} models with two costs described in the literature: molecular weight \cite{seligmann2003} and number of ATP and NAPH molecules \cite{akashi2002}.

\begin{figure}
\centering
\includegraphics*[width=13cm]{amino_acid_costs_scatterplot.eps}
\caption[Comparison of the genome scale model derived cost data sets.]{Comparison of estimated amino acid cost with number of ATP and NADPH molecules used in synthesis (left), and molecular weight (right). On the y axis are the amino acid costs estimated using flux balance analysis. Both \emph{S. cerevisiae} and \emph{E. coli} measures are included to illustrate correlation of cost estimates between species. Estimated cost value
es have been been rescaled around their mean value to allow comparisons across species. The trends in each plot are drawn using `loess' smoothing.}
\label{figure:amino_acid_costs_scatterplot}
\end{figure}

Figure~\ref{figure:amino_acid_costs_scatterplot} shows that the both the absolute costs show a small amount of variance in their correlation the example literature described cost measures. Both \emph{E. coli} and \emph{S. cerevisiae} absolute costs measures are particularly well correlated with the Akashi and Gojobori described measure of cost \cite{akashi2002}, however the relationship shows a parabolic relationship. In comparison molecular weight shows a linear relationship with the absolute measures of costs, but shows a greater degree of variation.

The comparison of relative measures of cost illustrates the small degree of correlation with previous estimates of cost. For the both the molecular weight and ATP+NADH cost estimates, neither \emph{E. coli} or \emph{S. cerevisiae} relative cost estimates show any relationship, and furthermore the figure illustrates the disparity in estimated costs between the species models.

\subsection{A novel approach to estimating gene cost}

\subsubsection{Estimated reaction constraint in glucose and ammonium limitation}

The above described 262 single-gene encoded reactions was assessed in various solutions spaces. The simulations used were glucose and ammonia nutrient limitations in using either a respiratory or fermentative growth model. All simulations were performed at a fixed growth rate of 0.3$hr^{-1}$. The use of each gene in the resulting model optimised solution space can be divided in four types based on the flux distribution in the model. Figure~\vref{figure:reaction_constraints} illustrates these types.

\begin{figure}
\centering
\includegraphics*[width=10cm]{reaction_constraints.eps}
\caption[Gene use in model optimised solution space]{The use of 1:1 gene associated reactions in four environmental solution spaces. Two types of nutrient limitation are shown: glucose and ammonia. Each limitation is explored using either a yeast respiratory, or fermentative growth. Four types of reaction uses are described: at maximum, constrained, variable, or zero flux. Maximum reactions are at the positive or negative limit of the allowable flux for the model optimisation. Constrained reactions are fixed at the current flux and absolute flux cannot be reduced to satisfy the model growth rate. Variable reactions' flux rate can be reduced whilst still maintaining the model growth rate. Zero flux reactions are not used in the given solution. All reaction estimates are performed at a fixed growth rate of 0.3hr$^{-1}$. }
\label{figure:reaction_constraints}
\end{figure}

\paragraph{Reaction rate at maximum}

The reactions are at the allowable limit for any reaction in the solution. The flux through the reaction the reaction is either at -1000/1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$. The number of these types of reactions was limited where only six total exist across all the model solutions.

\paragraph{Constrained reaction rate}

These reactions are constrained in their use in the given solution. The reaction is used in the solution, but the absolute reaction rate cannot be reduced whilst still maintaining maintaining a viable solution for a model growth rate at 0.3hr$^{-1}$ The number of these reactions ranged from 58 to 63 reactions.

\paragraph{Variable reaction rate}

These reactions are used in the solution, and the absolute reaction flux can be reduced whilst still maintaining the fixed growth rate. The number of these reactions ranged from 30 to 42.

\paragraph{Reaction unused}

This type of reaction was not used in the solution and had a reaction rate of zero. This is the largest proportion of reactions across all simulations ranging from 163 to 166.

\clearpage

\section{Discussion}

This chapter illustrated the methods used to estimate amino acid cost, and further showed the correspondence of these amino acid costs with previously described measures of amino acid cost.

\subsection{Flux balance analysis predicts amino acid cost}

The absolute amino acid costs estimated using using the genome scale model are as expected. The $A_{ammonium}$ and $A_{sulphate}$ costs are directly proportional to the content nitrogen and sulphur in the molecule. This indicates, as expected, when either of these two nutrients are limiting the cost of the amino acid is the quantity of the limiting nutrient. The estimated $A_{glucose}$ cost is slightly different as glucose is the source of both carbon atoms and energy. When compared with literature reported cost measures the $A_{glucose}$ cost is most correlated with the Akashi and Gojobori energetic cost. Similar to both these measures of amino acid cost, small less complex amino acids such as glycine and alanine are cheaper, which larger more complex amino acids such as tryptophan and tyrosine are energetically more expensive.

The relative amino acid cost estimations represent the absolute cost of an amino acid scaled by the quantity measured in $S. cerevisiae$ biomass. The relative measures of amino acid cost show little correlation with previous measures of amino acid cost. Furthermore amino acids that are expensive based on their absolute cost, are cheaper for a relative cost. For example tryptophan has a high $A_{glucose}$ cost, but is instead much cheaper when the $R_{glucose}$ cost is considered. This reflects that expensive amino acids may have their use in the genome minimised. On the other hand lysine has one of the most expensive $R_{glucose}$ costs amino acids, but is relative inexpensive in it's $A_{glucose}$ cost. This means that a proportional increase in the requirement of lysine would have a greater effect than of tryptophan, since lysine is used more in the genome.

\subsubsection{Correlation $R_{glucose}$ and $R_{ammonium}$ indicates genome optimisation}

An interesting point is that the $R_{glucose}$ and $R_{ammonium}$ costs have a high correlation (Spearman R = 0.63). This leads to the possibility that optimisation of energy/carbon based costs in the proteome also results in optimisation of the nitrogen cost. The benefits of this are intuitive where selective advantages for the optimisation of a given amino acid in carbon limitation will have advantages for nitrogen limitation. If there was little or negative correlation between carbon and nitrogen based costs, then optimisation of a protein for carbon, could result in the unoptimisation for nitrogen.

\subsubsection{Atomic material content as a biosynthetic cost}

Amino acid synthesis will require energy to power each step in the metabolic network, in addition to the energetic cost the molecules part of the amino acid structure may also be considered a cost of synthesis. Mazel \& Marliere \cite{mazel1989} showed that \emph{Calothrix} proteins expressed in sulphur limiting conditions are depleted for sulphur containing cysteine and methionine amino acids. This work indicating the atomic content of an amino acid should also be taken into consideration when attempting to define a measure of cost. Work by both Baudoin-Cornu \emph{et al.} \cite{baudoin2001} and Bragg \emph{et al.}	\cite{bragg2006} used atomic content as a measure of amino acid cost when examining protein content. When considering the evolution role of an amino acid cost, the role of energy should also be coupled with the availability of nutrients. Any limitation on a specific nutrient will play as an important role as that of energetic limitation.

\subsection{Predicting the cost of a gene}

The number of enzyme parameter needed to produce a kinetic model mean that the number of reactions included can be a magnitude smaller than that of genome scale models. The small number of reactions therefore means that the estimation of  control coefficients are limited to the small number of available reactions. For instance, at the time of writing this thesis, there exists no model where it would be possible to derive control coefficients for reactions across different parts of the cell metabolism such as both energy metabolism and amino acid. Existing genome scale models contain reactions contain the majority of the central metabolism for the organism they describe, which solves the problem of limited model size. The restriction of genome scale models is that they do not contain enzyme kinetic data, and so it is not possible to estimate metabolic control coefficients in the same way as kinetic models.

\subsubsection{Determining associations between genes and encoded metabolic reactions}

Estimating the cost of a gene on metabolism first requires determining the points of effect the gene has on the metabolic network. This analysis aimed to first determine which reactions have a single effect on metabolism, and thereby allowing a simpler analysis of their cost. The focus on 1:1 associations of gene to reactions excluded 488 genes but did allow a much simpler analysis of gene cost, where any effect on the gene may be expected to have a direct effect on single reaction which can then be analysed in through the effect on the corresponding phenotype.

Alternative approaches used in gene knockout studies have treated the reaction as unaffected if an associated paralog is still available, or have removed the reaction if the one unit of a multimeric enzyme is deleted \cite{pal2006}. In this analysis reactions are not being removed, but instead their effect on the model objective function estimated. There while it can be assumed that the removal of a enzyme subunit will render the complete enzyme ineffective, the same cannot by said by of reducing the effect of a single gene product within an multimeric enzyme. For considering paralogs catalysing the same reaction it could be assumed that a deficiency in one paralog will not affect the reaction rate, this may not necessarily be the case as some paralogs may only expressed under certain conditions \cite{carlson2007,perez2008}.

%TODO: Any other knockout studies can be cited?

%TODO: How did other knockout studies treat paralogs?

\subsubsection{Different constraints on reaction flux in glucose and ammonium limitation}

The reactions at maximum flux represent edges in solution space, where any reaction cannot exceed an absolute flux of 1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$. If the artificial constraint of reaction flux is increased the likely outcome is that the flux of these reactions would also increase, and the flux through the reactions is of an artificial limit imposed by the model optimisation, rather than because of the simulation of a biological phenotype. Therefore these reactions are not considered further in the analysis. The number of reactions at the limit is low, six across all conditions, and removal from the gene cost data set is not a significant impact on the size.

The constrained reactions represent those whose flux is constrained in the solution space. Forced reduction of the flux through any of these reactions results in a model will fail find an optimal solution. As the flux cannot be reduced, these reactions all represent inviable knockout mutants where removal from the model will prevent the model being optimised. These reactions may represent singleton pathways, the only available reactions for producing a required nutrient.

The variable reactions whose flux can be reduced and still maintain a viable model solution. Reasons that these fluxes can be reduced may be because optimisation finds an alternate pathway for the production of a required nutrient. Whether any of these reactions can be completely deleted from the model is not clear, only that the model is viable if the reaction flux is reduced.

The zero flux reactions are unused in the solution space and represent the majority of reactions. These reactions could be removed from the model without any effect on the model solution. These reactions could represent environment or nutrient specific reactions, required only when a specific nutrient is present.

The determination of each of these reaction phenotypes relies on the optimisation of the model. The solution space produced by the model may be over constrained the determination of the optimum. The \emph{in vivo} flux distribution may be more relaxed as highly constrained reaction fluxes may provide a competitive disadvantage in response to any environmental or genetic perturbations. Another consideration is whether \emph{in vivo} conditions are able to reach the optimised peak of the solution space, or instead reaching a point very close to the optimum but still maintaining a measure of flexibility in the metabolic network. This theory is discussed Mahadevan and Schilling \cite{mahadevan2003} where reaction fluxes are instead considered over a range of suboptimal solutions near the optimal. Future work could consider using testing the use of reactions over a range of suboptimal solutions to consider if this provides a different view of reactions use and constraint.
