\section*{Chapter summary}%{{{1

This chapter illustrates a novel approach to estimating amino acid biosynthetic cost and the functional importance of reaction encoded by single genes using the \emph{in silico} \emph{Saccharomyces cerevisiae}. The results described here apply existing systems biology control frameworks to a genome scale model of the yeast metabolism to first determine the cost of each amino acid in the context of environmental uptake of the nutrients required for synthesis. Second the fitness effect of a set of 262 yeast genes are determined by examining the constraints on gene-encoded reactions in the model in different environmental conditions.

\clearpage

\section{Introduction} %{{{1

\subsection{Understanding control in metabolic models}%{{{2

Early enzymology assumed the existence of rate limiting steps in biological pathways. The intuition was that the overall rate of a pathway is constrained by the rate of the slowest step. As the slowest reaction rate increases the whole pathway rate increases in proportion until another step becomes limiting. Niederberger \emph{et. al} \cite{niederberger1992} tested this assumption and found in contrast that individual up or down regulation of enzyme quantities at specific reaction steps had only marginal effect on overall tryptophan biosynthesis pathway in \emph{Saccharomyces cerevisiae}. The rate of the pathway was instead accelerated by increasing the quantity of five related enzymes in tandem. This research demonstrated control in a biological system is distributed over the system as a whole rather than concentrated at individual reactions.

The theory of metabolic control analysis (MCA) \cite{fell1992,fell1997} states that there is no rate limiting step in a pathway, but instead each reaction shares a measure of overall control. The control the rate of reaction $x$ has on reaction $y$ is determined through applying a small change in the rate of $x$ and measuring the degree of response in the rate of $y$. In metabolic control analysis this is described as the flux control of reaction $x$ on reaction $y$. The unitless coefficient of this control uses the symbol $C_{x}^{y}$ and is defined as:

\begin{equation}\label{fcc}
C_{x}^{y} = \frac{ \delta \ln x / \ln x}{\delta \ln y / \ln y} \, .
\end{equation}

The numerator ($\delta \ln x / \ln x$) represents the dimensionless effect of a small change in the reaction flux through $x$. The denominator $(\delta \ln y / \ln y$) represents the corresponding response in $y$ resulting from the change in $x$. The value of $C_{x}^{y}$ represents the ratio of response in reaction $y$ to changes in the rate of $x$. Larger values of $C_{x}^{y}$ indicate the reaction $x$ has a high degree of control on reaction $y$. Smaller values indicate reaction $y$ experiences only a small change in flux in response to changes in $x$. Using this approach a quantitative measure of control may be derived for each reaction in a biological system, whether \emph{in vivo} or \emph{in silico}.
\nomenclature{MCA}{Metabolic Control Analysis}
\nomenclature{$C_{x}^{y}$}{Flux control coefficient of reaction $x$ on reaction $y$}

Determining a numeric value for the relationships between reactions in a metabolic network allows quantitative exploration of biological systems. For example Rossell \emph{et al.} described the control of metabolism in terms of reactions whose rates are increased by changes in their substrate concentration or the reactions whose activity is instead controlled through changes in enzyme quantity and therefore via gene expression \cite{rossell2006,daran-lapujade2007}.

Example of the application MCA include engineering an increase in the production of commercial biomolecules \cite{niederberger1992}. MCA can also be used to identify possible drugs targets through the combination of reaction knock-downs which have the greatest effect on the metabolic network \cite{lehar2008,hopkins2008}. This can be further expanded in developing antibiotic drugs for human pathogens such as trypanosome parasites, where likely drug targets are the reactions with a significant effect on metabolism in the pathogen but where the orthologous human reaction has a much decreased effect in response to the drug. This difference in metabolic control for the same reaction in both species translates into lethal drug effect in the parasite with minimal impact on the host \cite{hornberg2007}.

%TODO: read Hopkins, Lehar papers

\subsection{Genome scale models}%{{{2

Derivation of a flux control coefficient \emph{in silico} requires the construction of a kinetic model of the system including enzyme kinetic parameters. In the case of reactions exhibiting simple Michaelis-Menten kinetics only substrate affinity (K$_{m}$) and maximum enzyme reaction rate (V$_{max}$) are necessary. More complex enzymatic kinetics more larger parameter which requires further experimental effort to derive. Even with all experimentally derived kinetic parameters, a model must still be tested to determine if simulated behaviour matches expected \emph{in vivo} behaviour \cite{teusink2000}. The total experimental effort required to derive complete sets of enzyme kinetic parameters means kinetic models are relatively small compared with the anticipated size of total cellular metabolism \cite{steuer2007}.
\nomenclature{K$_{m}$}{Substrate concentration where enzyme rate is at half maximum ($\frac{1}{2}$V$_{max}$)}
\nomenclature{V$_{max}$}{Maximum reactions per second per mole of enzyme}

%TODO: check Steuer paper

In comparison with kinetic models, a stoichiometric model does not require enzyme kinetic parameters, only a list of reactions and the metabolites used in reaction. A stoichiometric model is represented by an $m \times n$ sized matrix $S$. The matrix $S$ represents the reactions in metabolism and the substrates used and products produced by each reaction. The size of $m$ in the matrix is the total number of metabolites in the model, and $n$ is the total number of reactions. The position $S_{ij}$ is the participation of metabolite $i$ in reaction $j$. Positive values of $S_{ij}$ indicate the metabolite is produced by the reaction, negative values mean the metabolite is consumed. A zero indicates the metabolite does not participate in the reaction.

As stoichiometric models require less data to create it is easier to produce large metabolic representations. Stoichiometric models comprising a significant proportion of an organism's metabolism have be created, described as genome scale models. The construction of genome scale metabolic models is reviewed by Feist \emph{et al.} \cite{feist2009}, and the process is outlined in brief here.

The construction of a genome scale model is the production of the stoichiometric matrix detailing the participation of reactions and substrates in each reaction. Enzyme databases such as BRENDA \cite{chang2009} and KEGG \cite{okuda2008} may provide information on conserved enzymes and pathways while specific databases such as the \emph{Saccharomyces} Genome Database (SGD) \cite{cherry1997} allow the identification of reactions specific to yeast. The initial stages of construction can be automated by cross referencing metabolic maps with the presence of the enzyme in the genome. Expert knowledge of an organism can answer questions about the existence of specific metabolic pathway, particularly for organisms living in niche environments \cite{sun2009}.
\nomenclature{SGD}{\emph{Saccharomyces} Genome Database (see \cite{cherry1997})}

The first \emph{Escherichia coli} model was developed almost 20 years ago \cite{feist2008}. A simplified model of acetate production was described by Majewski \emph{et al.} in 1990, the recent 2007 genome scale model accounts for 1260 open reading frames \cite{feist2007}. The range of species for which a genome scale model exists continues to expand and includes attempts to reconstruct a human cell \cite{duarte2007}. The field of genome scale model construction does however suffer from the lack of a single standard for construction and model format, making model interchange and comparison difficult. There are however attempts to create a unified \emph{S. cerevisiae} model \cite{herrgard2008} using existing standards such the systems biology markup language (SBML) \cite{hucka2003}, minimum information requested in annotation of biochemical models (MIRIAM) \cite{lenovere2005} as a standard for model annotation, and standardised encoding and representation of chemicals using the international chemical identifier codes (InChI) \cite{coles2005}.
\nomenclature{InChI}{IUPAC International Chemical Identifier}
\nomenclature{SBML}{Systems Biology Markup Language}
\nomenclature{MIRIAM}{Minimum Information Requested In Annotation of biochemical Models}

When constructing a genome scale model the intended goal is often the simulation of \emph{in vivo} phenotypes which requires an additional reaction describing cellular growth. In many cases this reaction consumes lipids, proteins, nucleic acids and other cofactors to produce a per hour (hr$^{-1}$) unit of new biomass. The required quantities of each molecule in this growth reaction are estimated from \emph{in vivo} millimoles per gram of dry weight cellular biomass (mmol $^{-1}$ gDW $^{-1}$). The \emph{in silico} simulations of a newly constructed model must be validated against the observed \emph{in vivo} phenotypes, where \emph{in silico} behaviour should reflect that of \emph{in vivo} behaviour as closely as possible \cite{famili2003}. Differences between \emph{in silico} and \emph{in vivo} highlight gaps in the model which can be updated and followed by repeated model validation. The use and simulation of stoichiometric models is described in the next section.
\nomenclature{gDW}{Grams of Dry Weight biomass}
\nomenclature{mmol}{Millimoles, quantity equal to one thousandth number of atoms in 0.012 grams of carbon$^{12}$}
\nomenclature{hr}{Hour}

\subsubsection{Stoichiometric model simulation using flux balance analysis}%{{{3

Stoichiometric models are simulated using flux balance analysis (FBA). As described above a stoichiometric model is a matrix $S$ describing the substrates and products used in each reaction. FBA uses linear programming to solve the equation $S v = 0$ where $v$ represents a combination of reaction rates, known as fluxes, for each reaction in $S$. There are multiple solutions to this equation and so linear optimisation is used to determine the combination of fluxes for $v$ that maximises or minimises a particular value, this optimised value is the objective function. A common objective function used for genome scale models of unicellular organisms is maximising growth rate. Additional constraints can be placed on the flux balance solution to reflect \emph{in vivo} reaction kinetics, such as upper and lower bounds for individual reactions. Constraining a reaction to be $\geq 0$ restricts the solution space so the reaction may only proceed in one direction. Reaction constraints are also used to limit the flux of nutrients entering the cell and thereby simulate different environmental conditions.
\nomenclature{FBA}{Flux Balance Analysis}

The aim of flux balance analysis using a genome model is to simulate organism phenotypes. Minimisation of metabolic adjustment (MOMA) \cite{segre2002,burgard2003} is a technique that optimises the objective function while minimising the sum change in flux adjustment between two flux solutions. The aim of this method is to simulate perturbed metabolic phenotypes such a wild-type and gene knockout, where minimal flux changes more accurately reflect the \emph{in vivo} behaviour of how metabolic adjustments are made. A related approach minimises the number of on-off changes in reaction use (Regulatory on/off minimisation ROOM), whereby the number of reactions that switch between active or inactive between solutions is minimised \cite{shlomi2005}. This approach again mimics \emph{in vivo} behaviour where the use of already active reactions in different flux phenotypes is more realistic simulation compared to large scale metabolic readjustments through activation of inactivation of multiple reactions.
\nomenclature{MOMA}{Minimisation of Metabolic Adjustment}
\nomenclature{ROOM}{Regulatory on/off minimisation}

The simulation of an organism phenotype is dependent on how the model is optimised. The FBA solution may be considered constraint to the objective function where any changes in internal reaction flux results in a dramatic decrease in objective flux. Mahadevan and Schilling \cite{mahadevan2003} discuss that \emph{in vivo} metabolism may not be tightly optimised for a specific environment, but instead metabolise at a near optimal rate and retain a measure of flexibility in the metabolic network to adjust to changes in the environment. In their research Mahadevan and Schilling perform FBA to find a near optimal solution and showed this allowed a greater flexibility in the metabolic network.

\subsubsection{Predicting essential genes using genome scale models and flux balance analysis}%{{{3

Estimating metabolic control coefficients using stoichiometric models is difficult as the flux the through the reaction is determined by linear optimisation, rather than enzyme kinetics. A common approach to estimating the importance of a reaction in a genome scale model is to remove the reaction, and compare the resulting solution with the original wild-type simulation. The deletion of a reaction may be considered to confer a fitness defect if there is a significant growth rate reduction \cite{pal2006}. The deletion of a reaction may also prevent model optimisation indicating the reaction is essential \cite{becker2008}.  Even if a model solution can still be found, the reaction may still be considered essential if the growth defect is large enough \cite{papp2004,wang2009}.

Flux balance analysis of all possible gene deletions for \emph{S.cerevisiae} was calculated and compared with \emph{in vivo} observations. Of the knockout phenotypes up to 85\% were correctly predicted depending on the simulated media \cite{famili2003,forster2003}. \emph{In silico} knockout studies have been extended to detect metabolic redundancy by comparing all possible double reaction deletions and finding the pairs with no effect when deleted individually, but indicate a sick or lethal phenotype when deleted simultaneously \cite{harrison2007}. Reactions active in a specific range of conditions can also be determined through deletion in a wide range of simulated conditions, where a deletion may result in a fitness effect in only a small subset of biologically feasible conditions \cite{papp2004}.

The use of genome scale metabolic models is not limited to estimating the control of reactions on growth, but may also be used to estimate the importance of specific metabolites. Varma \emph{et al.} \cite{varma1993} used flux balance analysis and stoichiometric models to determine the role of oxygen in \emph{E. coli} energy metabolism through decreasing oxygen availability and identifying the corresponding change in the use of redox carriers. As oxygen availability decreased, the use of ethanol as the electron acceptor increased. The authors described this effect in terms of the price of the each molecule, where the price of oxygen use rises with decreasing availability while the price of ethanol use decreases. This highlights how systems biology models can be used to estimate the importance of metabolites to cell growth. Few subsequent studies have looked at the price of metabolite in a systems biology framework, though the biosynthetic price of amino acids in protein synthesis has been widely examined.

\subsection{Estimating amino acid biosynthetic cost}%{{{2

Craig \& Weber \cite{craig1998} estimated the biosynthetic cost of an amino acid as the sum of how many high energy phosphate bonds (e.g. ATP) and reducing molecules (e.g. NADPH) are diverted from glucose metabolism.

Akashi \& Gojobori \cite{akashi2002} explored a similar approach calculating cost on a range of different energy sources to estimate the role of amino acid cost in predicted \emph{E. coli} and \emph{Bacillus subtilis} gene expression. Heizer \emph{et al.} \cite{heizer2006} expanded this approach to estimate amino acid cost four prokaryotic species including photoautotrophs. Wagner considered biosynthetic amino acid cost in \emph{S. cerevisiae} using a similar approach but considering both respiratory and fermentative energy metabolism. Seligmann argued \cite{seligmann2003} the cost of amino acid biosynthesis must take into account more than just the energy required for synthesis, and used molecular weight as a proxy for amino acid cost arguing that this may take into account investments made in synthesising more complex molecular structures. Using molecular weight also has the advantage of being independent of a particular species. Table~\ref{table:literature_costs} summarises these described measures of cost.

\begin{table}
\centering
\begin{footnotesize}
  \begin{tabular}{ p{0.8cm} *{6}{p{1.8cm}} }
                                                                                \toprule
          & A\&G   & C\&W   & C\&W  & Wager        & Wagner      & Molecular \\
          & Energy & Energy & Steps & Fermentative & Respiratory & Weight    \\ \midrule
      ala & 11.7   & 12.5   & 1     & 2            & 14.5        & 89.1      \\
      arg & 27.3   & 18.5   & 10    & 13           & 20.5        & 174.2     \\
      asn & 14.7   & 4      & 1     & 6            & 18.5        & 132.1     \\
      asp & 12.7   & 1      & 1     & 3            & 15.5        & 133.1     \\
      cys & 24.7   & 24.5   & 9     & 13           & 26.5        & 121.2     \\
      gln & 16.3   & 9.5    & 2     & 3            & 10.5        & 146.2     \\
      glu & 15.3   & 8.5    & 1     & 2            & 9.5         & 147.1     \\
      gly & 11.7   & 14.5   & 4     & 1            & 14.5        & 75.1      \\
      his & 38.3   & 33     & 1     & 5            & 29          & 155.2     \\
      ile & 32.3   & 20     & 11    & 14           & 38          & 131.2     \\
      leu & 27.3   & 33     & 7     & 4            & 37          & 131.2     \\
      lys & 30.3   & 18.5   & 10    & 12           & 36          & 146.2     \\
      met & 34.3   & 18.5   & 9     & 24           & 36.5        & 149.2     \\
      phe & 52     & 63     & 9     & 10           & 61          & 165.2     \\
      pro & 20.3   & 12.5   & 4     & 7            & 14.5        & 115.1     \\
      ser & 11.7   & 15     & 3     & 1            & 14.5        & 105.1     \\
      thr & 18.7   & 6      & 6     & 9            & 21.5        & 119.1     \\
      trp & 74.3   & 78.5   & 12    & 14           & 75.5        & 204.2     \\
      tyr & 50     & 56.5   & 9     & 8            & 59          & 181.2     \\
      val & 23.3   & 25     & 4     & 4            & 29          & 117.2     \\ \bottomrule
  \end{tabular}
\end{footnotesize}
\caption[Amino acid costs described in the literature]{Amino acid costs described in the literature. The Akashi \& Gojobori \cite{akashi2002}, Craig \& Weber energy \cite{craig1998}, and two Wagner \cite{wagner2005} data sets are based on the curation of the number of high-energy molecules used during synthesis. The Craig \& Weber `steps' measure \cite{craig1998} is based on the number of biosynthetic steps between central metabolism and the produced amino acid. Molecular weight is in Daltons.}
\label{table:literature_costs}
\end{table}

The majority of cost estimates described require manual examination of a metabolic network to determine the number of energy molecules expended in the synthesis. This is a time-consuming activity and susceptible to human error. No existing approaches to estimating amino acid cost have considered using genome scale models in an approach similar to Varma \emph{et al.}.

\subsection{Summary of presented results}

The results presented in this chapter are in two sections. The first describes an approach to estimating amino acid cost used the \emph{S. cerevisiae} genome scale metabolic model. The approach used is similar to that described by Varma \emph{et al.} in their estimation of the oxygen price in \emph{E.coli} energy metabolism where each amino acid biosynthetic cost is estimated in the context of glucose, ammonium, and sulphate limitation. The estimated costs are validated by previous estimates of amino acid cost in the literature and show that this novel systems biology approach can be successfully used to estimate the biosynthetic cost of molecule in a variety of conditions.

The second part of results is a related approach to estimate the fitness effect of a gene. This is again used the \emph{S. cerevisiae} genome scale metabolic model. Gene fitness is examined in the metabolic model via enzymatic reactions known to be encoded by the gene. Each reaction is examined using three measures: the flux through the reaction, whether reaction flux may vary, and how reductions in reaction flux affect the model objective function. The represents a novel analysis of gene fitness using genome scale models, and is used for analysing gene function and evolution in later chapters.

\clearpage

\section{Materials and Methods}%{{{1

\subsection{Determination of gene to reaction association in \emph{S. cerevisiae} model}

Only a subset of the reactions in the \emph{S. cerevisiae} metabolic model have a direct gene association. This annotation of the model defines if there is one or more genes known to encode an enzyme catalysing one of the reactions in the model stoichiometric matrix. This association can include multimeric enzymes where multiple gene products form an enzyme complex or are isoenzymes where multiple gene encode enzymes for the same reactions.

The COBRA toolbox used in the analysis defines a sparse matrix specifying which \emph{S. cerevisiae} genes are associated with which model reactions. The matrix is sparse as not all reactions have an associated gene. This matrix was parsed to determine the reactions catalysed by only a single gene, where the gene is not associated with any other reactions.

\subsection{Flux balance analysis}

The genome scale models used in this work were \emph{S.cerevisiae} iND750 \cite{duarte2004a} and \emph{E.coli} iJR904 \cite{reed2003}. Flux balance analysis was performed using the COBRA toolbox \cite{becker2007} and the lpsolve optimisation library \cite{lpsolve}.
\nomenclature{COBRA}{Flux balance analysis toolbox for MATLAB (see \cite{becker2007})}
\nomenclature{iND750}{\emph{Saccharomyces cerevisiae} genome scale model (see \cite{duarte2004a})}
\nomenclature{iJR904}{\emph{Escherichia coli} genome scale model (see \cite{reed2003})}

Nutrient limitation was simulated by fixing the upper and lower bounds of the biomass production reaction to 0.3$^{-1}$ an optimising to minimise the entry of a specific nutrient into the cell. All other nutrient exchange fluxes were set to have bounds of -10000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$, effectively setting all other nutrients to be non-limiting on simulated growth. The flux balance analysis configuration therefore finds the minimum required amount of a specific nutrient to maintain cellular growth rate at 0.3 hr$^{-1}$.

For the \emph{S. cerevisiae} iND750 \cite{duarte2004a} genome scale model the reactions allowing entry of nutrients into the cell are for glucose, ammonium, sulphate, phosphate, oxygen and water. Each of these reactions represents the only external source for an essential nutrient and, for example, when simulating the glucose limitation there are no other high energy sugars available. The FBA simulations does however allow a variety of molecules to exit the cell, these are in the majority metabolic by products such as carbon dioxide or ethanol.

Fixing \emph{in silico} growth rate also simulates experimental conditions using continuous culture of microorganisms in the chemostat (see Castrillo \emph{et al.} \cite{castrillo2007}) where the minimisation of a specific nutrient as the \emph{in silico} objective mimics chemostat conditions where one nutrient is the limiting factor for growth and all other nutrient are available in excess.

\subsection{Flux balance analysis estimation of amino acid cost}

Estimation of amino acid cost was performed by making a percentage change ($\pm$0.0002\%) to the requirement of amino acid at position $S_{ij}$ in the model stoichiometric matrix, where $i$ is the amino acid and $j$ is the biomass reaction. When the model is optimised each change in $S_{ij}$ results in a corresponding change in the limiting nutrient uptake. Larger effects in nutrient uptake indicate a greater nutrient cost for the amino acid.  The slope between the change in amino acid requirement and the corresponding change in nutrient uptake flux was taken as the cost of the amino acid for that nutrient. This cost is a relative cost as it is derived from a percentage changes in amino acid requirement rather than constant changes in requirement. Division of the relative cost by the original amino acid requirement does however result in an absolute estimate of amino acid cost. Both relative and absolute amino costs were calculated for glucose, ammonium, and sulphate limitation. Finally each cost was divided by the growth rate at which it was estimated (0.3 hr$^{-1}$) to give an estimate of cost independent of growth. The nomenclature used for relative and absolute cost definitions is $R_{nutrient}$ and $A_{nutrient}$ respectively.
\nomenclature{$A_{nutrient}$}{Amino acid absolute nutrient cost}
\nomenclature{$R_{nutrient}$}{Amino acid relative nutrient cost}

The additional optimisation techniques described in the introduction, such as MOMA or suboptimal solutions, were not used in the estimation of amino acid cost. These techniques focus on the realistic \emph{in vivo} simulation reaction fluxes, while instead this analysis focused on the corresponding change in nutrient uptake flux given the change in amino acid stoichiometry.

\subsubsection{Units of absolute and relative measures of amino acid cost}

The relative costs are calculated value of $m$ in the linear relationship $y = mx + c$ where $x$ is the unit less percentage change in the amino acid requirement, and $y$ is corresponding mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ response in nutrient intake flux. When divided by the hr$^{-1}$ growth rate at which the cost was estimated, the units of the relative cost are mmol$^{-1}$ gDW$^{-1}$. This represents the change in nutrient uptake flux given the fractional change in amino acid requirement in biomass. The absolute cost is derived from the relative cost by division by the original mmol$^{-1}$ gDW$^{-1}$ requirement of the amino acid. The absolute amino acid is therefore unit less and represents what a mmol$^{-1}$ gDW$^{-1}$ increase in amino acid requirement has the mmol$^{-1}$ gDW$^{-1}$ uptake flux of the given nutrient.

\subsection{Flux balance analysis estimation of gene cost}

\subsubsection{Estimation of \emph{in vivo} flux distributions using suboptimal flux balance solutions}

The distribution reaction flux in the \emph{S. cerevisiae} model in of each three nutrient limited conditions, glucose, ammonium, and sulphate, was estimated as above where growth rate is fixed at 0.3 hr$^{-1}$ and the model optimised for the minimum intake of the given nutrient.

This solution was then used to find a suboptimal solution for the same nutrient limiting condition by a setting a lower boundary on the minimum flux of nutrient into the cell at 5\% of the minimum allowable flux. For example if the minimum glucose flux into the cell at growth rate 0.3 hr $^{-1}$ is $x$, then the lower boundary of glucose into the cell was set at 1.05$x$, where the amount of glucose entering the cell may increase, but not decrease below this value.

The model was re-optimised to determine the less optimal solution and corresponding distribution of internal reaction fluxes. This aim of approach is to simulate \emph{in vivo} flux distributions whereby the cellular state is very close to an optimum growth rate while still maintaining a measure of variability in the metabolic network \cite{mahadevan2003}.

\subsubsection{Determination of reaction variability in glucose, ammonium, and sulphate limitation}

The fluxes for all reactions with a single gene association were estimated for glucose, ammonium, and sulphate limited suboptimal flux balance analysis solutions in the yeast model. Reactions at zero flux in all three conditions were ignored. To identify changes in reaction fluxes between environments each reaction flux was scaled by subtracting the median value across all three environments.

In each nutrient limited condition, each reaction was classified based on use in the suboptimal solution space.  The first two categories were based on whether the reaction flux was zero (`zero flux'), or at the maximum allowable flux of $\pm$1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ (`at maximum').

If the reaction did not fall into either of these two categories it was then classified on whether the flux could vary in the FBA solution given the cell growth rate and nutrient limitation. For each remaining reaction, the model was re-optimised using MOMA to minimise the absolute flux through the reaction. If the reaction could not be reduced whilst maintaining the same growth and nutrient uptake flux, this indicated the reaction was constrained in the solution space (`constrained'), otherwise the the reaction was classified as variable (`variable').

This resulted in one of four classifications for each single gene-associated reaction in each of the three nutrient limitations considered.

\subsubsection{Determination of reaction flux control on in glucose, ammonium, and sulphate uptake}

The degree of control for each variable-type reaction was estimated for glucose, ammonium, and sulphate limiting conditions.

To estimate the reaction control on nutrient uptake, the flux through each reaction was fixed at one of five points in the range of 100\% - 95\% of original reaction flux in the initial suboptimal solution. For each point change in the the model was re-optimised using MOMA resulting in a corresponding reponse in the nutrient uptake flux, given the change in reaction flux.

The slope of the regression in change in the nutrient uptake flux as a response in to changes in reaction flux was the cost for the reaction in that nutrient limitation.

%TODO: what units?
%TODO: independent of growth?

\clearpage

\section{Results}%{{{1

\subsection{The \emph{S. cerevisiae} iND750 genome scale model}%{{{2

\subsubsection{Numbers of reactions and genes}%{{{3

To analyse reaction function and infer characteristics of \emph{S. cerevisiae} genes the relationship between model reactions and annotated genes must first be understood. Table~\vref{table:gene_associations} illustrates the number and types of associations between genes and reactions. Each reaction in the model can be categorised based on the type of associations with annotated genes.

\begin{table}[b]%{{{
  \centering
  \begin{tabular}{l r}
                                                                   \toprule
    Description                                      & Number   \\ \midrule
    Genes                                            & 750      \\
    Reactions                                        & 1266     \\
    Reactions with at least one associated gene      & 810      \\
    Reactions with a single gene associated          & 579      \\
    One-to-one association between reaction and gene & 262      \\ \bottomrule
  \end{tabular}
  \caption[Gene associations in the \emph{S. cerevisiae} iND750 model]{Comparison of numbers and types of associations between genes and reactions in the \emph{S. cerevisiae} iND750 model. }
  \label{table:gene_associations}
\end{table}%}}}

The total number genes included in the model is 750. Of the 1266 reactions, 810 are associated with one or more genes. This includes reactions that may be catalysed by paralogs, multimeric enzymes, and genes also associated other reactions. A subset of this category are the reactions associated to only a single gene containing 579 reactions. This excludes paralogs and multimeric enzymes, but still includes enzymes associated with more than one reaction. A further subset of this category are the reactions associated to a single gene, where the gene is associated with no other reactions. This comprises 262 genes which may be expected to have only a single point of effect in \emph{S. cerevisiae} metabolism are therefore useful for inferential analysis of the relationship between gene function and evolution, and corresponding metabolic effect.

\subsubsection{Composition of biomass requirement}%{{{3

As discussed in the introduction, the biomass equation in a stoichiometric model is an estimated requirement of essential metabolites for growth in a given organism. In flux balance analysis, the biomass equation can be used as the objective function to simulate the rate of growth given the available basic input nutrients. Table \vref{table:biomass_requirements} summarises the molecular requirement of the biomass reaction in the \emph{S. cerevisiae} iND750 model. These requirements include fats such as triglyceride and ergosterol, sugars such as glycogen and mannan, and small metabolites comprising sulphate, water, and ATP. The genome, transcriptome, and proteome content of the cell is included as the quantities of nucleotides and amino acids estimated from dry weight cellular composition.

\begin{table}%{{{
  \centering
  \begin{footnotesize}
  \begin{tabular}{l l r}
                                                                                              \toprule
    Type           & Molecule                       & Requirement (mmol$^{-1}$ gDW$^{-1}$) \\ \midrule
    Nucleic Acid   & dAMP                           &  0.00360                             \\
                   & dCMP                           &  0.00240                             \\
                   & dGMP                           &  0.00240                             \\
                   & dTMP                           &  0.00360                             \\
                   & AMP                            &  0.04600                             \\
                   & CMP                            &  0.04470                             \\
                   & GMP                            &  0.04600                             \\
                   & UMP                            &  0.05990                             \\ \midrule
      Amino Acid   & Alanine                        &  0.45880                             \\
                   & Arginine                       &  0.16070                             \\
                   & Asparagine                     &  0.10170                             \\
                   & Aspartate                      &  0.29750                             \\
                   & Cysteine                       &  0.00660                             \\
                   & Glutamate                      &  0.30180                             \\
                   & Glutamine                      &  0.10540                             \\
                   & Glycine                        &  0.29040                             \\
                   & Histidine                      &  0.06630                             \\
                   & Isoleucine                     &  0.19270                             \\
                   & Leucine                        &  0.29640                             \\
                   & Lysine                         &  0.28620                             \\
                   & Methionine                     &  0.05070                             \\
                   & Phenylalanine                  &  0.13390                             \\
                   & Proline                        &  0.16470                             \\
                   & Serine                         &  0.18540                             \\
                   & Threonine                      &  0.19140                             \\
                   & Tryptophan                     &  0.02840                             \\
                   & Tyrosine                       &  0.10200                             \\
                   & Valine                         &  0.26460                             \\ \midrule
      Carbohydrate & 1 3 beta D Glucan              &  1.13480                             \\
                   & Mannan                         &  0.80790                             \\
                   & Glycogen                       &  0.51850                             \\
                   & Trehalose                      &  0.02340                             \\ \midrule
      Lipid        & Ergosterol                     &  0.00070                             \\
                   & Triglyceride                   &  0.00007                             \\
                   & Zymosterol                     &  0.00150                             \\
      Phospholipid & Phosphatidyl 1D Myo Inositol   &  0.00005                             \\
                   & Phosphatidate                  &  0.00001                             \\
                   & Phosphatidylcholine            &  0.00006                             \\
                   & Phosphatidylethanolamine       &  0.00005                             \\
                   & Phosphatidylserine             &  0.00002                             \\ \midrule
      Small        & ATP                            & 59.27600                             \\
                   & H$_2$O                         & 59.27600                             \\
                   & Sulphate                       &  0.02000                             \\ \bottomrule
  \end{tabular}
  \end{footnotesize}
  \caption[Biomass requirements \emph{S. cerevisiae} iND750 model]{Biomass requirements of \emph{S. cerevisiae} iND750 model. Each molecule is categorised by type. The requirement of each molecule is millimoles per gram of dry weight biomass. The estimation of biomass quantities is outlined by Feist \emph{et al.} \cite{feist2009}, the \emph{S. cerevisiae} model construction is described by Duarte \emph{et al.} \cite{duarte2004a}.}
  \label{table:biomass_requirements}
\end{table}%}}}

\subsection{A novel approach using genome scale models to estimating amino acid cost}\label{section:amino_acid_cost_estimation}%{{{2

The introduction described how metabolic control analysis can be used to estimate the importance of each reaction in a kinetic model. This section shows how similar principles can be used to estimate the sensitivity of \emph{S. cerevisiae} growth to changes in amino acid requirements. This sensitivity is interpreted as the biosynthetic cost of the amino acid in the nutrient limiting condition it is estimated.

\begin{figure}%{{{
  \centering
  \subfloat[Cost estimation in glucose limited conditions]{
    \label{figure:curve_estimation:schematic}
    \includegraphics*[width=6.5cm]{cost_schematic.eps}
  }
  \hfill
  \subfloat[Supply-demand elasticity for tryptophan]{
    \label{figure:curve_estimation:curve}
    \includegraphics*[width=6cm]{curve_estimation.eps}
  }
  \hfill
  \subfloat[Comparison of costs for glycine, histidine, and tryptophan]{
    \label{figure:curve_estimation:costs}
    \includegraphics*[width=7cm]{comparison_of_costs.eps}
  }
  \hfill
  \caption[Method used for estimation of amino acid cost]{Outline of the method used for the estimation of amino acid cost. Figure~\ref{figure:curve_estimation:schematic} is a schematic representation of simulating glucose limited conditions using a genome scale model. The growth rate of the model is fixed at a constant value and the objective function of the simulation minimises the cellular entry of a specific nutrient, which in the example is glucose. All other nutrient entering the model, illustrated here with ammonia, are unbounded. Figure~\ref{figure:curve_estimation:curve} shows the relationship between altered tryptophan requirement and the response in both glucose and ammonium uptake. Figure~\ref{figure:curve_estimation:costs} contrasts the estimated costs for tryptophan, glycine and histidine in glucose and ammonium limiting conditions. }
  \label{figure:curve_estimation}
\end{figure}%}}}

Figure~\vref{figure:curve_estimation} illustrates the method used to estimate the amino acid biosynthetic cost using flux balance analysis. Figure~\ref{figure:curve_estimation:schematic} demonstrates the model configuration for the simulation of a glucose limited environment, amino acid costs were also calculated for ammonium and sulphate limiting conditions.

In each nutrient limiting environment the biomass requirement of each amino acid is manipulated around the original value in the range of $\pm$0.0002\%, and each time the model is re-optimised to determine the correspinding effect on the uptake flux of the limiting nutrient.  Figure~\ref{figure:curve_estimation:curve} provides an example of the relationship between changes in tryptophan requirement and the flux of either glucose and ammonium into the cell.

Two types of amino acid costs were estimated from changing biomass requirement. The first is described as a relative amino acid cost and is estimated from small percentage changes in requirement. This relative estimation of amino cost can be rescaled to estimate a cost based on absolute changes in amino acid requirement described as an absolute measure of amino acid cost.

One further consideration in estimating amino acid cost is the growth rate at which the costs were estimated. For instance the absolute cost of tryptophan estimated at 0.2 hr$^{-1}$ is twice the cost of tryptophan estimated at 0.1 hr$^{-1}$. To control for growth rate when estimating amino acid cost, the costs were estimated at a three levels of feasible yeast growth rates 0.1 hr$^{-1}$, 0.2 hr$^{-1}$ and 0.3 hr$^{-1}$. Each amino acid cost is then rescaled to be independent of the growth rate at which it was estimated. The amino acid costs presented in this work are those estimated at the 0.3 hr$^{-1}$ growth rate, where \emph{S. cerevisiae} uses respiratory energy production \cite{famili2003}.

Figure~\ref{figure:curve_estimation:costs} compares the estimated absolute costs for tryptophan, histidine and glycine in glucose and ammonium limitation. Tryptophan which is an expensive amino acid in glucose limitation, is cheaper in nitrogen limitation. In contrast to this histidine which is a cheap amino acid in glucose limitation is more expensive in ammonium limitation. Glycine, a small amino acid with a single hydrogen side chain is cheap in both nutrient limitations.

\subsubsection{Estimated amino acid costs in glucose, ammonium, and sulphate limitation}%{{{3

Using the above described method relative and absolute amino acid costs were estimated for all twenty amino acids in four simulated nutrient limiting conditions: glucose, ammonium, sulphate, and phosphate. For clarity costs are described as either $R_{nutrient}$ or $A_{nutrient}$ for relative and absolute costs respectively. The relative and absolute estimated costs for each of these amino acids is illustrated in Figure~\vref{figure:costs_dendrogram_dotplot} along with amino acid costs described in the literature. The estimated amino acid cost are provided in Table~\vref{table:estimated_costs} and the correlation between costs is shown in Table~\vref{table:costs_correlation}.

\begin{table}%{{{
\begin{footnotesize}
  \begin{tabular}{ p{1cm} *{6}{p{1.9cm}} }
  \toprule
      & $A_{glucose}$ & $A_{nitrogen}$ & $A_{sulphur}$ & $R_{glucose}$ & $R_{nitrogen}$ & $R_{sulphur}$ \\ \midrule
      ala & 0.50 & 1 & 0 & 0.23 & 0.46 & 0.00 \\
      arg & 1.39 & 4 & 0 & 0.22 & 0.64 & 0.00 \\
      asn & 0.79 & 2 & 0 & 0.08 & 0.20 & 0.00 \\
      asp & 0.61 & 1 & 0 & 0.18 & 0.30 & 0.00 \\
      cys & 0.75 & 1 & 1 & 0.00 & 0.01 & 0.01 \\
      gln & 0.92 & 2 & 0 & 0.10 & 0.21 & 0.00 \\
      glu & 0.86 & 1 & 0 & 0.26 & 0.30 & 0.00 \\
      gly & 0.31 & 1 & 0 & 0.09 & 0.29 & 0.00 \\
      his & 1.46 & 3 & 0 & 0.10 & 0.20 & 0.00 \\
      ile & 1.21 & 1 & 0 & 0.23 & 0.19 & 0.00 \\
      leu & 1.21 & 1 & 0 & 0.36 & 0.30 & 0.00 \\
      lys & 1.31 & 2 & 0 & 0.38 & 0.57 & 0.00 \\
      met & 1.25 & 1 & 1 & 0.06 & 0.05 & 0.05 \\
      phe & 1.84 & 1 & 0 & 0.25 & 0.13 & 0.00 \\
      pro & 0.99 & 1 & 0 & 0.16 & 0.16 & 0.00 \\
      ser & 0.49 & 1 & 0 & 0.09 & 0.19 & 0.00 \\
      thr & 0.69 & 1 & 0 & 0.13 & 0.19 & 0.00 \\
      trp & 2.39 & 2 & 0 & 0.07 & 0.06 & 0.00 \\
      tyr & 1.77 & 1 & 0 & 0.18 & 0.10 & 0.00 \\
      val & 0.96 & 1 & 0 & 0.25 & 0.26 & 0.00 \\ \bottomrule
  \end{tabular}
\end{footnotesize}
\caption[\emph{S. cerevisiae} estimated absolute and relative amino acid costs]{The units of the relative costs ($R_{uptake}$) are mmol of uptake nutrient per gram dry weight biomass (mmol$^{-1}$ gDW$^{-1}$). The absolute costs ($A_{uptake}$) are unit less. All costs are estimated in the iND750 \emph{S. cerevisiae} model using the COBRA toolbox and are rounded to two decimal places.}
\label{table:estimated_costs}
\end{table}%}}}

\begin{sidewaysfigure}%{{{
\centering
\includegraphics*[angle=0,height=13cm]{costs_dendrogram_dotplot.eps}
\caption[Comparison of amino acid cost estimates]{Amino acid cost estimates are shown as bar charts on the left hand side. Each bar chart axis shows the minimum and maximum value of each cost type. The correlations between costs are compared in a dendrogram on the right hand side computed by complete agglomerative clustering using Spearman's Rank correlation distance between data sets. The cost values are shown in Table~\vref{table:literature_costs} and the correlation between costs are given in Table~\vref{table:costs_correlation}}
\label{figure:costs_dendrogram_dotplot}
\end{sidewaysfigure}%}}}

\begin{sidewaystable}%{{{
  \centering
  \begin{footnotesize}
    \begin{tabular}{p{2.2cm} *{12}{p{1.2cm}} }
                                                                                \toprule
           & A\&G   & C\&W   & C\&W  & Wager        & Wagner      & Weight & $A_{glu}$ & $R_{glu}$ & $A_{amm}$ & $R_{amm}$ & $A_{sul}$ & $R_{sul}$ \\
           & Energy & Energy & Steps & Ferm. & Resp. &            &               &               &              &                &                &                \\ \midrule
A\&G Energy  & 1.00  & 0.81  & 0.70  & 0.77  & 0.87  & 0.75  & 0.94  & 0.06  & 0.51  & -0.45 & 0.62  & 0.37  \\
C\&W Energy  & 0.81  & 1.00  & 0.62  & 0.45  & 0.81  & 0.44  & 0.73  & 0.07  & 0.34  & -0.40 & 0.36  & 0.17  \\
C\&W Steps   & 0.70  & 0.62  & 1.00  & 0.80  & 0.72  & 0.37  & 0.59  & 0.03  & 0.34  & -0.33 & 0.38  & 0.27  \\
Wagner Ferm. & 0.77  & 0.45  & 0.80  & 1.00  & 0.71  & 0.52  & 0.65  & -0.16 & 0.52  & -0.46 & 0.68  & 0.50  \\
Wagner Resp. & 0.87  & 0.81  & 0.72  & 0.71  & 1.00  & 0.52  & 0.75  & 0.08  & 0.31  & -0.43 & 0.42  & 0.24  \\
Weight       & 0.75  & 0.44  & 0.37  & 0.52  & 0.52  & 1.00  & 0.82  & 0.02  & 0.57  & -0.16 & 0.46  & 0.20  \\
$A_{glu}$    & 0.94  & 0.73  & 0.59  & 0.65  & 0.75  & 0.82  & 1.00  & 0.18  & 0.56  & -0.25 & 0.54  & 0.26  \\
$R_{glu}$    & 0.06  & 0.07  & 0.03  & -0.16 & 0.08  & 0.02  & 0.18  & 1.00  & -0.25 & 0.63  & -0.55 & -0.30 \\
$A_{amm}$    & 0.51  & 0.34  & 0.34  & 0.52  & 0.31  & 0.57  & 0.56  & -0.25 & 1.00  & -0.11 & 0.55  & 0.09  \\
$R_{amm}$    & -0.45 & -0.40 & -0.33 & -0.46 & -0.43 & -0.16 & -0.25 & 0.63  & -0.11 & 1.00  & -0.71 & -0.50 \\
$A_{sul}$    & 0.62  & 0.36  & 0.38  & 0.68  & 0.42  & 0.46  & 0.54  & -0.55 & 0.55  & -0.71 & 1.00  & 0.76  \\
$R_{sul}$    & 0.37  & 0.17  & 0.27  & 0.50  & 0.24  & 0.20  & 0.26  & -0.30 & 0.09  & -0.50 & 0.76  & 1.00  \\ \bottomrule
    \end{tabular}
  \end{footnotesize}
  \caption[Correlations coefficients between amino acid cost estimates]{Spearman's Rank correlation coefficients between estimated and literature described amino acid costs. }
  \label{table:costs_correlation}
\end{sidewaystable}%}}}

The $A_{glucose}$ cost is highly correlated with several previous measures of amino acid cost. The energetic cost derived by Akashi and Gojobori \cite{akashi2002} has the highest correlation coefficient $A_{glucose}$ (Spearman R = 0.94), but $A_{glucose}$ is also correlated (Spearman R $>$ 0.7) with Craig and Weber's energetic cost \cite{craig1998}, Wagner's respiratory cost \cite{wagner2005} and molecular weight \cite{seligmann2003}. This indicates that an systems biology approach using flux balance analysis captures the energetic cost of synthesising an amino acid.

The $A_{ammonium}$ costs show a weak correlation (Spearman R between 0.5 - 0.6) with some of previous described measures of cost: Akashi \& Gojobori, Wagner fermentative, and molecular weight. It should be noted that the $A_{ammonium}$ and $A_{sulphate}$ costs are proportional to that nutrient content of the amino acid.

The $R_{glucose}$ and $R_{ammonium}$ costs show no correlation with previous measures of amino acid cost. The highest correlation is with Wagner fermentative growth (Spearman R = -0.16). The $R_{glucose}$ and $R_{ammonium}$ costs are however correlated with each other (Spearman R = 0.63).

Summarising both the relative and absolute cost types, the absolute costs reflect the per molecule biosynthetic cost, while the relative costs reflect the absolute cost scaled by the quantity of the amino acid in biomass. For example, the $A_{sulphate}$ costs of methionine and cysteine are both 1, which reflects the single sulphur atom in each amino acid. The $R_{sulphate}$ cost for methionine is however much greater than that of cysteine, as methionine is used proportionally more than cysteine.

\subsubsection{Comparison of amino acid costs across species in \emph{S. cerevisiae} and \emph{E. coli}}%{{{3

The relative ease at which amino acid costs can be calculated for species where there is a genome scale model, compared with previous manual curation of cost, allowed comparison of estimated amino acid costs in \emph{S. cerevisiae} and \emph{E. coli}. Figure~\vref{figure:amino_acid_costs_scatterplot} compares both $R_{glucose}$ and $A_{glucose}$ costs for \emph{E. coli} and \emph{S. cerevisiae} with two costs described in the literature: molecular weight \cite{seligmann2003} and number of ATP and NADH molecules \cite{akashi2002}.

\begin{figure}%{{{
\centering
\includegraphics*[width=13cm]{amino_acid_costs_scatterplot.eps}
\caption[Comparison of the genome scale model derived cost data sets.]{Comparison of estimated amino acid cost with number of ATP and NADPH molecules used in synthesis (left), and molecular weight (right). On the y axis are the amino acid costs estimated using flux balance analysis. Both \emph{S. cerevisiae} and \emph{E. coli} measures are included to illustrate correlation of cost estimates between species. Estimated cost values have been been rescaled around their mean value to allow comparisons across species. The trends in each plot are drawn using `loess' smoothing.}
\label{figure:amino_acid_costs_scatterplot}
\end{figure}%}}}

Figure~\ref{figure:amino_acid_costs_scatterplot} shows that the both the absolute costs show a small amount of variance in their correlation with the example literature described cost measures. Both \emph{E. coli} and \emph{S. cerevisiae} absolute costs measures are particularly well correlated with the Akashi and Gojobori described measure of cost \cite{akashi2002}. Comparing the variation between species the \emph{E. coli} $A_{glucose}$ cost is better correlated with this measure than the \emph{S. cerevisiae } equivalent (Spearman R = 0.99 vs. 0.94). This may be expected given the Akashi and Gojobori measures of cost were estimated for \emph{E. coli}. In comparison molecular weight shows a linear relationship with the absolute measures of costs, but shows a greater degree of variation with the costs from both species. Overall the species difference between the two $A_{glucose}$ cost measures is marginal and they are highly correlated (Spearman R = 0.94).

The comparison of relative measures of cost illustrates the small degree of correlation with previous estimates of cost. For the both molecular weight and ATP+NADH cost estimates, neither \emph{E. coli} or \emph{S. cerevisiae} relative cost estimates show any relationship, and furthermore the figure illustrates the disparity in relative costs between species (Spearman R = 0.74) relative to that of the absolute costs.

\subsection{A novel approach to estimating gene cost}%{{{2

The previous section described how genome scale models can be used to estimate biological cost in the context of amino acid synthesis. This section extends this analysis to estimate the fitness cost of a gene via examination of the corresponding \emph{in silico} enzymatic reaction. Gene dispensability has already been explored explored extensively in the literature through deletion of reactions in genome scale models \cite{deleted_reaction_studies}. In this section genome scale models are used to assess the fitness of a reaction in terms of flux, constraints on changes in flux, and how changes in flux produce the corresponding negative effects on the objective function. The aim of each of these analyses is determine a single value for reaction which may then be used to understand the control and impact of the encoding gene on \emph{S. cerevisiae} metabolism.

\subsubsection{Estimation of reaction flux in in glucose, ammonium and sulphate limitation}%{{{3

The first result examines reactions in terms of the differences in flux between three nutrient limiting conditions. The \emph{in silico} flux for of the each single gene-associated reaction, shown in Table~\vref{table:reaction_frequency}, was calculated for glucose, ammonium and sulphate limiting conditions. Each flux was estimated for the optimal and suboptimal flux balance analysis solutions for each nutrient limited environment. The optimal solution simulates the flux distribution across reaction for the minimum amount of the specific nutrient entering the cell. The suboptimal solution finds the flux distribution of a 5\% increase on the minimum amout nutrient entry into the cell, effectively a less optimal use of nutrients.

\begin{table}%{{{
  \centering
  \begin{tabular}{l r r}
                                                                  \toprule
    Reaction Type                       & Optimal & Suboptimal \\ \midrule
    Near zero flux in all conditions    & 176     & 171        \\
    Identical flux in all conditions    & 61      & 51         \\
    Non-trivial flux between conditions & 25      & 40         \\ \bottomrule
  \end{tabular}
  \caption[Comparison of reaction flux across environments]{Comparison of reaction flux in glucose, ammonium, and sulphate limiting conditions. The total number of reactions assessed was 262 and comprises all reactions in the \emph{S. cerevisiae} genome scale model with single gene-association. }
  \label{table:reaction_frequency}
\end{table}%}}}

For each optimal or suboptimal solution reactions with flux less than $\pm0.001$ mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ in all three nutrient limiting conditions were ignored, as were reactions with identical flux across all conditions. This identified the set of reactions which exhibited non-trivial flux differences between glucose, ammonium, and sulphate limiting conditions. These reactions are summarised in Table~\vref{table:reaction_frequency} The distribution of reaction fluxes for the three nutrient limiting conditions in optimal and suboptimal FBA solutions is illustrated in Figure~\vref{figure:flux_distribution}.

\begin{figure}%{{{
  \centering
  \includegraphics*[width=14cm]{reaction_flux_density.eps}
  \caption[Reaction flux in glucose, ammonium and sulphate limitation]{Reaction flux in glucose, ammonium and sulphate limiting conditions. The reactions shown have non-trivial flux differences between conditions. The $x$-axis is the absolute reaction flux on a log$_2$ scale. The leftmost peak in each plot corresponds to reactions with zero flux. }
  \label{figure:flux_distribution}
\end{figure}%}}}

In the optimal solution the ammonium and sulphate limiting have identical flux distributions (the ammonium distribution covers the sulphur distribution in the figure). The shape of the ammonium and sulphate flux distributions is trimodal and skewed towards reactions with a high flux value. The leftmost peak corresponds to unused reactions with zero flux where a small coefficient was added. The middle peak corresponds to reactions with a flux less than 1 while the rightmost peak indicates reactions with a greater flux. The ammonium an sulphate limiting condition have a number of reactions whose flux is greater than 1 in optimal conditions. The glucose optimal flux distribution is also trimodal but is more distributed towards reactions with a low flux or reactions which are unused.

This indicates that single gene reactions carry more flux and are more likely to be used in ammonium or sulphate limitation, when contrasted with glucose limitation. Furthermore the flux distribution in ammonium and sulphate limitation are identical indicating these reactions are used in the same way.

The right plot of Figure~\ref{figure:flux_distribution} illustrates reaction flux in the suboptimal FBA simulation. All three distributions are again trimodal, but in this instance use of flux in ammonium and sulphate limitation are no longer identical. The distribution of flux in ammonium limitation is still skewed towards high flux reactions but exhibit larger numbers of unused reactions. Sulphate limitation maintains a large numbers of both low and high flux reactions with few unused reactions, the glucose flux distribution remains relatively unchanged in the suboptimal solution. Comparing the optimal and suboptimal flux distributions, the number of high flux reactions i reduced in the suboptimal simulation of ammonium and sulphate limitation. Ammonium limitation shows a greater number of unused fluxes in the suboptimal solution. The indicates that sulphate and ammonium require overall less flux in the suboptimal simulation, glucose limitation on the other hand requires very similar flux distribution.

%TODO: Insert correlation values between flux distributions
%TODO: Change color points in flux scatter plot. Pale/Dark grey

\begin{figure}%{{{
  \centering
  \includegraphics*[width=14cm]{flux_scatterplot.eps}
  \caption[Reaction flux in glucose, ammonium and sulphate limitation]{Reaction flux in glucose, ammonium and sulphate limitation. The optimal FBA solution is shown in the pale grey, the suboptimal solution in dark grey. Each axis shows the absolute reaction flux on a log$_2$ scale. }
  \label{figure:flux_comparison}
\end{figure}%}}}

Figure \vref{figure:flux_comparison} compares the reaction fluxes each of the three limiting environments for both the optimal and suboptimal FBA solutions. In the optimal simulation, as expected given the previous result, the ammonium and sulphate flux distributions are perfectly correlated. The glucose limited optimal simulation is weakly correlated with these identical distributions. In the suboptimal simulations the ammonium and sulphate limiting conditions show no correlation, as is also the case for the glucose limited flux distribution with either other flux distribution. The overall picture is that the non-trivial fluxes in each simulation show little correlation with the exception of the ammonium and sulphate limitation which are identical in the optimal solution.

%TODO: Flux distributions cant be correlated, missing values

\subsubsection{Constraints on changes in reaction flux in glucose, ammonium  and sulphate limitation}%{{{3

The variability in flux for each of the single-gene encoded reactions was assessed in all three of the optimal and suboptimal nutrient limiting solutions. Each reaction was categorised based on whether the reaction was used in the solution and whether the reaction flux could be decreased whilst still maintaining the biomass growth rate. The reaction categories are summarised in Table~\vref{table:reaction_use}. Each reaction category is as follows.

\begin{table}%{{{
  \centering
  \begin{tabular}{l l r r}
                                                         \toprule
    Reaction type & Limitation & Optimal & Suboptimal \\ \midrule
    at maximum    & glucose    &       3 &          3 \\
                  & ammonium   &       4 &          4 \\
                  & sulphate   &       4 &          4 \\
    constrained   & glucose    &      60 &         58 \\
                  & ammonium   &      58 &         59 \\
                  & sulphate   &      58 &         59 \\
    variable      & glucose    &      25 &         27 \\
                  & ammonium   &      30 &         33 \\
                  & sulphate   &      30 &         34 \\
    zero flux     & glucose    &     174 &        174 \\
                  & ammonium   &     170 &        166 \\
                  & sulphate   &     170 &        165 \\ \bottomrule
  \end{tabular}
  \caption[Types of reaction constraints in glucose, ammonium and sulphate limitation]{Types of reaction constraints in glucose, ammonium and sulphate limiting simulations. Reactions at maximum are at $\pm$1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ flux. Flux through constrained reactions cannot be decreased in the FBA solution whilst flux through variable reactions can. Zero flux reactions have no flux and are unused in the given condition. The optimal solution find the minimum entry of the nutrient into the cell, the suboptimal solution finds the entry of nutrient into the cell that is 5\% more than required in the minimum entry. }
  \label{table:reaction_use}
\end{table}%}}}

\paragraph{Reaction rate at maximum.} Reactions at the allowable limit ($\pm$1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$) for any reaction in the FBA solution. These represent the boundaries of the FBA solution for each reaction. The numbers of these reactions was limited and no more than four in any condition.

\paragraph{Constrained reaction rate.} The flux through the reactions is constrained in the solution and cannot be reduced whilst still maintaining the FBA objective function. These represent fixed reactions in producing biomass. The numbers of these reactions was relatively invariant between all considered FBA solutions ranging from 58 to 60.

\paragraph{Variable reaction rate.} These reactions are used in the solution, and the absolute reaction flux can be reduced toward zero whilst still maintaining a viable FBA solution. These reactions represent flexibility in the flux phenotypes of the metabolic network. The number of these reactions type varied between solutions ranging from 25 to 34.

\paragraph{Reaction unused.} These reactions have zero flux and are unused in the FBA solution. This is the largest proportion of reactions across all simulations ranging from 165 to 174.

There is little difference between reaction categories when comparing the optimal and suboptimal FBA simulations. The only visible trend is for greater numbers of variable reactions in the suboptimal simulations compared with the optimal simulation. This is as might be expected given that the suboptimal solution is less constrained in determining the flux distribution.

\begin{figure}%{{{
  \centering
  \includegraphics*[width=15cm]{reaction_constraints.eps}
  \caption[Overlap of reaction type between glucose, ammonium and sulphate limiting conditions]{Overlap of single gene-associated reaction flux type between glucose, ammonium and sulphate limiting conditions in the suboptimal solution. Reactions at maximum are at $\pm$1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$. Constrained reaction flux cannot be reduced whilst variable reactions can. Zero flux reactions are unused in the solution. }
  \label{figure:reaction_constraints}
\end{figure}%}}}

The overlap of reaction categories between the suboptimal solution is illustrated in Figure~\vref{figure:reaction_constraints}. The reactions at maximum rate are not included given the limited number across conditions. The figure highlights that the majority of reactions display the same flux characteristics across all limiting environments. This is visible as the largest bar in each of the three subplots corresponds to reaction categories the same across all environments.

For the constrained reaction category there were only small numbers of reactions constrained in one environment but not the other two. There are almost no reactions constrained in two environments but not the third. In the variable reaction category the numbers of reactions specific to a single environment was almost half that of the reactions variable across all three. The numbers of reactions that were variable in two environments but not the third was almost the same as those consistently variable across all environments this indicates variable reactions constraints, in terms of relative proportions, tend to vary between environments. In the last category zero flux reactions, unused reactions across all environments was large majority. There were a limited number of reactions unused only in glucose or ammonium limitation, but almost no reactions unused only in sulphate limitation. The number of unused reactions mixed between environments was however much larger than those specific to a single limitation.

\subsubsection{Reaction control on glucose, ammonium, and sulphate uptake}%{{{3

Metabolic control analysis, described in the introduction, showed how the derivative between the change in one reaction rate and the correspond response in a second reaction can be used to measure reaction sensitivity. In this section, in a similar vein to MCA, perturbations are applied to each reaction to estimate the effect of the reaction on nutrient uptake. This analysis was applied to all the single gene-associated reactions in glucose, ammonium and sulphate limitation in both optimal and suboptimal solutions. Figure~\vref{figure:sensitivity_density} shows the distribution of reaction sensitivities for all nutrient limitations in both optimal and suboptimal FBA solutions. The median reactions sensitivities are shown in Table~\vref{table:median_sensitivity}

\begin{figure}%{{{
  \centering
  \includegraphics*[width=15cm]{reaction_sensitivity_density.eps}
  \caption[Reaction sensitivity in glucose, ammonium and sulphate limitation]{Reaction sensitivity in glucose, ammonium and sulphate limiting conditions. Each sensitivity is the log$_10$ absolute value of the slope between the decrease in reaction flux, and the corresponding effect on nutrient uptake flux. }
  \label{figure:sensitivity_density}
\end{figure}%}}}

\begin{table}%{{{
  \centering
  \begin{tabular}{l l r}
                                                    \toprule
    Limitation  & Solution   & Median (log$_10$) \\ \midrule
    glucose     & optimal    &  -9.34            \\
                & suboptimal & -10.35            \\
    ammonium    & optimal    & -11.53            \\
                & suboptimal &  -8.11            \\
    sulphate    & optimal    &  -7.73            \\
                & suboptimal &  -7.10            \\ \bottomrule
  \end{tabular}
  \caption[Median log$_10$ reaction sensitivities]{Median log$_10$ reaction sensitivity on nutrient uptake in each nutrient limiting FBA simulation. }
  \label{table:median_sensitivity}
\end{table}%}}}

In the optimal solution, glucose limitation is the environment with the most sensitive reactions having a peak furthermost to the right. Ammonium limitation exhibits one highly sensitive reaction while ammonium uptakes i relatively to changes in all other reactions. Sulphate limitation shows a distribution of reaction sensitivities in the mid range of reaction sensitivities. The median values of 10$^{-9}$, 10$^{-12}$, 10$^{-7}$ indicate that the reactions generally have little effect on nutrient uptake in the optimal solution.

The suboptimal distribution of reaction sensitivities shows a lesser degree of reaction sensitivities than those estimated in optimal solution. Glucose and sulphate limitation show a small number of sensitive reactions close to 10$^0$ while the remainder of the reactions are relatively insensitive across any condition.

%TODO: Changes figure axis to "Reaction sensitivity in the optimal solution"

\begin{figure}%{{{
  \centering
  \includegraphics*[width=8cm]{sensitivity_scatterplot.eps}
  \caption[Comparison of reaction sensitivity between optimal and suboptimal solutions]{Comparison of reaction sensitivity between optimal and suboptimal solutions. For each sensitivity is the log$_10$ absolute value of the slope between the decrease in reaction flux, and corresponding effect on nutrient uptake flux. Loess smoothing is used to indicated trend. }
  \label{figure:sensitivity_scatterplot}
\end{figure}%}}}

\begin{table}%{{{
  \centering
  \begin{tabular}{l r r}
                                                    \toprule
    Limitation  & Spearman's rank & Observations \\ \midrule
    glucose     & 0.367           & 23           \\
    ammonium    & 0.291           & 20           \\
    sulphate    & 0.243           & 22           \\ \bottomrule
  \end{tabular}
  \caption[Spearman's Rank between optimal and suboptimal solutions]{Spearman's Rank between optimal and suboptimal solutions. The count is the number of reactions with a calculated sensitivity for the same environment in both the optimal and suboptimal FBA solution. }
  \label{table:sensitivity_correlation}
\end{table}%}}}

%TODO: Not all values have a corresponding value in the other sub/optimal solution (Figure)

Figure~\vref{figure:sensitivity_scatterplot} compares suboptimal and optimal reaction sensitivity estimation in the same environment. The correlation between reaction sensitivities in optimal and suboptimal FBA solutions is shown in Table\vref{table:sensitivity_correlation}. The glucose sensitivity estimates are poorly correlated and show a group of reactions which are highly sensitive in the optimal solution, but with limited sensitivity in the suboptimal solution. This indicates these reactions are particularly constrained in the optimal solution space. The ammonium and sulphate reaction sensitivities are are relatively correlated between the optimal and suboptimal solutions, but do display non-linearity in the relationship.

\clearpage

\section{Discussion}%{{{1

This chapter illustrated the methods used to estimate amino acid cost, and further showed the correspondence of these amino acid costs with previously described measures of amino acid cost.

\subsection{Flux balance analysis predicts amino acid cost}%{{{2

The absolute amino acid costs estimated using the genome scale model are as might expected. The $A_{ammonium}$ and $A_{sulphate}$ costs are directly proportional to the nitrogen and sulphur content of the molecule. This indicates, when either of these two nutrients are limiting the cost of the amino acid is the material quantity of the limiting nutrient. The estimated $A_{glucose}$ cost is different as glucose is the source of both carbon atoms and energy to the cell. When compared with literature reported cost measures the $A_{glucose}$ cost is most correlated with the Akashi and Gojobori energetic cost. For both these measures of amino acid cost, as is also shown in the literature, small less complex amino acids such as glycine and alanine are cheaper while larger more complex amino acids such as tryptophan and tyrosine are biosynthetically more expensive.

The relative amino acid cost estimations represent the absolute cost of an amino acid scaled by the quantity measured in \emph{S. cerevisiae} biomass. The relative measures of amino acid cost show little correlation with previous measures of amino acid cost. Furthermore amino acids that are expensive based on their absolute cost, are cheaper in terms of relative cost. For example tryptophan has a high $A_{glucose}$ cost, but is instead much cheaper when the $R_{glucose}$ cost is considered. This reflects that expensive amino acids may be minimised in biomass composition. On the other hand lysine has one of the most expensive $R_{glucose}$ costs, but is relative inexpensive in $A_{glucose}$. This means that a proportional increase in the requirement of lysine would have a greater effect than of tryptophan, since lysine appears at greater quantities in biomass.

\subsubsection{Correlation $R_{glucose}$ and $R_{ammonium}$ indicates genome optimisation}%{{{3

An interesting point is that the $R_{glucose}$ and $R_{ammonium}$ costs have a high correlation (Spearman R = 0.63). This leads to the possibility that optimisation of energy/carbon based costs in the proteome also results in optimisation of the nitrogen cost. The benefits of this are intuitive where selective advantages for the optimisation of a given amino acid in carbon limitation will have advantages for nitrogen limitation. If there was little or negative correlation between carbon and nitrogen based costs, then optimisation of a protein for carbon limitation, could result in a non-optimisation of the protein under nitrogen limitation.

\subsubsection{Atomic material content as a biosynthetic cost}%{{{3

Amino acid synthesis will require energy to power each step in the metabolic network. In addition to these costs the atomic material content of the amino acid structure may also be considered a cost of synthesis. Mazel \& Marliere \cite{mazel1989} showed that \emph{Calothrix} proteins expressed in sulphur limiting conditions are depleted for the sulphur containing amino acids cysteine and methionine. This work indicating the atomic content of an amino acid should also be taken into consideration when attempting to define a measure of cost. Work by both Baudoin-Cornu \emph{et al.} \cite{baudoin2001} and Bragg \emph{et al.} \cite{bragg2006} used atomic content as a measure of amino acid cost when examining protein content, when considering the evolution role of an amino acid cost, the role of energy should also be coupled with the availability of nutrients. Any limitation on a specific nutrient may play as an important role as that of energetic limitation.

\subsection{Predicting the cost of a gene}%{{{2

\subsubsection{Interpreting gene cost using reaction phenotype} %{{{3

Large stoichiometric models are easier to construct than kinetic models, as less data is required for construction. This makes genome scale models more amendable to large scale of simulation of \emph{in silico} phenotypes, such as the understanding importance of individual genes and reaction to growth. Gene dispensability has been estimated using genome scale models through removal of the encoding reaction and measuring the resulting \emph{in silico} knockout phenotype. Gene loss is a relatively infrequent event compared with small mutations in the gene sequence \cite{pal2006}, in this chapter examine gene fitness was examined in terms of the reaction rather than complete removal of the reaction.

The different estimates of gene considered were: reaction flux, constraints on changes in reaction flux, and how reductions in reaction flux affect nutrient uptake. Each of these estimates was performed for glucose, ammonium, and sulphate limitation to relate to specific environments and nutrient metabolisms. These fitness effects were estimated for both optimal and suboptimal flux balance analysis to examine if a lesser level of constraint on model affects gene fitness estimates.

Estimating the cost of a gene in terms of metabolic phenotype first requires determining each gene's metabolic points of effect. This analysis was restricted to the set of 262 genes in the \emph{S. cerevisiae} model associated with only a single metabolic reaction. This excluded the set of 488 genes with multiple reaction associations. Alternative approaches used in \emph{in silico silico} gene knockout studies have treated reactions as functional paralog for the reactions, or have set the reaction as non-functional if the one unit of a multimeric enzyme is deleted \cite{pal2006}. In this analysis reactions are not being removed, but instead the size of the fitness effects on nutrient uptake estimated. Excluding paralogs or multimeric enzymes from this study creates a smaller dataset but allows a more direct interpretation of gene cost where any effect on the gene may be expected to have only a single corresponding effect on one reaction.

\subsubsection{Gene cost using reaction flux}%{{{3

Reaction flux was examined as a measure for gene cost. Mutational effects in genes encoding high flux reactions may be expected to have a resulting effect   in reactions with low flux. Of the all the single gene-associated reactions, only 10\%-14\% exhibited non-trivial flux difference between glucose, ammonium and sulphate limitation. Flux balance analysis the limited fluxes were ammonium and sulphate identical indicating the flux phenotype for single gene-associated reactions is the same for ammonium and sulphate limitation. The difference in flux distributions for the optimal solutions in ammonium and sulphate limitation must therefore be encoded by the enzymes not included in the analysised set comparing flux distributiions. The glucose limited flux distribution for single gene-associated reactions was relatively lower than than of ammonium and sulphate limitation indicating that single encoded reaction, on average, carry less flux in glucose limitation.

When estimating reaction flux using suboptimal simulation the glucose limited flux distribution is relatively unchanged compared with the optimal solution. The ammonium and sulphate limiting flux distributions do however vary between optimal and suboptimal flux balance solutions. This indicates that relaxing solution constraint has little effect on single gene-associated reactions in glucose limitation, but does change overall reaction flux in sulphate and ammonium limitation.

Comparing flux distributions between single gene associated reactions across nutrient limitation showed little correlation, with the exception of ammonium and sulphate limitation in optimal conditions which are identical. The compared data (seen in figure~\vref{figure:flux_scatterplot}) are non-trivial flux differences between environments and shows that the changes in these reactions are not correlated.

\subsubsection{Gene cost through constraints on reaction flux}%{{{3

The constraints on decreases in reaction flux represent the edges of the FBA solution space when optimising the \emph{S. cerevisiae} model. The reactions at maximum flux represent the boundaries on the linear optimisation where all reactions are restricted to within $\pm$1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$. The biological feasibility of whether a reaction may reach this level of flux is also questionable, given that the average reaction flux is much lower.  The number of observed reactions at this maximum level is low and discarding them does not significantly reduce the size of the gene fitness dataset for later use.

The constrained reactions represent those whose flux is at an edge in the FBA solution space. Forced reduction of the flux through any of these reactions prevents the production of a viable FBA solution where linear optimisation cannot be solved. As the flux cannot be reduced, these reactions all represent inviable knockout mutants where removal from the model produces a lethal phenotype. These reactions may be functionally important \emph{in vivo} as the \emph{in silico} simulation suggests that nutrient uptake is most sensitive to changes in these reaction fluxes. Comparing the frequency of constrained reactions in the suboptimal FBA simulation there was still a significant number, indicating constraint on reaction flux is present even in more relaxed model optimisation.

The variable reactions are those where the flux can be reduced and still produce a viable model solution. The variability observed in these reactions may be due to alternate pathways for the production of the same metabolite, or that the flux through a reaction may be increased to compensate. Changes in variable reaction fluxes represent non-unique flux distributions where multiple different combinations of flux distributions can produce the same \emph{in silico} phenotype. \emph{In vivo} it would be unclear as the role of these reactions, though it is likely the effect a reduction in reaction flux would have a lesser effect than of r a constrained reaction. Comparing the numbers of variable reactions in the suboptimal solution with optimal solution, the suboptimal solution identifies more flexible reactions, which may be expected as the solution space is less constrained.

The zero flux reactions are unused in the \emph{in silico} flux phenotype and are the majority of single gene-associated reactions. These reactions could be removed from the model without any effect on producing a viable FBA solution. These reactions could represent environment specific reactions, required in the combination nutrients not simulated in this analysis. The numbers of zero flux reaction sis relatively unchanged between optimal suboptimal FBA simulation, if these zero flux reactions are environment specific then changes in solutions constraints are unlikely to have any effect. The \emph{in vivo} fitness effect of these reactions is likely to specific to only the environments where the reaction is required.

Comparing the overlap of reaction type between each of the three nutrient limited environments highlights the majority of reaction categories are consistent across environments. This suggests, in the conditions and single gene-associated reactions examined, that reaction constraints do not differ between environments. One hypothesis is that the core metabolic requirements of the biomass are required independent of environment, placing consistent constraints on the same reactions. Alternatively significant changes in reactions constrained may be observed for multiple gene-associated reactions, or through observing a wider range of conditions.

\subsubsection{Gene cost through phenotypic effects on nutrient uptake}%{{{3

Given that only a small fraction of all single gene-associated reactions are variable in the considered FBA simulation, the number of reactions for which a sensitivity could be estimated was small compared with the total number of reactions. Furthermore of this small set of reactions few had a sensitivity close to 1, where the median reactions sensitivity was approximately $10^{-8}$. This indicates of the variable reactions of which sensitivity was estimated, the majority of reactions produced only minor effects on nutrient uptake and therefore fitness when reaction flux is reduced.

The most sensitive nutrient uptake to changes in reaction flux was glucose. This may be expected since glucose entry and metabolism is responsible for the production of the high energy molecules used in biosynthetic reactions. The ammonium and sulphate uptakes might be expected to only be sensitive to changes in reactions that produce metabolites requiring nitrogen or sulphur, however more analysis is required to confirm from these hypothesis.

One argument that most reactions have little effect on nutrient uptake flux could be attributed to biological robustness, where metabolism is able to buffer against deleterious changes in reaction activity. However should not be a large number of highly constrained reactions, where any changes in reaction flux relevant an optimal FBA solution. The estimation of reaction sensitivity is likely to highly dependent on the solution space when optimising a stoichiometric model. The sensitivity of reaction is highly dependent on how the model was optimised, this is readily apparent in he comparison of optimal and suboptimal reaction sensitivity for the same reactions, where there was little correlation.

\clearpage

\section{Conclusion}%{{{1

This chapter has outlined how genome scale models and flux balance analysis can be used to estimate both the cost of an amino acid and the fitness effect of a gene. Amino acid cost estimates are correlated with previous measures of amino acid cost described in the literature, and validates this method for estimating amino acid cost. In comparison, gene cost is examined in the literature in terms of dispensability: if reaction removal has a lethal phenotype the the gene is considered important. Here instead of dispensability gene cost is interpreted using reaction flux, constraints of reaction flux, and the effect of small changes in reaction flux on nutrient uptake. This is a novel approach to understanding the importance of a gene and compares to no existing methods in the literature.

Both of these costs are examined further chapters. In the following chapter amino acid cost in gene expression and evolution considered in the following chapter. Gene cost on evolutionary rate and observed \emph{in vivo} phenotypes is considered in the chapter after that.

% vim:foldmethod=marker
