\section*{Chapter summary}

This chapter illustrates a novel approach to estimating the cost and importance of amino acids and genes \emph{Saccharomyces cerevisiae}. The methods described apply existing systems biology control frameworks to a genome scale model of the yeast metabolism to determine the biosynthetic cost of amino acid synthesis, and the constraints on the use of gene-encoded functions in different environmental conditions.

The background section describes the theory of metabolic control analysis and how this applied to determine the structure and function of cellular metabolism. This is followed by an overview of the construction of genome scale models and their simulation using flux balance analysis.

The results section first outlines the novel method used to estimate the biosynthetic cost of amino acid synthesis in glucose, ammonium, and sulphur limitation. The second half of the results describes a method to determine the constraints on the use of gene-encoded reactions in glucose and ammonium limited conditions to understand the importance of a reaction and there the encoding gene.

\clearpage

\section{Introduction}

\subsection{Understanding control in metabolic models}

Early enzymology assumed the existence of rate limiting steps in biological pathways. The intuition was that the overall rate of a pathway is constrained by the rate of the slowest step. As the slowest reaction rate increases the whole pathway rate increases in proportion until the next step becomes limiting. Niederberger \emph{et. al} \cite{niederberger1992} tested this assumption and found that individual up or down regulation of enzyme quantities at specific reaction steps had only marginal effect on overall tryptophan biosynthesis in \emph{Saccharomyces cerevisiae}. The rate of the pathway could instead be accelerated by increasing the quantity of five related enzymes in tandem. This research demonstrated control in a biological system is distributed over the system as a whole rather than concentrated at individual reactions.

The theory of metabolic control analysis (MCA) \cite{fell1992,fell1997} states that there is no rate limiting step in a pathway, but instead each reaction shares a measure of overall control of metabolism. The control the rate of example reaction $x$ has on reaction $y$ is determined through applying a small change in the rate of $x$ and measuring the degree of response in the rate of $y$. In metabolic control analysis this is described as the flux control of reaction $x$ on reaction $y$. The coefficient of this control uses the symbol $C_{x}^{y}$ and is defined as:

\begin{equation}\label{fcc}
C_{x}^{y} = \frac{ \delta \ln x / \ln x}{\delta \ln y / \ln y} \, .
\end{equation}

The numerator ($\delta \ln x / \ln x$) represents the dimensionless effect of a small change in the reaction flux through $x$. The denominator $(\delta \ln y / \ln y$) represents the corresponding response in $y$ resulting from the change in $x$. The value of $C_{x}^{y}$ represents the ratio of response in reaction $y$ to changes in the rate of $x$. Larger values of $C_{x}^{y}$ indicate the reaction $x$ has a high degree of control on reaction $y$. Smaller values indicate reaction $y$ experiences only a small change in flux in response to changes in $x$. Using this approach a quantitative measure of control may be derived for each reaction in a biological system.
\nomenclature{MCA}{Metabolic Control Analysis}
\nomenclature{$C_{x}^{y}$}{Flux control coefficient of reaction $x$ on reaction $y$}

Determining a numeric value for the relationships between reactions in a metabolic network allows quantitative exploration of biological systems. For example Rossell \emph{et al.} described the control of metabolism in terms of reactions whose rates are increased by changes in their substrate concentration or the reactions whose activity is instead controlled through changes in enzyme quantity and therefore via gene expression \cite{rossell2006,daran-lapujade2007}.

An example of the application MCA is engineering an increase in the production of commercial biomolecules \cite{niederberger1992}. MCA can also be used to identify possible drugs targets through the combination of reactions knock-downs which have the greatest effect on the metabolic network \cite{lehar2008,hopkins2008}. This can be further expanded in developing antibiotic drugs for human pathogens such as trypanosome parasites, where drug targets are those reactions with a significant effect on metabolism in the pathogen but where the orthologous human reaction has a much decreased effect. This difference in metabolic control for the same reaction in both species translates into lethal drug effect in the parasite with minimal impact on the host \cite{hornberg2007}.

%TODO: read Hopkins, Lehar papers

\subsection{Genome scale models}

Derivation of a flux control coefficient \emph{in silico} requires a kinetic model including enzyme kinetic parameters, in the case of simple Michaelis-Menten kinetics only substrate affinity (K$_{m}$) and maximum enzyme reaction rate (V$_{max}$) are necessary. More complex enzymatic kinetics requires larger parameter sets which require further experimental effort to derive. Even with all experimentally derived kinetic parameters, a model must still be tested to determine if it matches expected \emph{in vivo} behaviour \cite{teusink2000}. The experimental effort required to derive enzyme kinetic parameters means the kinetic models of metabolic systems are relatively small compared with the anticipated size of cellular metabolism \cite{steuer2007}.
\nomenclature{K$_{m}$}{Substrate concentration where enzyme rate is at half maximum ($\frac{1}{2}$V$_{max}$)}
\nomenclature{V$_{max}$}{Maximum reactions per second per mole of enzyme}

%TODO: check Steuer paper

In comparison with kinetic models, a stoichiometric model does not require enzyme kinetic parameters, only the connectivity between reactions. A stoichiometric model is represented by an $m \times n$ sized matrix $S$. The matrix $S$ represents the reactions in metabolism and the metabolites that participate in the reactions. The size of of $m$ is the total number of metabolites in the model, and $n$ is the total number of reactions. The position $S_{ij}$ is the participation of metabolite $i$ in reaction $j$. Positive values of $S_{ij}$ indicate the metabolite is produced by the reaction, negative values mean the metabolite is consumed. A zero value indicates the metabolite does not participate in the reaction.

As stoichiometric models require less data to create it is easier to produce metabolic representations. Stoichiometric models comprising a significant proportion of an organism's metabolism have be created, described as genome scale models. The construction of genome scale metabolic models is reviewed by Feist \emph{et al.} \cite{feist2009}, and the process is outlined in brief here.

The construction of a genome scale model requires the production of the stoichiometric matrix detailing the participation of reactions and substrates in each reaction. Enzyme databases such as BRENDA \cite{chang2009} and KEGG \cite{okuda2008} may provide information on the known interactions enzymes and pathways while species specific databases such as the \emph{Saccharomyces} Genome Database (SGD) \cite{cherry1997} allow the identification of reactions specific to the organism. The initial stages construction can be automated, however expert knowledge of the organism can aid questions about the existence of specific metabolic pathway, particularly for organisms living in niche environments \cite{sun2009}.
\nomenclature{SGD}{\emph{Saccharomyces} Genome Database (see \cite{cherry1997})}

The first \emph{Escherichia coli} model was developed almost 20 years ago \cite{feist2008}. A simplified model of acetate production was described by Majewski \emph{et al.} in 1990, the recent 2007 genome scale model accounts for 1260 open reading frames \cite{feist2007}. The range of species for which a genome scale model exists continues to expand and includes attempts to reconstruct a human cell \cite{duarte2007}. The field of genome scale model construction does however suffer from the lack of a single standard for construction and model format, making model interchange and comparison difficult. There are however attempts to create a unified \emph{S. cerevisiae} model \cite{herrgard2008} using existing standards such the systems biology markup language (SBML) \cite{sbml} as the model format, minimum information requested in annotation of biochemical models (MIRIAM) \cite{lenovere2005} as a standard for model annotation, and standardised encoding and representation of chemicals using the international chemical identifier codes (InChI) \cite{coles2005}.
\nomenclature{InChI}{IUPAC International Chemical Identifier}
\nomenclature{SBML}{Systems Biology Markup Language}
\nomenclature{MIRIAM}{Minimum Information Requested In Annotation of biochemical Models}

When constructing a genome scale model the intended goal is often the simulation of \emph{in vivo} phenotypes which requires an additional reaction describing cellular growth. In many cases this reaction consumes the constituent lipids, proteins, nucleic acids and other cofactors required in a new cell and produces a per hour (hr$^{-1}$) unit of new biomass. The required quantities of each molecule in the growth reaction are estimated from \emph{in vivo} millimoles per gram of dry weight cellular biomass (mmol $^{-1}$ gDW $^{-1}$). The \emph{in silico} simulations of a newly constructed model must be validated against the observed \emph{in vivo} phenotypes, where \emph{in silico} behaviour should reflect that of \emph{in vivo} behaviour as closely as possible \cite{famili2003}. Differences between \emph{in silico} and \emph{in vivo} identify gaps in the model which can be updated followed by repeated model validation. The use and simulation of stoichiometric models is described in the next section.
\nomenclature{gDW}{Grams of Dry Weight biomass}
\nomenclature{mmol}{Millimoles, quantity equal to one thousandth number of atoms in 0.012 grams of carbon$^{12}$}
\nomenclature{hr}{Hour}

\subsubsection{Stoichiometric model simulation using flux balance analysis}

Stoichiometric models are simulated using flux balance analysis (FBA). As described above a stoichiometric model is defined by a matrix $S$ describing the substrates and products used in each reaction. FBA uses linear programming to solve the equation $S v = 0$ where $v$ represents a combination of reaction rates, known as fluxes, for each reaction in $S$. There are multiple solutions to this equation and so linear optimisation is used to determine the combination of fluxes for $v$ that maximises or minimises a particular value, this optimised value is described as the objective function. A common objective function used for genome scale models is maximising growth rate as this reflects \emph{in vivo} behaviour for most single cell organisms. Additional constraints can be placed on the flux balance solution, such as upper and lower bounds for individual reactions. Constraining a reaction to be $\geq 0$ restricts the solution space so the reaction may only proceed in one direction. Reaction constraints are often used in this way to limit the flux of nutrients entering the cell and thereby simulate different environmental conditions.
\nomenclature{FBA}{Flux Balance Analysis}

The aim of flux balance analysis using a genome model is to simulate of organism phenotypes. Minimisation of metabolic adjustment (MOMA) \cite{segre2002,burgard2003} is a technique that optimises the objective function while minimising the sum change in flux adjustment between two flux solutions. The aim of this method is to simulate perturbed metabolic phenotypes such a wild-type and gene knockout. MOMA attempts to simulate effects of changes in metabolism where minimal flux changes more accurately reflect the \emph{in vivo} behaviour of how metabolic adjustments are made. A related approach minimises the number of on-off changes in reaction use (Regulatory on/off minimisation ROOM), whereby the number of reactions that switch between active or inactive between solutions is minimised \cite{shlomi2005}. This approach again mimics \emph{in vivo} behaviour where the use of already active reactions in different flux phenotypes is more realistic simulation compared to large scale metabolic readjustments in the use of reactions.
\nomenclature{MOMA}{Minimisation of Metabolic Adjustment}
\nomenclature{ROOM}{Regulatory on/off minimisation}

The determination of each of these reaction phenotypes relies on the optimisation of the model. The solution space produced by the model may be over constrained the determination of the optimum. The \emph{in vivo} flux distribution may be more relaxed as highly constrained reaction fluxes may provide a competitive disadvantage in response to any environmental or genetic perturbations. Another consideration is whether \emph{in vivo} conditions are able to reach the optimised peak of the solution space, or instead reaching a point very close to the optimum but still maintaining a measure of flexibility in the metabolic network. This theory is discussed Mahadevan and Schilling \cite{mahadevan2003} where reaction fluxes are instead considered over a range of suboptimal solutions near the optimal.

\subsubsection{Predicting essential genes using genome scale models and flux balance analysis}

Estimating metabolic control coefficients using stoichiometric models is difficult as the flux the through the reaction is determined by linear optimisation, rather than enzyme kinetics. A common approach to estimating the importance of a reaction in a genome scale model is to remove a reaction, and compare the resulting solution with the original wild-type simulation. A reaction deletion may be considered to confer a fitness effect if there is a significant growth rate reduction \cite{pal2006}. The deletion of a reaction may prevent model optimisation indicating the reaction is essential for the objective function and that the gene knockout is lethal \cite{becker2008}.  Even if a model solution can still be found, the gene may still be considered essential if the growth defect is large enough \cite{papp2004,wang2009}.

Flux balance analysis of all possible gene deletions for \emph{S.cerevisiae} was calculated and compared with \emph{in vivo} observations. Of the knockout phenotypes up to $\sim$85\% were correctly predicted depending on the simulated media \cite{famili2003,forster2003}. \emph{In silico} knockout studies have been extended Reactions that buffer the effects of each other by comparing all possible double deletions and finding pairs with no phenotypic effect when deleted individually, but indicate a sick or lethal phenotype when deleted simultaneously \cite{harrison2007}. Reactions which may be active in a specific range of conditions can also be assessed through deletion of reactions in a wide range of simulated conditions, where a deletion may result in a fitness effect in only a small range of biologically feasible conditions \cite{papp2004}.

The use of genome scale metabolic models is not limited to estimating the control of reactions on growth, but may also be used to estimate the importance of specific metabolites. Varma \emph{et al.} \cite{varma1993} used flux balance analysis to determine the role of oxygen in \emph{E. coli} energy metabolism in through decreasing oxygen availability and identifying the corresponding change in the use of redox carriers. As oxygen availability decreased, the use of ethanol as the electron acceptor increased. The author described this effect in terms of the price of the each molecule, where the price of oxygen use rises with increasing availability while the price of ethanol decreases. This highlights how systems biology models can be used to estimate the importance of a set of metabolites to cell growth. Few subsequent studies have looked at the price of metabolite in a systems biology framework, though the biosynthetic price of metabolites is hypothesised to have an effect on their use in proteins.

\subsection{Estimating amino acid biosynthetic cost}

Craig \& Weber \cite{craig1998} estimated the cost of an amino acid as the sum of how many high energy phosphate bonds (e.g. ATP) and reducing molecules (e.g. NADPH) are diverted from the available energy pool for the synthesis of each amino acid from glucose. The aim of deriving these costs was to estimate the importance of amino acid cost on the evolution of a set of \emph{E. coli} proteins.

Akashi \& Gojobori \cite{akashi2002} explored a similar approach calculating cost on a range of different energy sources to estimate the role of amino acid cost in predicted \emph{E. coli} and \emph{B. subtilis} gene expression. Heizer \emph{et al.} \cite{heizer2006} expanded this estimating amino acid cost for inorganic sources in their analysis of the genomes of four prokaryotic species including photoautotrophs. Wagner considered the cost in \emph{S. cerevisiae} using a similar approach but considering both respiratory and fermentative energy metabolism. These estimates of cost are based on the curation of the metabolic network to account for the expenditure of high energy molecules during synthesis. Seligmann argued \cite{seligmann2003} the cost of amino acid biosynthesis must take into account more than just the energy required for synthesis. Seligmann used molecular weight as a proxy for amino acid cost arguing that this may take into account investments involved in additional complexity associated with larger amino acids. Using molecular weight also has the advantage where the metabolic network is not required to count the number of high energy molecules. Table~\ref{table:literature_costs} summarises the various measures of amino acid cost.

\begin{table}
\centering
\begin{footnotesize}
  \begin{tabular}{ p{0.8cm} *{6}{p{1.8cm}} }
                                                                                \toprule
          & A\&G   & C\&W   & C\&W  & Wager        & Wagner      & Molecular \\
          & Energy & Energy & Steps & Fermentative & Respiratory & Weight    \\ \midrule
      ala & 11.7   & 12.5   & 1     & 2            & 14.5        & 89.1      \\
      arg & 27.3   & 18.5   & 10    & 13           & 20.5        & 174.2     \\
      asn & 14.7   & 4      & 1     & 6            & 18.5        & 132.1     \\
      asp & 12.7   & 1      & 1     & 3            & 15.5        & 133.1     \\
      cys & 24.7   & 24.5   & 9     & 13           & 26.5        & 121.2     \\
      gln & 16.3   & 9.5    & 2     & 3            & 10.5        & 146.2     \\
      glu & 15.3   & 8.5    & 1     & 2            & 9.5         & 147.1     \\
      gly & 11.7   & 14.5   & 4     & 1            & 14.5        & 75.1      \\
      his & 38.3   & 33     & 1     & 5            & 29          & 155.2     \\
      ile & 32.3   & 20     & 11    & 14           & 38          & 131.2     \\
      leu & 27.3   & 33     & 7     & 4            & 37          & 131.2     \\
      lys & 30.3   & 18.5   & 10    & 12           & 36          & 146.2     \\
      met & 34.3   & 18.5   & 9     & 24           & 36.5        & 149.2     \\
      phe & 52     & 63     & 9     & 10           & 61          & 165.2     \\
      pro & 20.3   & 12.5   & 4     & 7            & 14.5        & 115.1     \\
      ser & 11.7   & 15     & 3     & 1            & 14.5        & 105.1     \\
      thr & 18.7   & 6      & 6     & 9            & 21.5        & 119.1     \\
      trp & 74.3   & 78.5   & 12    & 14           & 75.5        & 204.2     \\
      tyr & 50     & 56.5   & 9     & 8            & 59          & 181.2     \\
      val & 23.3   & 25     & 4     & 4            & 29          & 117.2     \\ \bottomrule
  \end{tabular}
\end{footnotesize}
\caption[Amino acid costs described in the literature]{Amino acid costs described in the literature. The Akashi \& Gojobori \cite{akashi2002}, Craig \& Weber energy \cite{craig1998}, and the two Wagner \cite{wagner2005} data sets are based on the curation of the number of high-energy molecules used during synthesis. The Craig \& Weber `steps' measure \cite{craig1998} is based on the number of the number of biosynthetic steps between central metabolism and the produced amino acid. Molecular weight is in Daltons.}
\label{table:literature_costs}
\end{table}

These studies however require manual examination of a metabolic network to determine the number of energy molecules expended in the synthesis of each amino acid. This is a time-consuming activity and susceptible to human error. No approaches to estimating amino acid cost have considered the using genome scale models is similar approach to Varma \emph{et al.} to determine the cost of an amino in varying considering taking into account the complete metabolic network of the organism.

\subsection{Results summary}

The results presented in this chapter are in two sections. The first describes an approach to estimating amino acid cost focusing on the \emph{S. cerevisiae} genome scale metabolic model. The approach used to estimate amino acid cost is similar to that described by Varma \emph{et al.} in their estimation of the oxygen price in \emph{E.coli} energy metabolism. The estimated costs are compared to the further estimates of amino acid cost that appear in the literature described above.

The second part of results section describes an related approach to estimate the cost of a gene. This is again used the \emph{S. cerevisiae} genome scale metabolic model. This approach performs robustness analysis of the encoded metabolic function of the gene. This analysis produces a robustness distribution describing how increasing and decreasing metabolic rate for a gene-encoded function affects the \emph{in silico} model fitness.

\clearpage

\section{Materials and Methods}

\subsection{Determination of gene to reaction association in \emph{S. cerevisiae} model}

Only a subset of the reactions in the \emph{S. cerevisiae} metabolic model have a direct gene association. This annotation of the model defines if there is one or more genes known to encode an enzyme catalysing one of the reactions in the model stoichiometric matrix. This association can include multimeric enzymes where multiple gene products form an enzyme complex or are isoenzymes where multiple gene encode enzymes for the same reactions.

The COBRA toolbox used in the analysis defines a sparse matrix specifying which \emph{S. cerevisiae} genes are associated with which model reactions. The matrix is sparse as not all reactions have an associated gene. This matrix was parsed to determine the reactions catalysed by only a single gene, where the gene is not associated with any other reactions.

\subsection{Flux balance analysis}

The genome scale models used in this work were \emph{S.cerevisiae} iND750 \cite{duarte2004a} and \emph{E.coli} iJR904 \cite{reed2003}. Flux balance analysis was performed using the COBRA toolbox \cite{becker2007} and the lpsolve optimisation library \cite{lpsolve}.
\nomenclature{COBRA}{Flux balance analysis toolbox for MATLAB (see \cite{becker2007})}
\nomenclature{iND750}{\emph{Saccharomyces cerevisiae} genome scale model (see \cite{duarte2004a})}
\nomenclature{iJR904}{\emph{Escherichia coli} genome scale model (see \cite{reed2003})}

Nutrient limitation was simulated by fixing the upper and lower bounds of the biomass production reaction to 0.3$^{-1}$ an optimising to minimise the entry of a specific nutrient into the cell. All other nutrient exchange fluxes were set to have bounds of -10000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$, effectively setting all other nutrients to be non-limiting on simulated growth. The flux balance analysis configuration therefore finds the minimum required amount of a specific nutrient to maintain cellular growth rate at 0.3 hr$^{-1}$.

For the \emph{S. cerevisiae} iND750 \cite{duarte2004a} genome scale model the reactions allowing entry of nutrients into the cell are for glucose, ammonium, sulphate, phosphate, oxygen and water. Each of these reactions represents the only external source for an essential nutrient and, for example, when simulating the glucose limitation there are no other high energy sugars available. The FBA simulations does however allow a variety of molecules to exit the cell, these are in the majority metabolic by products such as carbon dioxide or ethanol.

Fixing \emph{in silico} growth rate also simulates experimental conditions using continuous culture of microorganisms in the chemostat (see Castrillo \emph{et al.} \cite{castrillo2007}) where the minimisation of a specific nutrient as the \emph{in silico} objective mimics chemostat conditions where one nutrient is the limiting factor for growth and all other nutrient are available in excess.

\subsection{Flux balance analysis estimation of amino acid cost}

Estimation of amino acid cost was performed by making a percentage change ($\pm$0.0002\%) to the requirement of amino acid at position $S_{ij}$ in the model stoichiometric matrix, where $i$ is the amino acid and $j$ is the biomass reaction. When the model is optimised each change in $S_{ij}$ results in a corresponding change in the limiting nutrient uptake. Larger effects in nutrient uptake indicate a greater nutrient cost for the amino acid.  The slope between the change in amino acid requirement and the corresponding change in nutrient uptake flux was taken as the cost of the amino acid for that nutrient. This cost is a relative cost as it is derived from a percentage changes in amino acid requirement rather than constant changes in requirement. Division of the relative cost by the original amino acid requirement does however result in an absolute estimate of amino acid cost. Both relative and absolute amino costs were calculated for glucose, ammonium, and sulphate limitation. Finally each cost was divided by the growth rate at which it was estimated (0.3 hr$^{-1}$) to give an estimate of cost independent of growth. The nomenclature used for relative and absolute cost definitions is $R_{nutrient}$ and $A_{nutrient}$ respectively.
\nomenclature{$A_{nutrient}$}{Amino acid absolute nutrient cost}
\nomenclature{$R_{nutrient}$}{Amino acid relative nutrient cost}

The additional optimisation techniques described in the introduction, such as MOMA or suboptimal solutions, were not used in the estimation of amino acid cost. These techniques focus on the realistic \emph{in vivo} simulation reaction fluxes, while instead this analysis focused on the corresponding change in nutrient uptake flux given the change in amino acid stoichiometry.

\subsubsection{Units of absolute and relative measures of amino acid cost}

The relative costs are calculated value of $m$ in the linear relationship $y = mx + c$ where $x$ is the unit less percentage change in the amino acid requirement, and $y$ is corresponding mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ response in nutrient intake flux. When divided by the hr$^{-1}$ growth rate at which the cost was estimated, the units of the relative cost are mmol$^{-1}$ gDW$^{-1}$. This represents the change in nutrient uptake flux given the fractional change in amino acid requirement in biomass. The absolute cost is derived from the relative cost by division by the original mmol$^{-1}$ gDW$^{-1}$ requirement of the amino acid. The absolute amino acid is therefore unit less and represents what a mmol$^{-1}$ gDW$^{-1}$ increase in amino acid requirement has the mmol$^{-1}$ gDW$^{-1}$ uptake flux of the given nutrient.

\subsection{Flux balance analysis estimation of gene cost}

\subsubsection{Estimation of \emph{in vivo} flux distributions using suboptimal flux balance solutions}

The distribution reaction flux in the \emph{S. cerevisiae} model in of each three nutrient limited conditions, glucose, ammonium, and sulphate, was estimated as above where growth rate is fixed at 0.3 hr$^{-1}$ and the model optimised for the minimum intake of the given nutrient.

This solution was then used to find a suboptimal solution for the same nutrient limiting condition by a setting a lower boundary on the minimum flux of nutrient into the cell at 5\% of the minimum allowable flux. For example if the minimum glucose flux into the cell at growth rate 0.3 hr $^{-1}$ is $x$, then the lower boundary of glucose into the cell was set at 1.05$x$, where the amount of glucose entering the cell may increase, but not decrease below this value.

The model was re-optimised to determine the less optimal solution and corresponding distribution of internal reaction fluxes. This aim of approach is to simulate \emph{in vivo} flux distributions whereby the cellular state is very close to an optimum growth rate while still maintaining a measure of variability in the metabolic network \cite{mahadevan2003}.

\subsubsection{Determination of reaction flux in glucose, ammonium, and sulphate limitation}

The fluxes for all reactions with a single gene association were estimated for glucose, ammonium, and sulphate limited suboptimal flux balance analysis solutions in the yeast model. Reactions at zero flux in all three conditions were ignored. To identify changes in reaction fluxes between environments each reaction flux was scaled by subtracting the median value across all three environments.

\subsubsection{Determination of reaction variability in glucose, ammonium, and sulphate limitation}

In each nutrient limited condition, each reaction was classified based on use in the suboptimal solution space.  The first two categories were based on whether the reaction flux was zero (`zero flux'), or at the maximum allowable flux of $\pm$1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$ (`at maximum').

If the reaction did not fall into either of these two categories it was then classified on whether the flux could vary in the FBA solution given the cell growth rate and nutrient limitation. For each remaining reaction, the model was re-optimised using MOMA to minimise the absolute flux through the reaction. If the reaction could not be reduced whilst maintaining the same growth and nutrient uptake flux, this indicated the reaction was constrained in the solution space (`constrained'), otherwise the the reaction was classified as variable (`variable').

This resulted in one of four classifications for each single gene-associated reaction in each of the three nutrient limitations considered.

\subsubsection{Determination of reaction flux control on in glucose, ammonium, and sulphate uptake}

The degree of control for each variable-type reaction was estimated for glucose, ammonium, and sulphate limiting conditions.

To estimate the reaction control on nutrient uptake, the flux through each reaction was fixed at one of five points in the range of 100\% - 95\% of original reaction flux in the initial suboptimal solution. For each point change in the the model was re-optimised using MOMA resulting in a corresponding reponse in the nutrient uptake flux, given the change in reaction flux.

The slope of the regresion in change in the nutrient uptake flux as a response in to changes in reaction flux was the cost for the reaction in that nutrient limitation.

%TODO: what units?

\clearpage

\section{Results}

\subsection{The \emph{S. cerevisiae} iND750 genome scale model}

\subsubsection{Numbers of reactions and genes}

To analyse reaction function and infer characteristics of \emph{S. cerevisiae} genes the relationship between model reactions and annotated genes must first be understood. Table~\vref{table:gene_associations} illustrates the number and types of associations between genes and reactions. Each reaction in the model can be categorised based on the type of associations with annotated genes.

\begin{table}[b]
  \centering
  \begin{tabular}{l r}
                                                                   \toprule
    Description                                      & Number   \\ \midrule
    Genes                                            & 750      \\
    Reactions                                        & 1266     \\
    Reactions with at least one associated gene      & 810      \\
    Reactions with a single gene associated          & 579      \\
    One-to-one association between reaction and gene & 262      \\ \bottomrule
  \end{tabular}
  \caption[Gene associations in the \emph{S. cerevisiae} iND750 model]{Comparison of numbers and types of associations between genes and reactions in the \emph{S. cerevisiae} iND750 model. }
  \label{table:gene_associations}
\end{table}

The total number genes included in the model is 750. Of the 1266 reactions, 810 are associated with one or more genes. This includes reactions that may be catalysed by paralogs, multimeric enzymes, and genes also associated other reactions. A subset of this category are the reactions associated to only a single gene containing 579 reactions. This excludes paralogs and multimeric enzymes, but still includes enzymes associated with more than one reaction. A further subset of this category are the reactions associated to a single gene, where the gene is associated with no other reactions. This comprises 262 genes which may be expected to have only a single point of effect in \emph{S. cerevisiae} metabolism are therefore useful for inferential analysis of the relationship between gene function and evolution, and corresponding metabolic effect.

\subsubsection{Composition of biomass requirement}

As discussed in the introduction, the biomass equation in a stoichiometric model is an estimated requirement of essential metabolites for growth in a given organism. In flux balance analysis, the biomass equation can be used as the objective function to simulate the rate of growth given the available basic input nutrients. Table \vref{table:biomass_requirements} summarises the molecular requirement of the biomass reaction in the \emph{S. cerevisiae} iND750 model. These requirements include fats such as triglyceride and ergosterol, sugars such as glycogen and mannan, and small metabolites comprising sulphate, water, and ATP. The genome, transcriptome, and proteome content of the cell is included as the quantities of nucleotides and amino acids estimated from dry weight cellular composition.

\begin{table}
  \centering
  \begin{footnotesize}
  \begin{tabular}{l l r}
                                                                                              \toprule
    Type           & Molecule                       & Requirement (mmol$^{-1}$ gDW$^{-1}$) \\ \midrule
    Nucleic Acid   & dAMP                           &  0.00360                             \\
                   & dCMP                           &  0.00240                             \\
                   & dGMP                           &  0.00240                             \\
                   & dTMP                           &  0.00360                             \\
                   & AMP                            &  0.04600                             \\
                   & CMP                            &  0.04470                             \\
                   & GMP                            &  0.04600                             \\
                   & UMP                            &  0.05990                             \\ \midrule
      Amino Acid   & Alanine                        &  0.45880                             \\
                   & Arginine                       &  0.16070                             \\
                   & Asparagine                     &  0.10170                             \\
                   & Aspartate                      &  0.29750                             \\
                   & Cysteine                       &  0.00660                             \\
                   & Glutamate                      &  0.30180                             \\
                   & Glutamine                      &  0.10540                             \\
                   & Glycine                        &  0.29040                             \\
                   & Histidine                      &  0.06630                             \\
                   & Isoleucine                     &  0.19270                             \\
                   & Leucine                        &  0.29640                             \\
                   & Lysine                         &  0.28620                             \\
                   & Methionine                     &  0.05070                             \\
                   & Phenylalanine                  &  0.13390                             \\
                   & Proline                        &  0.16470                             \\
                   & Serine                         &  0.18540                             \\
                   & Threonine                      &  0.19140                             \\
                   & Tryptophan                     &  0.02840                             \\
                   & Tyrosine                       &  0.10200                             \\
                   & Valine                         &  0.26460                             \\ \midrule
      Carbohydrate & 1 3 beta D Glucan              &  1.13480                             \\
                   & Mannan                         &  0.80790                             \\
                   & Glycogen                       &  0.51850                             \\
                   & Trehalose                      &  0.02340                             \\ \midrule
      Lipid        & Ergosterol                     &  0.00070                             \\
                   & Triglyceride                   &  0.00007                             \\
                   & Zymosterol                     &  0.00150                             \\
      Phospholipid & Phosphatidyl 1D Myo Inositol   &  0.00005                             \\
                   & Phosphatidate                  &  0.00001                             \\
                   & Phosphatidylcholine            &  0.00006                             \\
                   & Phosphatidylethanolamine       &  0.00005                             \\
                   & Phosphatidylserine             &  0.00002                             \\ \midrule
      Small        & ATP                            & 59.27600                             \\
                   & H$_2$O                         & 59.27600                             \\
                   & Sulphate                       &  0.02000                             \\ \bottomrule
  \end{tabular}
  \end{footnotesize}
  \caption[Biomass requirements \emph{S. cerevisiae} iND750 model]{Biomass requirements of \emph{S. cerevisiae} iND750 model. Each molecule is categorised by type. The requirement of each molecule is millimoles per gram of dry weight biomass. The estimation of biomass quantities is outlined by Feist \emph{et al.} \cite{feist2009}, the \emph{S. cerevisiae} model construction is described by Duarte \emph{et al.} \cite{duarte2004a}.}
  \label{table:biomass_requirements}
\end{table}

\subsection{A novel approach using genome scale models to estimating amino acid cost}\label{section:amino_acid_cost_estimation}

The introduction described how metabolic control analysis can be used to estimate the importance of each reaction in a kinetic model. This section shows how similar principles can be used to estimate the sensitivity of \emph{S. cerevisiae} growth to changes in amino acid requirements. This sensitivity is interpreted as the biosynthetic cost of the amino acid in the nutrient limiting condition it is estimated.

\begin{figure}
  \centering
  \subfloat[Cost estimation in glucose limited conditions]{
    \label{figure:curve_estimation:schematic}
    \includegraphics*[width=6.5cm]{cost_schematic.eps}
  }
  \hfill
  \subfloat[Supply-demand elasticity for tryptophan]{
    \label{figure:curve_estimation:curve}
    \includegraphics*[width=6cm]{curve_estimation.eps}
  }
  \hfill
  \subfloat[Comparison of costs for glycine, histidine, and tryptophan]{
    \label{figure:curve_estimation:costs}
    \includegraphics*[width=7cm]{comparison_of_costs.eps}
  }
  \hfill
  \caption[Method used for estimation of amino acid cost]{Outline of the method used for the estimation of amino acid cost. Figure~\ref{figure:curve_estimation:schematic} is a schematic representation of simulating glucose limited conditions using a genome scale model. The growth rate of the model is fixed at a constant value and the objective function of the simulation minimises the cellular entry of a specific nutrient, which in the example is glucose. All other nutrient entering the model, illustrated here with ammonia, are unbounded. Figure~\ref{figure:curve_estimation:curve} shows the relationship between altered tryptophan requirement and the response in both glucose and ammonium uptake. Figure~\ref{figure:curve_estimation:costs} contrasts the estimated costs for tryptophan, glycine and histidine in glucose and ammonium limiting conditions. }
  \label{figure:curve_estimation}
\end{figure}

Figure~\vref{figure:curve_estimation} illustrates the method used to estimate the amino acid biosynthetic cost using flux balance analysis. Figure~\ref{figure:curve_estimation:schematic} demonstrates the model configuration for the simulation of a glucose limited environment, amino acid costs were also calculated for ammonium and sulphate limiting conditions.

In each nutrient limiting environment the biomass requirement of each amino acid is manipulated around the original value in the range of $\pm$0.0002\%, and each time the model is re-optimised to determine the correspinding effect on the uptake flux of the limiting nutrient.  Figure~\ref{figure:curve_estimation:curve} provides an example of the relationship between changes in tryptophan requirement and the flux of either glucose and ammonium into the cell. 
 
Two types of amino acid costs were estimated from changing biomass requirement. The first is described as a relative amino acid cost and is estimated from small percentage changes in requirement. This relative estimation of amino cost can be rescaled to estimate a cost based on absolute changes in amino acid requirement described as an absolute measure of amino acid cost.

One further consideration in estimating amino acid cost is the growth rate at which the costs were estimated. For instance the absolute cost of tryptophan estimated at 0.2 hr$^{-1}$ is twice the cost of tryptophan estimated at 0.1 hr$^{-1}$. To control for growth rate when estimating amino acid cost, the costs were estimated at a three levels of feasible yeast growth rates 0.1 hr$^{-1}$, 0.2 hr$^{-1}$ and 0.3 hr$^{-1}$. Each amino acid cost is then rescaled to be independent of the growth rate at which it was estimated. The amino acid costs presented in this work are those estimated at the 0.3 hr$^{-1}$ growth rate, where \emph{S. cerevisiae} uses respiratory energy production \cite{famili2003}.

Figure~\ref{figure:curve_estimation:costs} compares the estimated absolute costs for tryptophan, histidine and glycine in glucose and ammonium limitation. Tryptophan which is an expensive amino acid in glucose limitation, is cheaper in nitrogen limitation. In contrast to this histidine which is a cheap amino acid in glucose limitation is more expensive in ammonium limitation. Glycine, a small amino acid with a single hydrogen side chain is cheap in both nutrient limitations.

\subsubsection{Estimated amino acid costs in glucose, ammonium, and sulphate limitation}

Using the above described method relative and absolute amino acid costs were estimated for all twenty amino acids in four simulated nutrient limiting conditions: glucose, ammonium, sulphate, and phosphate. For clarity costs are described as either $R_{nutrient}$ or $A_{nutrient}$ for relative and absolute costs respectively. The relative and absolute estimated costs for each of these amino acids is illustrated in Figure~\vref{figure:costs_dendrogram_dotplot} along with amino acid costs described in the literature. The estimated amino acid cost are provided in Table~\vref{table:estimated_costs} and the correlation between costs is shown in Table~\vref{table:costs_correlation}.

\begin{table}
\begin{footnotesize}
  \begin{tabular}{ p{1cm} *{6}{p{1.9cm}} }
  \toprule
      & $A_{glucose}$ & $A_{nitrogen}$ & $A_{sulphur}$ & $R_{glucose}$ & $R_{nitrogen}$ & $R_{sulphur}$ \\ \midrule
      ala & 0.50 & 1 & 0 & 0.23 & 0.46 & 0.00 \\
      arg & 1.39 & 4 & 0 & 0.22 & 0.64 & 0.00 \\
      asn & 0.79 & 2 & 0 & 0.08 & 0.20 & 0.00 \\
      asp & 0.61 & 1 & 0 & 0.18 & 0.30 & 0.00 \\
      cys & 0.75 & 1 & 1 & 0.00 & 0.01 & 0.01 \\
      gln & 0.92 & 2 & 0 & 0.10 & 0.21 & 0.00 \\
      glu & 0.86 & 1 & 0 & 0.26 & 0.30 & 0.00 \\
      gly & 0.31 & 1 & 0 & 0.09 & 0.29 & 0.00 \\
      his & 1.46 & 3 & 0 & 0.10 & 0.20 & 0.00 \\
      ile & 1.21 & 1 & 0 & 0.23 & 0.19 & 0.00 \\
      leu & 1.21 & 1 & 0 & 0.36 & 0.30 & 0.00 \\
      lys & 1.31 & 2 & 0 & 0.38 & 0.57 & 0.00 \\
      met & 1.25 & 1 & 1 & 0.06 & 0.05 & 0.05 \\
      phe & 1.84 & 1 & 0 & 0.25 & 0.13 & 0.00 \\
      pro & 0.99 & 1 & 0 & 0.16 & 0.16 & 0.00 \\
      ser & 0.49 & 1 & 0 & 0.09 & 0.19 & 0.00 \\
      thr & 0.69 & 1 & 0 & 0.13 & 0.19 & 0.00 \\
      trp & 2.39 & 2 & 0 & 0.07 & 0.06 & 0.00 \\
      tyr & 1.77 & 1 & 0 & 0.18 & 0.10 & 0.00 \\
      val & 0.96 & 1 & 0 & 0.25 & 0.26 & 0.00 \\ \bottomrule
  \end{tabular}
\end{footnotesize}
\caption[\emph{S. cerevisiae} estimated absolute and relative amino acid costs]{The units of the relative costs ($R_{uptake}$) are mmol of uptake nutrient per gram dry weight biomass (mmol$^{-1}$ gDW$^{-1}$). The absolute costs ($A_{uptake}$) are unit less. All costs are estimated in the iND750 \emph{S. cerevisiae} model using the COBRA toolbox and are rounded to two decimal places.} 
\label{table:estimated_costs}
\end{table}

\begin{sidewaysfigure}
\centering
\includegraphics*[angle=0,height=13cm]{costs_dendrogram_dotplot.eps}
\caption[Comparison of amino acid cost estimates]{Amino acid cost estimates are shown as bar charts on the left hand side. Each bar chart axis shows the minimum and maximum value of each cost type. The correlations between costs are compared in a dendrogram on the right hand side computed by complete agglomerative clustering using Spearman's Rank correlation distance between data sets. The cost values are shown in Table~\vref{table:literature_costs} and the correlation between costs are given in Table~\vref{table:costs_correlation}}
\label{figure:costs_dendrogram_dotplot}
\end{sidewaysfigure}

\begin{sidewaystable}
  \centering
  \begin{footnotesize}
    \begin{tabular}{p{2.2cm} *{12}{p{1.2cm}} }
                                                                                \toprule
           & A\&G   & C\&W   & C\&W  & Wager        & Wagner      & Weight & $A_{glu}$ & $R_{glu}$ & $A_{amm}$ & $R_{amm}$ & $A_{sul}$ & $R_{sul}$ \\
           & Energy & Energy & Steps & Ferm. & Resp. &            &               &               &              &                &                &                \\ \midrule
A\&G Energy  & 1.00  & 0.81  & 0.70  & 0.77  & 0.87  & 0.75  & 0.94  & 0.06  & 0.51  & -0.45 & 0.62  & 0.37  \\
C\&W Energy  & 0.81  & 1.00  & 0.62  & 0.45  & 0.81  & 0.44  & 0.73  & 0.07  & 0.34  & -0.40 & 0.36  & 0.17  \\
C\&W Steps   & 0.70  & 0.62  & 1.00  & 0.80  & 0.72  & 0.37  & 0.59  & 0.03  & 0.34  & -0.33 & 0.38  & 0.27  \\
Wagner Ferm. & 0.77  & 0.45  & 0.80  & 1.00  & 0.71  & 0.52  & 0.65  & -0.16 & 0.52  & -0.46 & 0.68  & 0.50  \\
Wagner Resp. & 0.87  & 0.81  & 0.72  & 0.71  & 1.00  & 0.52  & 0.75  & 0.08  & 0.31  & -0.43 & 0.42  & 0.24  \\
Weight       & 0.75  & 0.44  & 0.37  & 0.52  & 0.52  & 1.00  & 0.82  & 0.02  & 0.57  & -0.16 & 0.46  & 0.20  \\
$A_{glu}$    & 0.94  & 0.73  & 0.59  & 0.65  & 0.75  & 0.82  & 1.00  & 0.18  & 0.56  & -0.25 & 0.54  & 0.26  \\
$R_{glu}$    & 0.06  & 0.07  & 0.03  & -0.16 & 0.08  & 0.02  & 0.18  & 1.00  & -0.25 & 0.63  & -0.55 & -0.30 \\
$A_{amm}$    & 0.51  & 0.34  & 0.34  & 0.52  & 0.31  & 0.57  & 0.56  & -0.25 & 1.00  & -0.11 & 0.55  & 0.09  \\
$R_{amm}$    & -0.45 & -0.40 & -0.33 & -0.46 & -0.43 & -0.16 & -0.25 & 0.63  & -0.11 & 1.00  & -0.71 & -0.50 \\
$A_{sul}$    & 0.62  & 0.36  & 0.38  & 0.68  & 0.42  & 0.46  & 0.54  & -0.55 & 0.55  & -0.71 & 1.00  & 0.76  \\
$R_{sul}$    & 0.37  & 0.17  & 0.27  & 0.50  & 0.24  & 0.20  & 0.26  & -0.30 & 0.09  & -0.50 & 0.76  & 1.00  \\ \bottomrule
    \end{tabular}
  \end{footnotesize}
  \caption[Correlations coefficients between amino acid cost estimates]{Spearman's Rank correlation coefficients between estimated and literature described amino acid costs. }
  \label{table:costs_correlation}
\end{sidewaystable}

The $A_{glucose}$ cost is highly correlated with several previous measures of amino acid cost. The energetic cost derived by Akashi and Gojobori \cite{akashi2002} has the highest correlation coefficient $A_{glucose}$ (Spearman R = 0.94), but $A_{glucose}$ is also correlated (Spearman R $>$ 0.7) with Craig and Weber's energetic cost \cite{craig1998}, Wagner's respiratory cost \cite{wagner2005} and molecular weight \cite{seligmann2003}. This indicates that an systems biology approach using flux balance analysis captures the energetic cost of synthesising an amino acid. 

The $A_{ammonium}$ costs show a weak correlation (Spearman R between 0.5 - 0.6) with some of previous described measures of cost: Akashi \& Gojobori, Wagner fermentative, and molecular weight. It should be noted that the $A_{ammonium}$ and $A_{sulphate}$ costs are proportional to that nutrient content of the amino acid.

The $R_{glucose}$ and $R_{ammonium}$ costs show no correlation with previous measures of amino acid cost. The highest correlation is with Wagner fermentative growth (Spearman R = -0.16). The $R_{glucose}$ and $R_{ammonium}$ costs are however correlated with each other (Spearman R = 0.63). 

Summarising both the relative and absolute cost types, the absolute costs reflect the per molecule biosynthetic cost, while the relative costs reflect the absolute cost scaled by the quantity of the amino acid in biomass. For example, the $A_{sulphate}$ costs of methionine and cysteine are both 1, which reflects the single sulphur atom in each amino acid. The $R_{sulphate}$ cost for methionine is however much greater than that of cysteine, as methionine is used proportionally more than cysteine.

\subsubsection{Comparison of amino acid costs across species in \emph{S. cerevisiae} and \emph{E. coli}}

The relative ease at which amino acid costs can be calculated for species where there is a genome scale model, compared with previous manual curation of cost, allowed comparison of estimated amino acid costs in \emph{S. cerevisiae} and \emph{E. coli}. Figure~\vref{figure:amino_acid_costs_scatterplot} compares both $R_{glucose}$ and $A_{glucose}$ costs for \emph{E. coli} and \emph{S. cerevisiae} with two costs described in the literature: molecular weight \cite{seligmann2003} and number of ATP and NADH molecules \cite{akashi2002}.

\begin{figure}
\centering
\includegraphics*[width=13cm]{amino_acid_costs_scatterplot.eps}
\caption[Comparison of the genome scale model derived cost data sets.]{Comparison of estimated amino acid cost with number of ATP and NADPH molecules used in synthesis (left), and molecular weight (right). On the y axis are the amino acid costs estimated using flux balance analysis. Both \emph{S. cerevisiae} and \emph{E. coli} measures are included to illustrate correlation of cost estimates between species. Estimated cost values have been been rescaled around their mean value to allow comparisons across species. The trends in each plot are drawn using `loess' smoothing.}
\label{figure:amino_acid_costs_scatterplot}
\end{figure}

Figure~\ref{figure:amino_acid_costs_scatterplot} shows that the both the absolute costs show a small amount of variance in their correlation with the example literature described cost measures. Both \emph{E. coli} and \emph{S. cerevisiae} absolute costs measures are particularly well correlated with the Akashi and Gojobori described measure of cost \cite{akashi2002}. Comparing the variation between species the \emph{E. coli} $A_{glucose}$ cost is better correlated with this measure than the \emph{S. cerevisiae } equivalent (Spearman R = 0.99 vs. 0.94). This may be expected given the Akashi and Gojobori measures of cost were estimated for \emph{E. coli}. In comparison molecular weight shows a linear relationship with the absolute measures of costs, but shows a greater degree of variation with the costs from both species. Overall the species difference between the two $A_{glucose}$ cost measures is marginal and they are highly correlated (Spearman R = 0.94).

The comparison of relative measures of cost illustrates the small degree of correlation with previous estimates of cost. For the both molecular weight and ATP+NADH cost estimates, neither \emph{E. coli} or \emph{S. cerevisiae} relative cost estimates show any relationship, and furthermore the figure illustrates the disparity in relative costs between species (Spearman R = 0.74) relative to that of the absolute costs.

\subsection{A novel approach to estimating gene cost}

%TODO: Why do this analysis?

\subsubsection{Estimated reaction constraint in glucose and ammonium limitation}

The above described 262 single-gene encoded reactions was assessed in various solutions spaces. The simulations used were glucose and ammonia nutrient limitations in using either a respiratory or fermentative growth model. All simulations were performed at a fixed growth rate of 0.3 hr$^{-1}$. The use of each gene in the resulting model optimised solution space can be divided in four types based on the flux distribution in the model. Figure~\vref{figure:reaction_constraints} illustrates these types.

\begin{figure}
\centering
\includegraphics*[width=10cm]{reaction_constraints.eps}
\caption[Gene use in model optimised solution space]{The use of 1:1 gene associated reactions in four environmental solution spaces. Two types of nutrient limitation are shown: glucose and ammonia. Each limitation is explored using either a yeast respiratory, or fermentative growth. Four types of reaction uses are described: at maximum, constrained, variable, or zero flux. Maximum reactions are at the positive or negative limit of the allowable flux for the model optimisation. Constrained reactions are fixed at the current flux and absolute flux cannot be reduced to satisfy the model growth rate. Variable reactions' flux rate can be reduced whilst still maintaining the model growth rate. Zero flux reactions are not used in the given solution. All reaction estimates are performed at a fixed growth rate of 0.3 hr$^{-1}$. }
\label{figure:reaction_constraints}
\end{figure}

\paragraph{Reaction rate at maximum}

The reactions are at the allowable limit for any reaction in the solution. The flux through the reaction the reaction is either at $\pm$1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$. The number of these types of reactions was limited where only six total exist across all the model solutions.

\paragraph{Constrained reaction rate}

These reactions are constrained in their use in the given solution. The reaction is used in the solution, but the absolute reaction rate cannot be reduced whilst still maintaining maintaining a viable solution for a model growth rate at 0.3 hr$^{-1}$ The number of these reactions ranged from 58 to 63 reactions.

\paragraph{Variable reaction rate}

These reactions are used in the solution, and the absolute reaction flux can be reduced whilst still maintaining the fixed growth rate. The number of these reactions ranged from 30 to 42.

\paragraph{Reaction unused}

This type of reaction was not used in the solution and had a reaction rate of zero. This is the largest proportion of reactions across all simulations ranging from 163 to 166.

\clearpage

\section{Discussion}

This chapter illustrated the methods used to estimate amino acid cost, and further showed the correspondence of these amino acid costs with previously described measures of amino acid cost.

\subsection{Flux balance analysis predicts amino acid cost}

The absolute amino acid costs estimated using the genome scale model are as might expected. The $A_{ammonium}$ and $A_{sulphate}$ costs are directly proportional to the nitrogen and sulphur content of the molecule. This indicates, when either of these two nutrients are limiting the cost of the amino acid is the material quantity of the limiting nutrient. The estimated $A_{glucose}$ cost is different as glucose is the source of both carbon atoms and energy to the cell. When compared with literature reported cost measures the $A_{glucose}$ cost is most correlated with the Akashi and Gojobori energetic cost. For both these measures of amino acid cost, as is also shown in the literature, small less complex amino acids such as glycine and alanine are cheaper while larger more complex amino acids such as tryptophan and tyrosine are biosynthetically more expensive.

The relative amino acid cost estimations represent the absolute cost of an amino acid scaled by the quantity measured in \emph{S. cerevisiae} biomass. The relative measures of amino acid cost show little correlation with previous measures of amino acid cost. Furthermore amino acids that are expensive based on their absolute cost, are cheaper in terms of relative cost. For example tryptophan has a high $A_{glucose}$ cost, but is instead much cheaper when the $R_{glucose}$ cost is considered. This reflects that expensive amino acids may be minimised in biomass composition. On the other hand lysine has one of the most expensive $R_{glucose}$ costs, but is relative inexpensive in $A_{glucose}$. This means that a proportional increase in the requirement of lysine would have a greater effect than of tryptophan, since lysine appears at greater quantities in biomass.

\subsubsection{Correlation $R_{glucose}$ and $R_{ammonium}$ indicates genome optimisation}

An interesting point is that the $R_{glucose}$ and $R_{ammonium}$ costs have a high correlation (Spearman R = 0.63). This leads to the possibility that optimisation of energy/carbon based costs in the proteome also results in optimisation of the nitrogen cost. The benefits of this are intuitive where selective advantages for the optimisation of a given amino acid in carbon limitation will have advantages for nitrogen limitation. If there was little or negative correlation between carbon and nitrogen based costs, then optimisation of a protein for carbon limitation, could result in a non-optimisation of the protein under nitrogen limitation.

\subsubsection{Atomic material content as a biosynthetic cost}

Amino acid synthesis will require energy to power each step in the metabolic network. In addition to these costs the atomic material content of the amino acid structure may also be considered a cost of synthesis. Mazel \& Marliere \cite{mazel1989} showed that \emph{Calothrix} proteins expressed in sulphur limiting conditions are depleted for the sulphur containing amino acids cysteine and methionine. This work indicating the atomic content of an amino acid should also be taken into consideration when attempting to define a measure of cost. Work by both Baudoin-Cornu \emph{et al.} \cite{baudoin2001} and Bragg \emph{et al.} \cite{bragg2006} used atomic content as a measure of amino acid cost when examining protein content, when considering the evolution role of an amino acid cost, the role of energy should also be coupled with the availability of nutrients. Any limitation on a specific nutrient may play as an important role as that of energetic limitation.

\subsection{Predicting the cost of a gene}

The number of enzyme parameter needed to produce a kinetic model mean that the number of reactions included can be a magnitude smaller than that of genome scale models. The small number of reactions therefore means that the estimation of control coefficients are limited to the small number of available reactions. For instance, at the time of writing this thesis, there exists no model where it would be possible to derive control coefficients for reactions across different parts of the cell metabolism such as both energy metabolism and amino acid. Existing genome scale models contain reactions contain the majority of the central metabolism for the organism they describe, which solves the problem of limited model size. The restriction of genome scale models is that they do not contain enzyme kinetic data, and so it is not possible to estimate metabolic control coefficients in the same way as kinetic models.

\subsubsection{Determining associations between genes and encoded metabolic reactions}

Estimating the cost of a gene on metabolism first requires determining the points of effect the gene has on the metabolic network. This analysis aimed to first determine which reactions have a single effect on metabolism, and thereby allowing a simpler analysis of their cost. The focus on 1:1 associations of gene to reactions excluded 488 genes but did allow a much simpler analysis of gene cost, where any effect on the gene may be expected to have a direct effect on single reaction which can then be analysed in through the effect on the corresponding phenotype.

Alternative approaches used in gene knockout studies have treated the reaction as unaffected if an associated paralog is still available, or have removed the reaction if the one unit of a multimeric enzyme is deleted \cite{pal2006}. In this analysis reactions are not being removed, but instead their effect on the model objective function estimated. There while it can be assumed that the removal of a enzyme subunit will render the complete enzyme ineffective, the same cannot by said by of reducing the effect of a single gene product within an multimeric enzyme. For considering paralogs catalysing the same reaction it could be assumed that a deficiency in one paralog will not affect the reaction rate, this may not necessarily be the case as some paralogs may only expressed under certain conditions \cite{carlson2007,perez2008}.

%TODO: Any other knockout studies can be cited?

%TODO: How did other knockout studies treat paralogs?

\subsubsection{Different constraints on reaction flux in glucose and ammonium limitation}

The reactions at maximum flux represent edges in solution space, where any reaction cannot exceed an absolute flux of 1000 mmol$^{-1}$ gDW$^{-1}$ hr$^{-1}$. If the artificial constraint of reaction flux is increased the likely outcome is that the flux of these reactions would also increase, and the flux through the reactions is of an artificial limit imposed by the model optimisation, rather than because of the simulation of a biological phenotype. Therefore these reactions are not considered further in the analysis. The number of reactions at the limit is low, six across all conditions, and removal from the gene cost data set is not a significant impact on the size.

The constrained reactions represent those whose flux is constrained in the solution space. Forced reduction of the flux through any of these reactions results in a model will fail find an optimal solution. As the flux cannot be reduced, these reactions all represent inviable knockout mutants where removal from the model will prevent the model being optimised. These reactions may represent singleton pathways, the only available reactions for producing a required nutrient.

The variable reactions whose flux can be reduced and still maintain a viable model solution. Reasons that these fluxes can be reduced may be because optimisation finds an alternate pathway for the production of a required nutrient. Whether any of these reactions can be completely deleted from the model is not clear, only that the model is viable if the reaction flux is reduced.

The zero flux reactions are unused in the solution space and represent the majority of reactions. These reactions could be removed from the model without any effect on the model solution. These reactions could represent environment or nutrient specific reactions, required only when a specific nutrient is present.

