\section*{Summary}

Summary of the chapter...

\clearpage

\section{Introduction}

\subsection{Metabolic cost as a selection pressure}

The amino acid composition of a predicted proteome varies across species, and is related to GC content and environemnt \cite{tekaia2006}. The selective pressures influencing the use of amino acids in individual protein sequence are reviewed by Pal \emph{et al.} \cite{pal2006} and are illustrated in Table~\vref{table:protein_selective_pressures}.

The premise of the cost selection hypothesis \cite{akashi2002} is that if two amino acids can perform a similar function at a position in a protein sequence, selection prefers the biosnthetically cheaper amino acid. The difference in biosynthetic materials is diverted to other processes. The correctness of this assumption of cost as a selective pressure is still unclear, though two studies suggests that cost selection plays only a minor role in protein evolution \cite{rocha2004} or expression \cite{raiford2008}.

\begin{table}
  \centering
  \begin{tabular}{ l l l }
  \toprule
  Selection & Name                   & Description \\ \midrule
  Purifying & Fitness density        & Text \\
            & Dispensibility         & Text \\
            & Structure              & Text \\
            & Network position       & Text \\
            & Developmental position & Text \\
            & Expression             & Text \\
  Positive  & Competitive            & Text \\
            & Compensatory           & Text \\
            & Adaptive               & Text \\ \bottomrule
  \end{tabular}
  \caption[Factors in protein evolution]{Long caption}
  \label{table:protein_selective_pressures}
\end{table}

\subsection{Genome composition}

The genome composition of an organism may reflect trends for preferential amino acid usage. The cyanobacterium emph{Calothrix} sp. 7601 encodes sulphur eradicated versions of proteins abundant in sulphur limiting conditions, where the only sulphur containing residues are the start methionine and a single cysteine binding site \cite{mazel1989}. In \emph{Eschericia coli} and \emph{Saccharomyces cerevisiae} proteins in pathways responsible for recruiting carbon and sulpur from the environment are depleted for the same nutrient \cite{baudoin2001}. The advantage of depleting the enzymes of for the nutrient is that the protein may be easier expressed for in conditons limiting for the nutrient. Both of these studies indicate, in a small subset of the total proteome, the use of amino acids with specific atoms may come at a cost when availability of a specific atom is restricted.

The trend for specific amino acid use is considered using a larger dataset acorss 141 predicted proteomes derived from available genome sequences \cite{bragg2006}. The difference in mean sulphur content between othologs was correlated with mean sulphur content in the proteome, suggesting a selection pressure for reducing sulphur content. Thermophiles have a lower sulphur content than non-thermophiles, the synthesis and maintainence of methionine and cysteine residues is the likely factor in the utilisation of these residues. Sulphur containing amino acids also followed specific trends related to GC content, though the usage of all amino acids is associated with GC content \cite{tekaia2006}.

Analysis of large numbers of complete genomes was also used to determine whether the molecular weight of amino acids is minimised \cite{seligmann2002}. Across a large set of genomes the use of heavier amino acids is minimised. Furthermore cost minimisation is a greater factor for organisms synthesising amino acids, than for organisms absorbing amino acids from the environment.

\subsection{Gene expression}

The encoding of amino acids is expressed in the proteome, and therefore cost trends should be related to gene expression. A large proportion of cellular energy is spent of protein synthesis, and the cost minimisation theory predicts that minimising the use of expensive amino acids provides a fitness advantage. The highly expressed genes may be under particular pressure to minimise cost since the cost of synthesis is multiplied by the number of protein copies. The selection pressure for cost minimisation should also therefore decrease with expression level.

The selection pressure for cost minimisation can be estimated from the cost of the amino acid residues encoded in highly expressed genes. Akashi and Gojobori \cite{akashi2002} used Craig and Weber's amino acid cost estimates \cite{craig1998} to determine whether the strength of the trend between expression level and cost. Comparison of per residue eneretic cost with major codon usage predicted gene expression \cite{kanaya1999} indicated protein cost does decrease with level of expression.

Akashi and Gojobori performed their analysis in two mesophiles: \emph{E. coli} and \emph{B. subtillis}. The importance of cost minimisation in gene expression may be related to organism lifestyle, where species with large differences in metabolism may have different measures of amino cost and selection pressures to minimise these costs. Heizer \emph{et al.} \cite{heizer2006} examined cost related trends in an additional four organims including chemoheterophic, photoautotrophic, and thermophilic lifestyles. Across all included species the trend between predicted gene epxression and biosythetic cost indicated a pressure to minimise costly amino acids at high expression levels.

TODO: Is mesophile the correct term?

TODO: Include descriptions of the meaning of each life style

Changes in the expression level of a gene may also expected to have an effect on the use of cellular energy. An increase in gene expression will require the greater energy diversion to the synthesis of the required amino acids, while a decrease will free cellular energy for other processes. Wagner \cite{wager2005} examined whether gene duplication events or mutations in regulatory regions are also constrained by a cost selection pressure. The resulting cost increase form a doubling of gene expression from a duplication event may be expected to be greater than the negative selection pressure for the \emph{S. cerevisiae} effective population size. This indicates that cost is a factor in the retention or loss of gene duplicates.

The previous described studies examined the cost minimisation in gene expression using indicators of expression based on codon usage. Validation of the extent of cost minimisation must examine \emph{in vivo} expression. Raiford \emph{et al.} \cite{raiford2008} compared the trend to minimise the use of expensive amino acids with \emph{S. cerevisiae} codon adaptation, transcript levels, and protein levels. This analysis correlated the aerobic and anerobic costs with the usage of each amino acids in expression. The results of the study showed a trend for cost minimisation, but the trend varies according to amino acid class, and explains only a limited degree of variation in gene expression.

Such results indicate the selective for cost minimisation may have only a small impact on the evolution of proteins. Bragg and Wagner \cite{bragg2007} compared the carbon content of upregulated genes versus the rest of the proteome, and found no indication of cost minimisation in the upregulated set. Futhermore a yeast strain evolved in carbon limitation was found to use more carbon atoms per residue than the unadapted strain.

The current literature describes a mixed view of the strength of cost minimisation as a selective pressure in protein evolution. Analysis of the genome data, comparing predicted expression level with predicted per residue biosynthetic cost indicates that cost minimisation may be a strong selective force. Cost minimisation however appears to be a much weaker force when comparing actual expression data. Analysis of enzymes protein sequences, and simulation of various nutrient limiting conditions indicates cheaper metabolically inefficient enzymes are maintained for expression in nutrient scarcity. Biosynthetically more expensive efficient copies may then be used when nutrients become more available \cite{carlson2007}.

\subsection{Protein sequence evolution}

If there is a selective force for the optimisation of protein sequence for cheaper residues, then biosynthetic cost should be a factor in amino acid subsitution rates. Whether this is true or not is still unclear and remains to be investigated. The literature described so far has examined either genome compostion or gene expression for trends in cost minimisation; the existence of such indicates that cost minimisation is a selective pressure.

Craig and Weber \cite{craig1998} found that replacement amino acids in a small set of \emph{E. coli} genes were cheaper than the mean or median amino acid cost. Amino acid cost in another examined set of \emph{E. coli} proteins were found to the same as the amino acid cost, presenting a mixed view of the importance of cost minimisation.

Metabolic cost was compared directly in a multivariate regression of possible \emph{E. coli} and \emph{B. subtillis} protein evolution determinants. Of all the protein characteristics included: expression level, functional catergory, essentiality, and metabolic cost (Akashi and Gojobori energetic cost\cite{akashi2002}); cost minimisation is a weak predictor of protein evolution \cite{rocha2004}.

Metabolic cost is often interpreted as the energy required for syntheis or the use of specific atoms such as nitrogen or sulphur. But what if the costs of amino acid selection is more complex than existing current estimates of energy or atomic content? Amino acid composition analysis of 208 encoded proteomes indicates trends for amino usage associated with organism life style \cite{tekaia}. Comparison of paralogs between thermophile and non-thermophile related species, controlling for protein structure and function, highlights amino acid preferences dependent on the organism's lifestyle. One possible reason, as Swire discusses, is hypothermophiles are subject to different energetic constraints given the tempurate of the environment that may affect Gibb's free energy of metabolic reactions \cite{swire2006}.

Sequencing samples of environments provides varing genome coverage of species living in the same environment. Reconstruction of the metabolic pathways can be used to determine how pathways vary between environment. Amino acid biosynthetic pathways vary between oceanic environments, and the variance is unrelated to biosynthetic cost. Instead the usage of a amino acid synthetic pathway, in particular methionine, may instead depend on the availability of the cofactors in the synthesis of the amino acid. A negative correlation between the number of synthetic pathways and number transporters of the same acid indicate a trade-off between synthesis and import \cite{gianoulis2009}. The metabolic use of cofactors or the import of amino acids in the environment are not readily apparent from the metabolic map of an organism, but may have an intimate effect on the usage of an amino acids not explained by energetic or atomic cost.

\subsection{Summary of results}

\clearpage

\section{Materials and Methods}

\subsubsection{Protein characteristics}

\emph{S. cerevisiae} codon adaptation index (CAI) data was produced by Wall \emph{et al.} \cite{wall2005}. Genome encoded tRNA count data was produced by Akashi \cite{akashi_trna}. The $A_{glucose}$ and $R_{glucose}$ costs used were those described in the previous chapter. The other amino acid costs used are reported in the literature by Craig \& Weber \cite{craig1998}, Akashi \& Gojobori \cite{akashi2002}, Wagner \cite{wagner2005} and Seligmann \cite{seligmann2004}. Atomic content for each amino acid was taken according to universally available descriptions.

\subsubsection{Gene expression experimental data}

The transcript and metabolomic data used in this analysis was taken from Castrillo \emph{et al.} \cite{castrillo2007}. The data was measured from \emph{S. cerevisiae} cultured continuously in a chemostat. Each measurement was taken at one of three dilution rates 0.1h$^-1$, 0.2h$^-1$ and 0.3h$^-1$ and four different nutrient limiting conditions, glucose, ammonium, sulphate and phosphate. The transcript data was produced from replicate microarray measurements of total RNA. Following measurement the RNA data was processed using with RMA quantile normalisation \cite{RMA}. The metabolite data was measured using Gas chromatography followed by time of flight mass spectrometry (GC/TOF-MS). The metabolite data was normalised using median absolute deviation. Missing metabolite data were inferred from replicates in the same conditions.

The protein data used were produced by Ghaemmaghami \emph{et al.} \cite{ghaemmaghami2003}. These data were derived from tandem affinity purification of TAP-tagged \emph{S.cerevisiae} ORFs. Each tagged ORF was measured using antibody quantification of the encoded tag. Absolute levels of protein per cell were estimated by comparing the \emph{S.cerevisiae} gene measurements with a purified \emph{E. coli} INFA-TAP construct scale.

\subsubsection{Gene expression multivariate regression}

Each variable used in the regression transformed by the natural logarithm. The exception to this was the transcript data which was logged during the original processing of the data. The sulphur content of individual amino acids contained zero values, therefore a small value (0.0001) was added so the logarithm could be taken. All datasets were scaled to have the same mean and variance. The aim of this data processing was to minimise heteroscedasticity or over-variation in any of the data influencing regression fitting. The metabolite data was averaged for each set of experimental parameters to prevent pseudo-replication resulting from the inference of missing values.

The multivariate regression was performed using the R language for statistical computing \cite{R}. For each dataset, transcript, protein or metabolite, the measured level of the entity in the cell was the response variable in the regression equation. The explanatory variables in the regression equation were carbon, nitrogen, and sulphur content, amino acid cost, codon adaptation index. Environment and dilution rate were also included for the transcript and metabolite data.

For each dataset the complete model equation was fitted including all factors and interactions between factors. The R function \emph{lm} was used as the regression function. Stepwise interaction/variable removable was then performed where interactions were removed from the compete model, to produce a reduced model. The reduced model was compare with the previous complete model, and if the reduced model was considered more parsimonious based on the AIC difference, then the reduced model was then used as the current model. This process was repeated until no more interactions could be removed.

The importance of each explanatory variable in the regression equation was assessed by by repeating the above process but with one of the explanatory variables removed. The regression model with the variable removed was then compared to the complete regression model with all factors included. The AIC difference between the two equations was used as a measure of importance for the removed variable. This process was repeated, where each of the cost types was cycled between those in the introduction.

\subsection{Amino acid cost in \emph{Saccharomyces} proteomes}

The verified open reading frames from seven \emph{Saccharomyces} related species were downloaded from the \emph{Saccharomyces} genome database. The species were \emph{paradoxus}, \emph{bayanus}, \emph{castellii}, \emph{cerevisiae}, \emph{kluyveri}, \emph{kudriavzevii}, and \emph{mikatae}. The genomes of \emph{paradoxus} and \emph{bayanus} were sequenced by MIT \emph{et al.} \cite{mit_paper}. The \emph{castellii}, \emph{kluyveri}, \emph{kudriavzevii}, and \emph{mikatae} genomes were sequenced by WASHU \emph{et al.} \cite{washu}.

The percentage amino acid content of each species was calculated by summing the frequency of each amino acid over the total predicted proteome and dividing by the total number of amino acids. Stop codons and ambiguous amino acid definitions were excluded from the analysis. Only the twenty standard amino acids were counted. Open reading frames containing an internal stop codon were also excluded from the analysis.

The variation in amino acid usage between species was calculated using median absolute deviation (MAD). MAD is a more robust measure of variance for non-normally distributed, or small datasets. MAD is calcualted similarly to standard deviation, except the median absolution deviations from the median is calculated.

Hierchical clustering of data was calculated using the complete linkage agglomerative clustering. Non-parametric Spearman's rank correlation was used to calculate the distance between either species or amino acids, based on percentage amino acid usage in each species. The correlation between variables was coverted to a distance matrix using the normal square (Euclidean) distance between two correlation estimates.

\clearpage

\section{Results}

\subsection{Amino acid cost in genome composition}

\subsection{Amino acid cost in gene expression}

Amino acid cost in gene expression was estimated using multivariate regression on transcriptomic, proteomic and metabolomic datasets. The aim of this analysis was to estimate the importance of amino acid in the protein production, of which each of these datasets are related.

The data in this analysis was the taken from a large scale systems biology analysis in \emph{S.cerevisiae} \cite{castrillo2007}. This dataset contained yest proteomic, transcriptomic and metabolomic data from four nutrient limiting conditions carbon, nitrogen, sulphur and phosphorus. The aim of this analysis was to determine the effects of each of the nutrient limitations on the yeast gene expression.

\subsubsection{Multivariate Regression}

Multivariate regression was used to estimate the importance of amino acid cost in gene expression. For the regression the response variable was the measured level of the data in the original analysis, this being either transcripts, proteins or free amino acids in the metabolome. The explanatory variables were as follows: mean amino acid carbon, nitrogen and sulphur content for the protein, codon adaptation index of the encoding transcript, average genome encoded tRNA number for each codon in the transcript, and average amino acid cost.

Average carbon, nitrogen and sulphur content represents the total content for each atom in the protein divided by the length of the protein excluding stop codons. For the transcriptomic data this represents the atomic content of the encoded protein. In the free amino acid data the atomic content represents the content for each individual amino acid. The codon adaptation index (CAI) represents the optimisation of the transcript for translation, where the use of certain codons are preferential for translation \cite{ikemura1982}. For the protein data the CAI of the encoding transcript was used. CAI was not applicable for the free amino acid data. The tRNA count for a transcript represents the total number of genome encoded tRNAs for each codon divided by the length of the transcript excluding stop codons. For the protein data the average tRNA count of the encoding transcript was used. For the free amino acid data, the total number of tRNAs was used. The amino acid cost variable used was one of those described in the previous chapter. The multivariate regression was repeated where each amino acid cost set was cycled as the variable used in the analysis. The costs used were as follows: Akashi \& Gojobori energy \cite{akashi2002}, Craig \& Weber energetic cost, and biosynthetic steps \cite{craig1998}, Wager fermentative and respiratory energetic cost \cite{wagner2005}, amino acid molecular weight (Da) as used by Seligmann \cite{seligmann2004}, and the $A_{glucose}$ and $R_{glucose}$ costs derived in Section\vref{section:amino_acid_cost_estimation}. The aim of cycling the costs was determine which cost has the most explanatory power for explaining variation in cell physiological data.

The importance of each variable in the multivariate regression was analysed by removing the variable then comparing the reduced model with the model containing all the variables. Akaiki's information criterion (AIC) \cite{akaike1974} was used to the compare the explanatory power of the two models. The AIC is a log likelihood score for the regression, which penalises models with more parameters. This allows models with differing numbers of parameters to be compared. For each regression, a complete model was fitted containing all interaction terms, stepwise automated variable removal was then used to reduce interactions and variables based on the AIC score. A negative AIC score indicates the reduced model should be used as it is more parsimonious. A positive score indicates the larger model provides a better fit to the data. The results of this analysis for each dataset are shown in Figure~\vref{figure:regression_analysis} and in Table~\vref{table:regression_analysis}.

\begin{figure}
  \centering
  \includegraphics*[height=18cm]{regression_analysis.eps}
  \caption[Transcript, proteomic and metabolomic multivariate regression results.]{Transcript, proteomic and metabolomic multivariate regression results. The y-axis indicates the variable removed from the multivariate regression. The x-axis indicates the log$_{10}$ AIC difference between the complete model, and the model with all variables included. The legend indicates which amino acid cost type was used to calculate the per-residue cost. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{figure:regression_analysis}
\end{figure}

\begin{table}
  \centering
  \begin{footnotesize}
    \begin{tabular}{ l *{4}{ p{2cm}} }
      \toprule
      Cost type                          & Transcripts  & Proteins & Amino Acids \\ \midrule
      \emph{S. cerevisiae} $A_{glucose}$ & 0.389        & 0.406    & 0.782       \\
      \emph{S. cerevisiae} $R_{glucose}$ & 0.383        & 0.408    & 0.875       \\
      Akashi \& Gojobori (2002)          & 0.398        & 0.405    & 0.805       \\
      Craig \& Weber (1998) Energy       & 0.416        & 0.40     & 0.835       \\
      Craig \& Weber (1998) Steps        & 0.375        & 0.404    & 0.866       \\
      Wagner (2005) Respiratory          & 0.382        & 0.405    & 0.822       \\
      Wagner (2005) Fermentative         & 0.377        & 0.406    & 0.851       \\
      Molecular Weight                   & 0.422        & 0.405    & 0.767       \\ \bottomrule
    \end{tabular}
  \end{footnotesize}
  \caption[Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets.]{Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets. Each value in the table is the $R^2$ for the complete multivariate regression model using the indicated cost type, and including the carbon, nitrogen, sulphur content of the amino acid, average tRNA count, and codon adaptation index for the transcript and protein data. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{table:regression_analysis}
\end{table}

\paragraph{Transcriptome and proteome analysis}

The regression analysis of the transcript data from Castrillo \emph{et al.} \cite{castrillo2007} and the proteome data from \cite{ghaemmaghami2003} yielded similar results. In both these data codon adaptation index was the strongest predictor of measure levels of either transcript or protein. Figure~\ref{figure:regression_analysis} shows the CAI of the transcript was approximately half a magnitude greater as a predictor of transcription than the other included factors. In the protein data CAI was a full magnitude greater as a predictor of protein production. Together these results indicate the optimisation of a transcript for translation is a strong predictor of both transcript and protein levels.

Comparing other factors in the regression, amino acid cost, carbon content and nitrogen appear to show a similar level of importance in explaining variation in both transcript and protein data. The variables for tRNA count and and sulphur content appear slightly less important. Overall all though the regression analysis indicates that the majority of the variation in the transcript and protein data is explained by codon adaption index of the transcript, as opposed to the variables of costs, carbon, nitrogen, sulphur content, or corresponding genome encoded tRNAs.

As the first two columns of Table~\vref{table:regression_analysis} show, each of the different amino acid cost types had little effect on the $R^2$ of the multivariate regression of transcript and protein data. The difference in which cost type used in the transcript analysis was 4.7\%, ranging from 37.5\% for the regression using Craig \& Weber's step cost measures \cite{craig1998}, to 42.2\% for the regression using molecular weight as suggested Seligmann \cite{seligmann2004}. In the analysis of the protein data, the difference in variation explained dependent of the cost measure was even smaller 0.8\% with all cost measures having approximately the same effect in the multivariate regression of the proteome data.

\paragraph{Metabolome analysis}

In the regression of the metabolome data from Castrillo \emph{et al.} \cite{castrillo2007} codon adaptation index is not applicable and therefore the regression is based on genome encoded tRNAs, estimated cost and carbon, nitrogen and sulphur content of the amino acid. The third plot of Figure~\ref{figure:regression_analysis} illustrates that in the majority of each regression each of these factors appears to contribute equally to model fitting of free amino acid data in yeast. The exception to this is sulphur content which in half the regressions explains less regression than the other factors.

Compared with the multivariate regression of the transcript and protein data, the multivariate regression of the metabolite regression of the free amino acid data was able to explain a much larger degree of variation. The variation explained differed by 10.8\% between the best and worst regression model. The regression that explained the least used molecular weight for amino acid cost, while the best fit used $R_{glucose}$ explaining 87.5\%. This result indicates that a large degree of the variation in amino acid levels can be explained based on their cost, atomic content, and the corresponding tRNAs encoded in the genome.

\subsection{Amino acid cost in protein evolution}

\begin{table}
  \centering
  \begin{tabular}{ l r }
                                                  \toprule
    \emph{Saccharomyces} species & Amino acids \\ \midrule
    \emph{bayanus}               & 2921220     \\
    \emph{castellii}             & 2328652     \\
    \emph{cerevisiae}            & 2916055     \\
    \emph{kluyveri}              & 1248804     \\
    \emph{kudriavzevii}          & 1546252     \\
    \emph{mikatae}               & 1172495     \\
    \emph{paradoxus}             & 2933095     \\ \bottomrule
  \end{tabular}
  \caption{Total number of amino acids for each \emph{Saccharomyces} species.}
  \label{table:total_amino_acids}
\end{table}


\begin{figure}
  \centering
  \includegraphics*[height=10cm]{amino_acid_usage.eps}
  \caption[Comparison of amino acid usage with molecular weight]{Comparison of amino acid usage with molecular weight. Each point is the median percentage amino acid usage across 6 \emph{Saccharomyces} species: \emph{cerevisiae}, \emph{paradoxus}, \emph{mikatae}, \emph{kudriavzevii}, \emph{bayanus}, \emph{castelli}, and \emph{kluyveri}. The grey line indicates the range of values across species.}
  \label{figure:amino_acid_usage}
\end{figure}

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{amino_acid_variance.eps}
  \caption[Comparison of deviation in amino acid usage with molecular weight]{Comparison of deviation in amino acid usage with molecular weight. Each point is the median absolute deviation in percentage amino acid usage across 6 \emph{Saccharomyces} species: \emph{cerevisiae}, \emph{paradoxus}, \emph{mikatae}, \emph{kudriavzevii}, \emph{bayanus}, \emph{castelli}, and \emph{kluyveri}. Loess smoothing is used to indicate trend.}
  \label{figure:amino_acid_variance}
\end{figure}

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{heatmap.eps}
  \caption[Amino acid usage heatmap]{Amino acid usage heatmap. Each block is the percentage amino acid usage for the species. Darker grey indicates greater usage. Each column is scaled by the column mean. The dendrogram highlights the correlation between amino acid usage. Correlation is performed using Spearman's Rank.}
  \label{figure:phylogeny}
\end{figure}

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{phylogeny.eps}
  \caption[Hierarchical clustering of species amino acid usage]{Hierarchical clustering of percentage amino acid usage between \emph{Saccharomyces} species. The distance measure is Spearman's Rank correlation using percentage amino acid compostion. Hierarchical clustering is performed using the complete linkage algorithm.}
  \label{figure:phylogeny}
\end{figure}

\clearpage

\section{Discussion}

\subsection{A small factor in protein composition}

TODO: Metagenomics paper, variance in amino acid compostion not related to cost. Instead may related to limitation of sparse cofactors related to synthesis.

TODO: Regulation vs cost minimisation. Peligbacter ublique has minimimal genome. Streamlined for minimal replication costs? Is cost minimisation become a greater adaptive advantage for organisms living in less varied environments? Otherwise adaptation to mulitple environments outweighs any specific advantage of cost. Attenuate cost across a broad range of environments as much as possible. Or just not important at all?

\subsection{A small factor in yeast gene expression}

\subsubsection{Predictors of transcript and protein levels}

The multivariate analysis indicated that amino acid cost is a small factor explanatory factor in the analysed \emph{S. cerevisiae} transcript and protein data. Regardless of which cost type was used in the regression the CAI of the encoding transcript was the most important variable for predicting transcript and protein levels. The importance of codon adaptation for predicting transcript levels is not a new discovery, but this analysis compares codon adaptation with other predictors of transcript and protein levels to show that codon adaptation is a much greater predictor of transcript and protein levels than factors such as amino acid costs \cite{akashi2002}, atomic content \cite{atomic_content}, or tRNA levels \cite{akashi_trna}.

After codon adaptation is considered the other factors in the regression all appear to play approximately a similar role in explaining variation in protein and transcript expression. In general across each cost regression model of the transcript data cost carbon and nitrogen appear more important than tRNA or sulphur content. There is more variation in the protein data, but a similar trend across these variables is apparent. In some cases the removal of tRNA or sulphur content, and even carbon content in one instance produces a more parsimonious regression model. 

Interpreting the results of the cost regression analysis, codon adaptation is the strongest predictor of both the transcript and protein expression data. The other factors in the regression equation, while do show some effect in prediction are much less important than that of codon adaptation.

\subsubsection{Predictors of metabolite levels}

Codon adaptation is not applicable to metabolite levels, and therefore all explanatory power is based on amino acid cost, genome encoded tRNA, and carbon, nitrogen and sulphur content. The variation explained in the metabolite data was much greater than that of the transcript and protein data, more than twice as much depending on which cost measure was used. This result indicates that the combination of factors used in that analysis are strong predictors of the variation in metabolomic data, and that amino acids are maintained in the cell at levels related to cost, atomic content and genome encoded tRNA levels.

\subsection{Amino acid cost is a small factor in yeast protein evolution}
