\section{Introduction}

\subsection{Measuring amino acid cost in gene expression}

\subsection{Amino acid cost in protein evolution}

\section{Results}

\subsection{Estimating the importance of amino acid cost in gene expression}

Amino acid cost in gene expression was estimated using multivariate regression on transcriptomic, proteomic and metabolomic datasets. The aim of this analysis was to estimate the importance of amino acid in the protein production, of which each of these datasets are related.

The data in this analysis was the taken from a large scale systems biology analysis in \emph{S.cerevisiae} \cite{castrillo2007}. This dataset contained yest proteomic, transcriptomic and metabolomic data from four nutrient limiting conditions carbon, nitrogen, sulphur and phosphorus. The aim of this analysis was to determine the effects of each of the nutrient limitations on the yeast gene expression.

\subsubsection{Multivariate Regression}

Multivariate regression was used to estimate the importance of amino acid cost in gene expression. For the regression the response variable was the measured level of the data in the original analysis, this being either transcripts, proteins or free amino acids in the metabolome. The explanatory variables were as follows: mean amino acid carbon, nitrogen and sulphur content for the protein, codon adaptation index of the encoding transcript, average genome encoded tRNA number for each codon in the transcript, and average amino acid cost.

Average carbon, nitrogen and sulphur content represents the total content for each atom in the protein divided by the length of the protein excluding stop codons. For the transcriptomic data this represents the atomic content of the encoded protein. In the free amino acid data the atomic content represents the content for each individual amino acid. The codon adaptation index (CAI) represents the optimisation of the transcript for translation, where the use of certain codons are preferential for translation \cite{ikemura1982}. For the protein data the CAI of the encoding transcript was used. CAI was not applicable for the free amino acid data. The tRNA count for a transcript represents the total number of genome encoded tRNAs for each codon divided by the length of the transcript excluding stop codons. For the protein data the average tRNA count of the encoding transcript was used. For the free amino acid data, the total number of tRNAs was used. The amino acid cost variable used was one of those described in the previous chapter. The multivariate regression was repeated where each amino acid cost set was cycled as the variable used in the analysis. The costs used were as follows: Akashi \& Gojobori energy \cite{akashi2002}, Craig \& Weber energetic cost, and biosynthetic steps \cite{craig1998}, Wager fermentative and respiratory energetic cost \cite{wagner2005}, amino acid molecular weight (Da) as used by Seligmann \cite{seligmann2004}, and the $A_{glucose}$ and $R_{glucose}$ costs derived in Section\vref{section:amino_acid_cost_estimation}. The aim of cycling the costs was determine which cost has the most explanatory power for explaining variation in cell physiological data.

The importance of each variable in the multivariate regression was analysed by removing the variable then comparing the reduced model with the model containing all the variables. Akaiki's information criterion (AIC) \cite{akaike1974} was used to the compare the explanatory power of the two models. The AIC is a log likelihood score for the regression, which penalises models with more parameters. This allows models with differing numbers of parameters to be compared. For each regression, a complete model was fitted containing all interaction terms, stepwise automated variable removal was then used to reduce interactions and variables based on the AIC score. A negative AIC score indicates the reduced model should be used as it is more parsimonious. A positive score indicates the larger model provides a better fit to the data. The results of this analysis for each dataset are shown in Figure~\vref{figure:regression_analysis} and in Table~\vref{table:regression_analysis}.

\begin{figure}
  \centering
  \includegraphics*[height=18cm]{regression_analysis.eps}
  \caption[Transcript, proteomic and metabolomic multivariate regression results.]{Transcript, proteomic and metabolomic multivariate regression results. The y-axis indicates the variable removed from the multivariate regression. The x-axis indicates the log$_{10}$ AIC difference between the complete model, and the model with all variables included. The legend indicates which amino acid cost type was used to calculate the per-residue cost. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{figure:regression_analysis}
\end{figure}

\begin{table}
  \centering
  \begin{footnotesize}
    \begin{tabular}{ l *{4}{ p{2cm}} }
      \toprule
      Cost type                          & Transcripts  & Proteins & Amino Acids \\ \midrule
      \emph{S. cerevisiae} $A_{glucose}$ & 0.389        & 0.406    & 0.782       \\
      \emph{S. cerevisiae} $R_{glucose}$ & 0.383        & 0.408    & 0.875       \\
      Akashi \& Gojobori (2002)          & 0.398        & 0.405    & 0.805       \\
      Craig \& Weber (1998) Energy       & 0.416        & 0.40     & 0.835       \\
      Craig \& Weber (1998) Steps        & 0.375        & 0.404    & 0.866       \\
      Wagner (2005) Respiratory          & 0.382        & 0.405    & 0.822       \\
      Wagner (2005) Fermentative         & 0.377        & 0.406    & 0.851       \\
      Molecular Weight                   & 0.422        & 0.405    & 0.767       \\ \bottomrule
    \end{tabular}
  \end{footnotesize}
  \caption[Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets.]{Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets. Each value in the table is the $R^2$ for the complete multivariate regression model using the indicated cost type, and including the carbon, nitrogen, sulphur content of the amino acid, average tRNA count, and codon adaptation index for the transcript and protein data. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{table:regression_analysis}
\end{table}

\paragraph{Transcriptome analysis}

\paragraph{Proteome analysis}

\paragraph{Metabolome analysis}

\subsection{The role of amino acid cost in yeast gene evolution}

\subsubsection{Protein wide cost}

\subsubsection{Single residue cost}

\section{Discussion}

\subsection{Amino acid cost is a small factor in yeast gene expression}

\subsection{Amino acid cost is a small factor in yeast protein evolution}

\section{Materials and Methods}
