\section{Introduction}

\subsection{Measuring amino acid cost in gene expression}

\subsection{Amino acid cost in protein evolution}

\section{Results}

\subsection{Estimating the importance of amino acid cost in gene expression}

Amino acid cost in gene expression was estimated using multivariate regression on transcriptomic, proteomic and metabolomic datasets. The aim of this analysis was to estimate the importance of amino acid in the protein production, of which each of these datasets are related.

The data in this analysis was the taken from a large scale systems biology analysis in \emph{S.cerevisiae} \cite{castrillo2007}. This dataset contained yest proteomic, transcriptomic and metabolomic data from four nutrient limiting conditions carbon, nitrogen, sulphur and phosphorus. The aim of this analysis was to determine the effects of each of the nutrient limitations on the yeast gene expression.

\subsubsection{Multivariate Regression}

Multivariate regression was used to estimate the importance of amino acid cost in gene expression. For the regression the response variable was the measured level of the data in the original analysis, this being either transcripts, proteins or free amino acids in the metabolome. The explanatory variables were as follows: mean amino acid carbon, nitrogen and sulphur content for the protein, codon adaptation index of the encoding transcript, average genome encoded tRNA number for each codon in the transcript, and average amino acid cost.

Average carbon, nitrogen and sulphur content represents the total content for each atom in the protein divided by the length of the protein excluding stop codons. For the transcriptomic data this represents the atomic content of the encoded protein. In the free amino acid data the atomic content represents the content for each individual amino acid. The codon adaptation index (CAI) represents the optimisation of the transcript for translation, where the use of certain codons are preferential for translation \cite{ikemura1982}. For the protein data the CAI of the encoding transcript was used. CAI was not applicable for the free amino acid data. The tRNA count for a transcript represents the total number of genome encoded tRNAs for each codon divided by the length of the transcript excluding stop codons. For the protein data the average tRNA count of the encoding transcript was used. For the free amino acid data, the total number of tRNAs was used. The amino acid cost variable used was one of those described in the previous chapter. The multivariate regression was repeated where each amino acid cost set was cycled as the variable used in the analysis. The costs used were as follows: Akashi \& Gojobori energy \cite{akashi2002}, Craig \& Weber energetic cost, and biosynthetic steps \cite{craig1998}, Wager fermentative and respiratory energetic cost \cite{wagner2005}, amino acid molecular weight (Da) as used by Seligmann \cite{seligmann2004}, and the $A_{glucose}$ and $R_{glucose}$ costs derived in Section\vref{section:amino_acid_cost_estimation}. The aim of cycling the costs was determine which cost has the most explanatory power for explaining variation in cell physiological data.

The importance of each variable in the multivariate regression was analysed by removing the variable then comparing the reduced model with the model containing all the variables. Akaiki's information criterion (AIC) \cite{akaike1974} was used to the compare the explanatory power of the two models. The AIC is a log likelihood score for the regression, which penalises models with more parameters. This allows models with differing numbers of parameters to be compared. For each regression, a complete model was fitted containing all interaction terms, stepwise automated variable removal was then used to reduce interactions and variables based on the AIC score. A negative AIC score indicates the reduced model should be used as it is more parsimonious. A positive score indicates the larger model provides a better fit to the data. The results of this analysis for each dataset are shown in Figure~\vref{figure:regression_analysis} and in Table~\vref{table:regression_analysis}.

\begin{figure}
  \centering
  \includegraphics*[height=18cm]{regression_analysis.eps}
  \caption[Transcript, proteomic and metabolomic multivariate regression results.]{Transcript, proteomic and metabolomic multivariate regression results. The y-axis indicates the variable removed from the multivariate regression. The x-axis indicates the log$_{10}$ AIC difference between the complete model, and the model with all variables included. The legend indicates which amino acid cost type was used to calculate the per-residue cost. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{figure:regression_analysis}
\end{figure}

\begin{table}
  \centering
  \begin{footnotesize}
    \begin{tabular}{ l *{4}{ p{2cm}} }
      \toprule
      Cost type                          & Transcripts  & Proteins & Amino Acids \\ \midrule
      \emph{S. cerevisiae} $A_{glucose}$ & 0.389        & 0.406    & 0.782       \\
      \emph{S. cerevisiae} $R_{glucose}$ & 0.383        & 0.408    & 0.875       \\
      Akashi \& Gojobori (2002)          & 0.398        & 0.405    & 0.805       \\
      Craig \& Weber (1998) Energy       & 0.416        & 0.40     & 0.835       \\
      Craig \& Weber (1998) Steps        & 0.375        & 0.404    & 0.866       \\
      Wagner (2005) Respiratory          & 0.382        & 0.405    & 0.822       \\
      Wagner (2005) Fermentative         & 0.377        & 0.406    & 0.851       \\
      Molecular Weight                   & 0.422        & 0.405    & 0.767       \\ \bottomrule
    \end{tabular}
  \end{footnotesize}
  \caption[Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets.]{Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets. Each value in the table is the $R^2$ for the complete multivariate regression model using the indicated cost type, and including the carbon, nitrogen, sulphur content of the amino acid, average tRNA count, and codon adaptation index for the transcript and protein data. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{table:regression_analysis}
\end{table}

\paragraph{Transcriptome and proteome analysis}

The regression analysis of the transcript data from Castrillo \emph{et al.} \cite{castrillo2007} and the proteome data from \cite{ghaemmaghami2003} yielded similar results. In both these data codon adaptation index was the strongest predictor of measure levels of either transcript or protein. Figure~\ref{figure:regression_analysis} shows the CAI of the transcript was approximately half a magnitude greater as a predictor of transcription than the other included factors. In the protein data CAI was a full magnitude greater as a predictor of protein production. Together these results indicate the optimisation of a transcript for translation is a strong predictor of both transcript and protein levels.

Comparing other factors in the regression, amino acid cost, carbon content and nitrogen appear to show a similar level of importance in explaining variation in both transcript and protein data. The variables for tRNA count and and sulphur content appear slightly less important. Overall all though the regression analysis indicates that the majority of the variation in the transcript and protein data is explained by codon adaption index of the transcript, as opposed to the variables of costs, carbon, nitrogen, sulphur content, or corresponding genome encoded tRNAs.

As the first two columns of Table~\vref{table:regression_analysis} show, each of the different amino acid cost types had little effect on the $R^2$ of the multivariate regression of transcript and protein data. The difference in which cost type used in the transcript analysis was 4.7\%, ranging from 37.5\% for the regression using Craig \& Weber's step cost measures \cite{craig1998}, to 42.2\% for the regression using molecular weight as suggested Seligmann \cite{seligmann2004}. In the analysis of the protein data, the difference in variation explained dependent of the cost measure was even smaller 0.8\% with all cost measures having approximately the same effect in the multivariate regression of the proteome data.

\paragraph{Metabolome analysis}

In the regression of the metabolome data from Castrillo \emph{et al.} \cite{castrillo2007} codon adaptation index is not applicable and therefore the regression is based on genome encoded tRNAs, estimated cost and carbon, nitrogen and sulphur content of the amino acid. The third plot of Figure~\ref{figure:regression_analysis} illustrates that in the majority of each regression each of these factors appears to contribute equally to model fitting of free amino acid data in yeast. The exception to this is sulphur content which in half the regressions explains less regression than the other factors.

Compared with the multivariate regression of the transcript and protein data, the multivariate regression of the metabolite regression of the free amino acid data was able to explain a much larger degree of variation. The variation explained differed by 10.8\% between the best and worst regression model. The regression that explained the least used molecular weight for amino acid cost, while the best fit used $R_{glucose}$ explaining 87.5\%. This result indicates that a large degree of the variation in amino acid levels can be explained based on their cost, atomic content, and the corresponding tRNAs encoded in the genome.

\subsection{The role of amino acid cost in yeast gene evolution}

\subsubsection{Protein wide cost}

\subsubsection{Single residue cost}

\section{Discussion}

\subsection{Amino acid cost is a small factor in yeast gene expression}

\subsection{Amino acid cost is a small factor in yeast protein evolution}

\section{Materials and Methods}

\subsubsection{Amino acid characteristics}

\emph{S. cerevisiae} codon adaptation index (CAI) data was produced by Wall \emph{et al.} \cite{wall2005}. Genome encoded tRNA count data was produced by Akashi \cite{akashi_trna}. The $A_{glucose}$ and $R_{glucose}$ costs used were those described in the previous chapter. The other amino acid costs used are reported in the literature by Craig \& Weber \cite{craig1998}, Akashi \& Gojobori \cite{akashi2002}, Wagner \cite{wagner2005} and Seligmann \cite{seligmann2004}. Atomic content for each amino acid was taken according to universally available descriptions.

\subsubsection{Experimental data}

The transcript and metabolomic data used in this analysis was taken from Castrillo \emph{et al.} \cite{castrillo2007}. The data was measured from \emph{S. cerevisiae} cultured continuously in a chemostat. Each measurement was taken at one of three dilution rates 0.1h$^-1$, 0.2h$^-1$ and 0.3h$^-1$ and four different nutrient limiting conditions, glucose, ammonium, sulphate and phosphate. The transcript data was produced from replicate microarray measurements of total RNA. Following measurement the RNA data was processed using with RMA quantile normalisation \cite{RMA}. The metabolite data was measured using Gas chromatography followed by time of flight mass spectrometry (GC/TOF-MS). The metabolite data was normalised using median absolute deviation. Missing metabolite data were inferred from replicates in the same conditions.

The protein data used were produced by Ghaemmaghami \emph{et al.} \cite{ghaemmaghami2003}. These data were derived from tandem affinity purification of TAP-tagged \emph{S.cerevisiae} ORFs. Each tagged ORF was measured using antibody quantification of the encoded tag. Absolute levels of protein per cell were estimated by comparing the \emph{S.cerevisiae} gene measurements with a purified E. coli INFA-TAP construct scale.

\subsubsection{Multivariate regression}

Each variable used in the regression transformed by the natural logarithm.  The exception to this was the transcript data which was logged during the original processing of the data. The sulphur content of individual amino acids contained zero values, therefore a small value (0.0001) was added so the logarithm could be taken. All datasets were scaled to have the same mean and variance. The aim of this data processing was to minimise heteroscedasticity or over-variation in any of the data influencing regression fitting. The metabolite data was averaged for each set of experimental parameters to prevent pseudo-replication resulting from the inference of missing values.

The multivariate regression was performed using the R language for statistical computing \cite{R}. For each dataset, transcript, protein or metabolite, the measured level of the entity in the cell was the response variable in the regression equation. The explanatory variables in the regression equation were carbon, nitrogen, and sulphur content, amino acid cost, codon adaptation index. Environment and dilution rate were also included for the transcript and metabolite data.

For each dataset the complete model equation was fitted including all factors and interactions between factors. The R function \emph{lm} was used as the regression function. Stepwise interaction/variable removable was then performed where interactions were removed from the compete model, to produce a reduced model. The reduced model was compare with the previous complete model, and if the reduced model was considered more parsimonious based on the AIC difference, then the reduced model was then used as the current model. This process was repeated until no more interactions could be removed.

The importance of each explanatory variable in the regression equation was assessed by by repeating the above process but with one of the explanatory variables removed. The regression model with the variable removed was then compared to the complete regression model with all factors included. The AIC difference between the two equations was used as a measure of importance for the removed variable. This process was repeated, where each of the cost types was cycled between those in the introduction.
