\section*{Summary}

This chapter examines if amino acid cost is force in the evolution of protein sequence. Comparing the usage of each amino acid in the \emph{Saccharomyces cerevisiae} proteome with the predict cost of the amino acid indicates that cost minimisation is a weak selective pressure. Further comparison of amino acid cost with transcript and protein production levels show that cost minimisation is a much weaker force than compared with transcript optimisation. Free amino acids in the metabolite pool are however maintained at levels related to their cost of synthesis.

These findings show cost minimisation is a weak in shaping the composition of the genome and expressed transcripts in \emph{S. cerevisiae}. Comparison of the predicted proteome of related \emph{Saccharomyces} species however indicates the mutation rates of amino acids is correlated with amino acid cost suggesting a purifying selection pressure to maintain expensive amino acids for a specific functional role, while allowing greater flexibility in the use of cheaper amino acids.

\clearpage

\section{Introduction}

\subsection{Metabolic cost as a selection pressure}

The amino acid composition of a predicted proteome varies across species, and is related to GC content and environment \cite{tekaia2006}. The selective pressures influencing the use of amino acids in individual protein sequence are reviewed by Pal \emph{et al.} \cite{pal2006} and are illustrated in Table~\vref{table:protein_selective_pressures}.

The premise of the cost selection hypothesis \cite{akashi2002} is that if two amino acids can perform a similar function at a position in a protein sequence, selection prefers the biosynthetically cheaper amino acid. The difference in biosynthetic materials is diverted to other processes. The correctness of this assumption of cost as a selective pressure is still unclear, though two studies suggests that cost selection plays only a minor role in protein evolution \cite{rocha2004} or expression \cite{raiford2008}.

\begin{table}
  \centering
  \begin{tabular}{ l l l }
  \toprule
  Selection & Name                   & Description \\ \midrule
  Purifying & Fitness density        & Text \\
            & Dispensibility         & Text \\
            & Structure              & Text \\
            & Network position       & Text \\
            & Developmental position & Text \\
            & Expression             & Text \\
  Positive  & Competitive            & Text \\
            & Compensatory           & Text \\
            & Adaptive               & Text \\ \bottomrule
  \end{tabular}
  \caption[Factors in protein evolution]{Long caption}
  \label{table:protein_selective_pressures}
\end{table}

\subsection{Genome composition}

The genome composition of an organism may reflect trends for preferential amino acid usage. The cyanobacterium \emph{Calothrix} sp. 7601 encodes sulphur eradicated versions of proteins abundant in sulphur limiting conditions, where the only sulphur containing residues are the start methionine and a single cysteine binding site \cite{mazel1989}. In \emph{Escherichia coli} and \emph{Saccharomyces cerevisiae} proteins in pathways responsible for recruiting carbon and sulphur from the environment are depleted for the same nutrient \cite{baudoin2001}. The advantage of depleting the enzymes of for the nutrient is that the protein may be easier expressed for in conditions limiting for the nutrient. Both of these studies indicate, in a small subset of the total proteome, the use of amino acids with specific atoms may come at a cost when availability of a specific atom is restricted.

The trend for specific amino acid use is considered using a larger dataset across 141 predicted proteomes derived from available genome sequences \cite{bragg2006}. The difference in mean sulphur content between orthologs was correlated with mean sulphur content in the proteome, suggesting a selection pressure for reducing sulphur content. Thermophiles have a lower sulphur content than non-thermophiles, the synthesis and maintenance of methionine and cysteine residues is the likely factor in the utilisation of these residues. Sulphur containing amino acids also followed specific trends related to GC content, though the usage of all amino acids is associated with GC content \cite{tekaia2006}.

Analysis of large numbers of complete genomes was also used to determine whether the molecular weight of amino acids is minimised \cite{seligmann2003}. Across a large set of genomes the use of heavier amino acids is minimised. Furthermore cost minimisation is a greater factor for organisms synthesising amino acids, than for organisms absorbing amino acids from the environment.

\subsection{Gene expression}

The encoding of amino acids is expressed in the proteome, and therefore cost trends should be related to gene expression. A large proportion of cellular energy is spent of protein synthesis, and the cost minimisation theory predicts that minimising the use of expensive amino acids provides a fitness advantage. The highly expressed genes may be under particular pressure to minimise cost since the cost of synthesis is multiplied by the number of protein copies. The selection pressure for cost minimisation should also therefore decrease with expression level.

The selection pressure for cost minimisation can be estimated from the cost of the amino acid residues encoded in highly expressed genes. Akashi and Gojobori \cite{akashi2002} used Craig and Weber's amino acid cost estimates \cite{craig1998} to determine whether the strength of the trend between expression level and cost. Comparison of per residue energetic cost with major codon usage predicted gene expression \cite{kanaya1999} indicated protein cost does decrease with level of expression.

Akashi and Gojobori performed their analysis in two mesophiles: \emph{E. coli} and \emph{B. subtilis}. The importance of cost minimisation in gene expression may be related to organism lifestyle, where species with large differences in metabolism may have different measures of amino cost and selection pressures to minimise these costs. Heizer \emph{et al.} \cite{heizer2006} examined cost related trends in an additional four organisms including chemoheterotrophic, photoautotrophic, and thermophilic lifestyles. Across all included species the trend between predicted gene expression and biosynthetic cost indicated a pressure to minimise costly amino acids at high expression levels.

TODO: Is mesophile the correct term?

TODO: Include descriptions of the meaning of each life style

Changes in the expression level of a gene may also expected to have an effect on the use of cellular energy. An increase in gene expression will require the greater energy diversion to the synthesis of the required amino acids, while a decrease will free cellular energy for other processes. Wagner \cite{wagner2005} examined whether gene duplication events or mutations in regulatory regions are also constrained by a cost selection pressure. The resulting cost increase form a doubling of gene expression from a duplication event may be expected to be greater than the negative selection pressure for the \emph{S. cerevisiae} effective population size. This indicates that cost is a factor in the retention or loss of gene duplicates.

The previous described studies examined the cost minimisation in gene expression using indicators of expression based on codon usage. Validation of the extent of cost minimisation must examine \emph{in vivo} expression. Raiford \emph{et al.} \cite{raiford2008} compared the trend to minimise the use of expensive amino acids with \emph{S. cerevisiae} codon adaptation, transcript levels, and protein levels. This analysis correlated the aerobic and anaerobic costs with the usage of each amino acids in expression. The results of the study showed a trend for cost minimisation, but the trend varies according to amino acid class, and explains only a limited degree of variation in gene expression.

Such results indicate the selective for cost minimisation may have only a small impact on the evolution of proteins. Bragg and Wagner \cite{bragg2007} compared the carbon content of up-regulated genes versus the rest of the proteome, and found no indication of cost minimisation in the up-regulated set. Furthermore a yeast strain evolved in carbon limitation was found to use more carbon atoms per residue than the unadapted strain.

The current literature describes a mixed view of the strength of cost minimisation as a selective pressure in protein evolution. Analysis of the genome data, comparing predicted expression level with predicted per residue biosynthetic cost indicates that cost minimisation may be a strong selective force. Cost minimisation however appears to be a much weaker force when comparing actual expression data. Analysis of enzymes protein sequences, and simulation of various nutrient limiting conditions indicates cheaper metabolically inefficient enzymes are maintained for expression in nutrient scarcity. Biosynthetically more expensive efficient copies may then be used when nutrients become more available \cite{carlson2007}.

\subsection{Protein sequence evolution}

If there is a selective force for the optimisation of protein sequence for cheaper residues, then biosynthetic cost should be a factor in amino acid substitution rates. Whether this is true or not is still unclear and remains to be investigated. The literature described so far has examined either genome composition or gene expression for trends in cost minimisation; the existence of such indicates that cost minimisation is a selective pressure.

Craig and Weber \cite{craig1998} found that replacement amino acids in a small set of \emph{E. coli} genes were cheaper than the mean or median amino acid cost. Amino acid cost in another examined set of \emph{E. coli} proteins were found to the same as the amino acid cost, presenting a mixed view of the importance of cost minimisation.

Metabolic cost was compared directly in a multivariate regression of possible \emph{E. coli} and \emph{B. subtilis} protein evolution determinants. Of all the protein characteristics included: expression level, functional category, essentiality, and metabolic cost (Akashi and Gojobori energetic cost\cite{akashi2002}); cost minimisation is a weak predictor of protein evolution \cite{rocha2004}.

Metabolic cost is often interpreted as the energy required for synthesis or the use of specific atoms such as nitrogen or sulphur. But what if the costs of amino acid selection is more complex than existing current estimates of energy or atomic content? Amino acid composition analysis of 208 encoded proteomes indicates trends for amino usage associated with organism life style \cite{tekaia2006}. Comparison of paralogs between thermophile and non-thermophile related species, controlling for protein structure and function, highlights amino acid preferences dependent on the organism's lifestyle. One possible reason, as Swire discusses, is hyperthermophile are subject to different energetic constraints given the temperature of the environment that may affect Gibb's free energy of metabolic reactions \cite{swire2007}.

Sequencing samples of environments provides varying genome coverage of species living in the same environment. Reconstruction of the metabolic pathways can be used to determine how pathways vary between environment. Amino acid biosynthetic pathways vary between oceanic environments, and the variance is unrelated to biosynthetic cost. Instead the usage of a amino acid synthetic pathway, in particular methionine, may instead depend on the availability of the cofactors in the synthesis of the amino acid. A negative correlation between the number of synthetic pathways and number transporters of the same acid indicate a trade-off between synthesis and import \cite{gianoulis2009}. The metabolic use of cofactors or the import of amino acids in the environment are not readily apparent from the metabolic map of an organism, but may have an intimate effect on the usage of an amino acids not explained by energetic or atomic cost.

\subsection{Summary of results}

In this chapter the genome composition of \emph{S. cerevisiae} is compared with several measures of amino acid cost to determine if amino acid cost has shaped the sequences of the encoded proteins. The cellular levels of transcripts, proteins, and free amino acids is determine the extent of cost minimisation compared with other indicators of protein production. The amino acid usage across all open reading frames is compared between seven \emph{Saccharomyces} to establish the role of cost in the closely related species. The last analysis in this chapter focuses on the role of amino acid cost on mutation rates by analysing the amino acid changes in the protein sequence alignments of four \emph{Saccharomyces} species.

\clearpage

\section{Materials and Methods}

\subsubsection{Protein characteristics}

\emph{S. cerevisiae} codon adaptation index (CAI) data was produced by Wall \emph{et al.} \cite{wall2005}. Genome encoded tRNA count data was produced by Akashi \cite{akashi2003}. The $A_{glucose}$ and $R_{glucose}$ costs used were those described in the previous chapter. The other amino acid costs used are reported in the literature by Craig \& Weber \cite{craig1998}, Akashi \& Gojobori \cite{akashi2002}, Wagner \cite{wagner2005} and Seligmann \cite{seligmann2003}. Atomic content for each amino acid was taken according to universally available descriptions.

\subsubsection{Gene expression experimental data}

The transcript and metabolomic data used in this analysis was taken from Castrillo \emph{et al.} \cite{castrillo2007}. The data was measured from \emph{S. cerevisiae} cultured continuously in a chemostat. Each measurement was taken at one of three dilution rates 0.1h$^-1$, 0.2h$^-1$ and 0.3h$^-1$ and four different nutrient limiting conditions, glucose, ammonium, sulphate and phosphate. The transcript data was produced from replicate microarray measurements of total RNA. Following measurement the RNA data was processed using with RMA quantile normalisation \cite{bolstad2003}. The metabolite data was measured using Gas chromatography followed by time of flight mass spectrometry (GC/TOF-MS). The metabolite data was normalised using median absolute deviation. Missing metabolite data were inferred from replicates in the same conditions.

The protein data used were produced by Ghaemmaghami \emph{et al.} \cite{ghaemmaghami2003}. These data were derived from tandem affinity purification of TAP-tagged \emph{S.cerevisiae} ORFs. Each tagged ORF was measured using antibody quantification of the encoded tag. Absolute levels of protein per cell were estimated by comparing the \emph{S.cerevisiae} gene measurements with a purified \emph{E. coli} INFA-TAP construct scale.

\subsubsection{Gene expression multivariate regression}

Each variable used in the regression transformed by the natural logarithm. The exception to this was the transcript data which was logged during the original processing of the data. The sulphur content of individual amino acids contained zero values, therefore a small value (0.0001) was added so the logarithm could be taken. All datasets were scaled to have the same mean and variance. The aim of this data processing was to minimise heteroscedasticity or over-variation in any of the data influencing regression fitting. The metabolite data was averaged for each set of experimental parameters to prevent pseudo-replication resulting from the inference of missing values.

The multivariate regression was performed using the R language for statistical computing \cite{R}. For each dataset, transcript, protein or metabolite, the measured level of the entity in the cell was the response variable in the regression equation. The explanatory variables in the regression equation were carbon, nitrogen, and sulphur content, amino acid cost, codon adaptation index. Environment and dilution rate were also included for the transcript and metabolite data.

For each dataset the complete model equation was fitted including all factors and interactions between factors. The R function \emph{lm} was used as the regression function. Stepwise interaction/variable removable was then performed where interactions were removed from the compete model, to produce a reduced model. The reduced model was compare with the previous complete model, and if the reduced model was considered more parsimonious based on the AIC difference, then the reduced model was then used as the current model. This process was repeated until no more interactions could be removed.

The importance of each explanatory variable in the regression equation was assessed by by repeating the above process but with one of the explanatory variables removed. The regression model with the variable removed was then compared to the complete regression model with all factors included. The AIC difference between the two equations was used as a measure of importance for the removed variable. This process was repeated, where each of the cost types was cycled between those in the introduction.

\subsection{Amino acid usage in \emph{Saccharomyces} proteomes}

The verified open reading frames from seven \emph{Saccharomyces} related species were downloaded from the \emph{Saccharomyces} genome database. The species were \emph{paradoxus}, \emph{bayanus}, \emph{castellii}, \emph{cerevisiae}, \emph{kluyveri}, \emph{kudriavzevii}, and \emph{mikatae}. The genomes of \emph{paradoxus} and \emph{bayanus} were sequenced by Kellis \emph{et al.} \cite{kellis2003}. The \emph{castellii}, \emph{kluyveri}, \emph{kudriavzevii}, and \emph{mikatae} genomes were sequenced by Cliften \emph{et al.} \cite{cliften2003}.

The percentage amino acid content of each species was calculated by summing the frequency of each amino acid over the total predicted proteome and dividing by the total number of amino acids. Stop codons and ambiguous amino acid definitions were excluded from the analysis. Only the twenty standard amino acids were counted. Open reading frames containing an internal stop codon were also excluded from the analysis.

The variation in amino acid usage between species was calculated using median absolute deviation (MAD). MAD is a more robust measure of variance for non-normally distributed, or small datasets. MAD is calculated similarly to standard deviation, except the median absolution deviations from the median is calculated.

\subsubsection{Hierarchical clustering}

Hierarchical clustering of data was calculated using the complete linkage agglomerative clustering. Non-parametric Spearman's rank correlation was used to calculate the distance between either species or amino acids, based on percentage amino acid usage in each species. The correlation between variables was converted to a distance matrix using the normal square (Euclidean) distance between two correlation estimates.

\subsubsection{Singular value decomposition}
 
Singular value decomposition was applied to the percentage frequency matrix of amino acid across species described above. Singular value decomposition breaks the amino acid usage matrix $X$ into three matrices, $U$ and $V$ represent the vector contributions for amino acid acids and species towards the vector $D$ the singular values. Each value in $D$ the weight of an unobserved factor in the matrix $X$, and each variable in $U$ and $V$  represents the contribution to a greater or lesser extent of species and amino acid to each value in $D$ such that $X = U \dot D \dot V'$. This function was performed in R using the ``svd'' function.

TODO: More detail for SVD

\subsubsection{Canonical correlation}

Canonical correlation was used to determine the maximum correlation between the four amino acid characteristics and the usage of amino acids across species (matrix $Y$). A matrix $X$ was produced containing twenty rows, one for each amino acid, and four columns: molecular weight (Da), and the measures $A_{glucose}$, $A_{ammonium}$, and $A_{sulphate}$ derived in the previous chapter. Each observation was rounded to two decimal places. Canonical correlation was used to determined the maximum correlation between the linear combinations of unobserved variables in $X$ with those of $Y$. This analysis was performed in R using the ``cancor'' function.

TODO: More detail on the estimation of a derived cost

\subsection{Amino acid cost in \emph{Saccharomyces} protein evolution}

The codon aligned gene sequences of the \emph{cerevisiae}, \emph{mikatae}, \emph{bayanus}, and \emph{paradoxus} species generated by Someone \emph{et al.} \cite{wall2005}. The evolutionary rate of the entire alignment and each position in the alignment was estimated using the codeml tool, part of the PAML package \cite{yang2007}. The tool rind \cite{bruno1996} was used to estimate the phylogeny-weighted frequency each amino acid at each site in each alignment. A phylogeny weighted cost of each position in each alignment was estimated by estimating the mean of each amino acid in the alignment weighted by the cost of the amino acid.

TODO: Insert weighted mean equation

TODO: What sequences were excluded? Check validations

The frequency of amino acid fixation across species was estimated by determining each amino acid in \emph{cerevisiae} protein in each alignment, and counting the number of instances each amino acid was used at the same position in the other alignment proteins. The ratio of fixation was then calculated by dividing the number of alignment columns where the same amino was used across species, by the total number of instances of the amino acid was used in \emph{cerevisiae} across all alignments. Alignment columns containing gaps were ignored.

A similar method was used to calculate the average mutation rate of each amino acid. For each of the twenty amino acids in turn, the mutation rate of each alignment column was summed over the all sites where the amino acid occurred at the \emph{cerevisiae} position in the alignment, this divided by the total number of sites to produce a mean mutation rate for that amino acid. Alignment columns containing gaps were ignored.

TODO: What are the units of mutation rate

The residuals of the relationship between ratio of fixed sites to average mutation rate was estimated using linear regression. A direct linear model of mean mutation rate as a function of fixed sites was used as this model provided the lowest AIC when compared with modelling a quadratic or logarithmic relationship between the two models. Inferring residuals from a regression can be dangerous as each observation may influence the estimation of the relationship to a greater of lesser degree, which in turn influences residual errors. Robust linear modelling used as this iteratively reduces the least squares of the model estimation reducing the influence of single points on model estimation.

TODO: Read up on method of robust linear modelling. How exactly does it work?

\clearpage

\section{Results}

\subsection{Amino acid cost in gene expression}

Amino acid cost in gene expression was estimated using multivariate regression on transcriptomic, proteomic and metabolomic datasets. The aim of this analysis was to estimate the importance of amino acid in the protein production, of which each of these datasets are related.

The data in this analysis was the taken from a large scale systems biology analysis in \emph{S.cerevisiae} \cite{castrillo2007}. This dataset contained yest proteomic, transcriptomic and metabolomic data from four nutrient limiting conditions carbon, nitrogen, sulphur and phosphorus. The aim of this analysis was to determine the effects of each of the nutrient limitations on the yeast gene expression.

\subsubsection{Multivariate Regression}

Multivariate regression was used to estimate the importance of amino acid cost in gene expression. For the regression the response variable was the measured level of the data in the original analysis, this being either transcripts, proteins or free amino acids in the metabolome. The explanatory variables were as follows: mean amino acid carbon, nitrogen and sulphur content for the protein, codon adaptation index of the encoding transcript, average genome encoded tRNA number for each codon in the transcript, and average amino acid cost.

Average carbon, nitrogen and sulphur content represents the total content for each atom in the protein divided by the length of the protein excluding stop codons. For the transcriptomic data this represents the atomic content of the encoded protein. In the free amino acid data the atomic content represents the content for each individual amino acid. The codon adaptation index (CAI) represents the optimisation of the transcript for translation, where the use of certain codons are preferential for translation \cite{ikemura1982}. For the protein data the CAI of the encoding transcript was used. CAI was not applicable for the free amino acid data. The tRNA count for a transcript represents the total number of genome encoded tRNAs for each codon divided by the length of the transcript excluding stop codons. For the protein data the average tRNA count of the encoding transcript was used. For the free amino acid data, the total number of tRNAs was used. The amino acid cost variable used was one of those described in the previous chapter. The multivariate regression was repeated where each amino acid cost set was cycled as the variable used in the analysis. The costs used were as follows: Akashi \& Gojobori energy \cite{akashi2002}, Craig \& Weber energetic cost, and biosynthetic steps \cite{craig1998}, Wagner fermentative and respiratory energetic cost \cite{wagner2005}, amino acid molecular weight (Da) as used by Seligmann \cite{seligmann2003}, and the $A_{glucose}$ and $R_{glucose}$ costs derived in Section\vref{section:amino_acid_cost_estimation}. The aim of cycling the costs was determine which cost has the most explanatory power for explaining variation in cell physiological data.

The importance of each variable in the multivariate regression was analysed by removing the variable then comparing the reduced model with the model containing all the variables. Akaiki's information criterion (AIC) \cite{akaike1974} was used to the compare the explanatory power of the two models. The AIC is a log likelihood score for the regression, which penalises models with more parameters. This allows models with differing numbers of parameters to be compared. For each regression, a complete model was fitted containing all interaction terms, stepwise automated variable removal was then used to reduce interactions and variables based on the AIC score. A negative AIC score indicates the reduced model should be used as it is more parsimonious. A positive score indicates the larger model provides a better fit to the data. The results of this analysis for each dataset are shown in Figure~\vref{figure:regression_analysis} and in Table~\vref{table:regression_analysis}.

\begin{figure}
  \centering
  \includegraphics*[height=18cm]{regression_analysis.eps}
  \caption[Transcript, proteomic and metabolomic multivariate regression results.]{Transcript, proteomic and metabolomic multivariate regression results. The y-axis indicates the variable removed from the multivariate regression. The x-axis indicates the log$_{10}$ AIC difference between the complete model, and the model with all variables included. The legend indicates which amino acid cost type was used to calculate the per-residue cost. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{figure:regression_analysis}
\end{figure}

\begin{table}
  \centering
  \begin{footnotesize}
    \begin{tabular}{ l *{4}{ p{2cm}} }
      \toprule
      Cost type                          & Transcripts  & Proteins & Amino Acids \\ \midrule
      \emph{S. cerevisiae} $A_{glucose}$ & 0.389        & 0.406    & 0.782       \\
      \emph{S. cerevisiae} $R_{glucose}$ & 0.383        & 0.408    & 0.875       \\
      Akashi \& Gojobori (2002)          & 0.398        & 0.405    & 0.805       \\
      Craig \& Weber (1998) Energy       & 0.416        & 0.40     & 0.835       \\
      Craig \& Weber (1998) Steps        & 0.375        & 0.404    & 0.866       \\
      Wagner (2005) Respiratory          & 0.382        & 0.405    & 0.822       \\
      Wagner (2005) Fermentative         & 0.377        & 0.406    & 0.851       \\
      Molecular Weight                   & 0.422        & 0.405    & 0.767       \\ \bottomrule
    \end{tabular}
  \end{footnotesize}
  \caption[Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets.]{Variation explained by multivariate regression model fitting of transcriptomic, proteomic, and metabolomic datasets. Each value in the table is the $R^2$ for the complete multivariate regression model using the indicated cost type, and including the carbon, nitrogen, sulphur content of the amino acid, average tRNA count, and codon adaptation index for the transcript and protein data. The transcript and metabolomic data are described by Castrillo \emph{et al}. \cite{castrillo2007}. The proteomic data is described by Ghaemmaghami \emph{et al}. \cite{ghaemmaghami2003}.}
  \label{table:regression_analysis}
\end{table}

\paragraph{Transcriptome and proteome analysis}

The regression analysis of the transcript data from Castrillo \emph{et al.} \cite{castrillo2007} and the proteome data from \cite{ghaemmaghami2003} yielded similar results. In both these data codon adaptation index was the strongest predictor of measure levels of either transcript or protein. Figure~\ref{figure:regression_analysis} shows the CAI of the transcript was approximately half a magnitude greater as a predictor of transcription than the other included factors. In the protein data CAI was a full magnitude greater as a predictor of protein production. Together these results indicate the optimisation of a transcript for translation is a strong predictor of both transcript and protein levels.

Comparing other factors in the regression, amino acid cost, carbon content and nitrogen appear to show a similar level of importance in explaining variation in both transcript and protein data. The variables for tRNA count and and sulphur content appear slightly less important. Overall all though the regression analysis indicates that the majority of the variation in the transcript and protein data is explained by codon adaption index of the transcript, as opposed to the variables of costs, carbon, nitrogen, sulphur content, or corresponding genome encoded tRNAs.

As the first two columns of Table~\vref{table:regression_analysis} show, each of the different amino acid cost types had little effect on the $R^2$ of the multivariate regression of transcript and protein data. The difference in which cost type used in the transcript analysis was 4.7\%, ranging from 37.5\% for the regression using Craig \& Weber's step cost measures \cite{craig1998}, to 42.2\% for the regression using molecular weight as suggested Seligmann \cite{seligmann2003}. In the analysis of the protein data, the difference in variation explained dependent of the cost measure was even smaller 0.8\% with all cost measures having approximately the same effect in the multivariate regression of the proteome data.

\paragraph{Metabolome analysis}

In the regression of the metabolome data from Castrillo \emph{et al.} \cite{castrillo2007} codon adaptation index is not applicable and therefore the regression is based on genome encoded tRNAs, estimated cost and carbon, nitrogen and sulphur content of the amino acid. The third plot of Figure~\ref{figure:regression_analysis} illustrates that in the majority of each regression each of these factors appears to contribute equally to model fitting of free amino acid data in yeast. The exception to this is sulphur content which in half the regressions explains less regression than the other factors.

Compared with the multivariate regression of the transcript and protein data, the multivariate regression of the metabolite regression of the free amino acid data was able to explain a much larger degree of variation. The variation explained differed by 10.8\% between the best and worst regression model. The regression that explained the least used molecular weight for amino acid cost, while the best fit used $R_{glucose}$ explaining 87.5\%. This result indicates that a large degree of the variation in amino acid levels can be explained based on their cost, atomic content, and the corresponding tRNAs encoded in the genome.

\subsection{Amino acid cost in protein evolution}

\subsubsection{Amino acid variance between \emph{Saccharomyces} species}

The percentage amino acid usage across seven \emph{Saccharomyces} species was calculated. Table~\vref{table:total_amino_acids} shows the total number of amino acids included for each species. Figure~\vref{figure:amino_acid_usage} compares the median percentage usage in the predicted proteome with the molecular weight of each amino acid.

\begin{table}
  \centering
  \begin{tabular}{ l r }
                                                  \toprule
    \emph{Saccharomyces} species & Amino acids \\ \midrule
    \emph{bayanus}               & 2921220     \\
    \emph{castellii}             & 2328652     \\
    \emph{cerevisiae}            & 2916055     \\
    \emph{kluyveri}              & 1248804     \\
    \emph{kudriavzevii}          & 1546252     \\
    \emph{mikatae}               & 1172495     \\
    \emph{paradoxus}             & 2933095     \\ \bottomrule
  \end{tabular}
  \caption{Total number of amino acids for each \emph{Saccharomyces} species.}
  \label{table:total_amino_acids}
\end{table}

Figure~\ref{figure:amino_acid_usage} illustrates a weak trend between molecular weight and percentage usage in the \emph{Saccharomyces} genomes. Spearman's rank correlation indicates a non-significant correlation explaining 40\% of the variation in amino acid usage ($r = 0.4$,$p = 0.082$). The range of values for each point in the plot indicates variation in amino acid usage increases with molecular weight.

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{amino_acid_usage.eps}
  \caption[Comparison of amino acid usage with molecular weight]{Comparison of amino acid usage with molecular weight. Each point is the median percentage amino acid usage across 6 \emph{Saccharomyces} species: \emph{cerevisiae}, \emph{paradoxus}, \emph{mikatae}, \emph{kudriavzevii}, \emph{bayanus}, \emph{castellii}, and \emph{kluyveri}. The grey line indicates the maximum and minimum values across species.}
  \label{figure:amino_acid_usage}
\end{figure}

Figure~\vref{figure:amino_acid_variance} compares the median absolute deviation in percentage with molecular weight for each amino acid. Spearman's rank correlation indicates a stronger significant trend between the usage of amino acids in \emph{Saccharomyces} encoded proteomes and the molecular weight of the amino acid ($r = - 0.58$,$p = 0.0075$). This indicates a possible selection pressure to limit relatively more complex amino acids, while allowing greater variance in the use of simpler amino acids.

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{amino_acid_variance.eps}
  \caption[Comparison of deviation in amino acid usage with molecular weight]{Comparison of deviation in amino acid usage with molecular weight. Each point is the median absolute deviation in percentage amino acid usage across seven \emph{Saccharomyces} species: \emph{cerevisiae}, \emph{paradoxus}, \emph{mikatae}, \emph{kudriavzevii}, \emph{bayanus}, \emph{castellii}, and \emph{kluyveri}. Loess smoothing is used to indicate trend.}
  \label{figure:amino_acid_variance}
\end{figure}

\subsubsection{Multivariate analysis of amino acid usage}

The usage of amino acids across each of the seven species is contrasted in Figure~\vref{figure:heat}. The usage of each amino acid is displayed as a colour on the heatmap, each row is an amino acid and each column is a species. The darker grey the greater the relative usage of the amino acid in the genome. The two dendrograms in figure illustrates the hierarchical clustering of both amino acids and species based on the usage of each amino acid.

\begin{figure}
  \centering
  \includegraphics*[height=15cm]{heatmap.eps}
  \caption[Heatmap of amino acid usage across \emph{Saccharomyces} species]{Heatmap of amino acid usage across \emph{Saccharomyces} species. Each row is scaled by the mean where darker grey indicates a greater amino acid usage in that species relative to the other species. Agglomerative complete linkage hierarchical clustering produces dendrograms showing the Spearman's Rank relationship between both species, and individual amino acids.}
  \label{figure:heat}
\end{figure}

The upper species dendrogram shows a partition of the yeast species with \emph{kluyveri} in one group and the remaining species in another. The yeast species do not follow a phylogenetic distribution \emph{kluyveri} and \emph{castellii} are the two outermost species in the hierarchy which does follow the phylogeny. The other species \emph{cerevisiae}, \emph{paradoxus}, \emph{mikatae}, \emph{kudriavzevii} and \emph{bayanus} appear at different positions in the hierarchy based on amino acid usage, compared to the hierarchy based on genetic distance. This indicates amino acid usage may indicate broad speciation, but more smaller changes in amino acid usage may not necessarily relate to the species phylogeny.

Singular value decomposition was performed on the amino acid usage matrix. The percentage variation explained by each of the singular values is shown in Figure~\vref{figure:components}. Each value of the first component is positive indicating for both rows and columns of the usage matrix indicating the first component does not discriminate between either amino acids or species based on usage. The size of the component indicates a broad trend for amino acid usage across species explaining ~90\% of the variation.

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{components.eps}
  \caption[Components of amino acid usage singular value decomposition]{The percentage weight of each singular value of amino usage across seven \emph{Saccharomyces} species.}
  \label{figure:components}
\end{figure}

The first singular value of the decomposition of amino acid usage contained no negative values and so does not discriminate between either amino acids or species. The second singular value is the first to contain both negative and positive loadings. The second and third component loadings for each amino acid are plotted in Figure~\vref{figure:acid_decomp}. Each amino acid in the figure is plotted by a symbol proportional in size to molecular weight. A circle is added to the lot to indicate the values with a high loading and therefore contribute most to the variation between species. There is no distinct trend, but the figure indicates smaller amino acids have larger absolute loadings and explain a greater degree in variation. The higher molecular weight amino acids cluster towards the origin, and therefore are fixed between species.

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{acid_difference.eps}
  \caption[Amino acid projection onto first two discriminatory components of \emph{Saccharomyces} decomposition]{Amino acid projection onto first two discriminatory components of \emph{Saccharomyces} decomposition. The width of each square is proportional to amino acid molecular weight. The dotted lines indicate a loading of 0, where observations closer to 0 have a lesser contribution to the singular value. The dashed circle crosses both components at -0.3 and 0.3 and is added to highlight discriminatory observations.}
  \label{figure:acid_decomp}
\end{figure}

Canonical correlation analysis was used to find the maximum correlation between a set of unobserved variables in the matrix amino acid usage across species and a matrix of amino acid characteristics consisting of molecular weight $A_{glucose}$, $A_{ammonimum}$, and $A_{sulphur}$. The first canonical correlation explained 90\% of the variation between the two matrices. The weights for the amino acid characteristics are shown in Figure~\vref{figure:canon_decomp:loadings}. Glucose and sulphate are given similar high weights, while molecular weight and ammonium cost are relatively close to zero. This indicates amino acid usage trends across the \emph{Saccharomyces} species are best interpreted in terms of energy and sulphur content.

TODO: Change figure, sulphur to sulphate

\begin{figure}
  \centering
  \subfloat[Contribution of amino acid characteristics to first canonical correlation]{
    \label{figure:canon_decomp:loadings}
    \includegraphics*[height=8cm]{acid_correlation.eps}
  }
  \vfill
  \subfloat[Comparison of \emph{S. cerevisiae} amino acid usage with derived amino acid cost]{
    \label{figure:canon_decomp:usage}
    \includegraphics*[height=8cm]{canonical_cost.eps}
  }
  \caption[Canonical correlation of amino acid usage]{Canonical correlation of amino acid usage across seven \emph{Saccharomyces} species. Figure a illustrates the contributions of each of the amino acid characteristics to the first canonical correlation with the amino acid usages across the species. Figure a illustrates the derived amino acid cost derived from the first canonical component with amino acid usage in \emph{S. cerevisiae}. Loess smoothing is used to indicate trend.}
  \label{figure:canon_decomp}
\end{figure}

An combined estimate of amino acid cost was derived from the canonical weights of each amino acid characteristic. This cost is compared with amino acid usage for \emph{S. cerevisiae} in Figure~\ref{figure:canon_decomp:usage}. The combined cost explains a greater degree of variation in amino acid usage (Spearman R = 0.55, $p$ = 0.013) than using molecular weight alone indicating that a combined cost derived from multiple measures may provide the greatest predictor of the cost of amino acid use.

TODO: Insert molecular weight correlation with usage

\subsubsection{Cost trends in amino acid mutation rates}

Per site mutation rates were calculated for all columns in protein sequence alignments used by Someone \emph{et. al} \cite{wall2005}. The mutation rate was used to calculate a per-site cost of each alignment position, weighted by the estimated amino acid fixation time at each site. The distribution evolutionary weighted cost is shown in Figure~\vref{figure:mutation_density}.

TODO: How many protein alignments?

TODO: How many alignment columns?

TODO: How many ungapped alignment columns?

\begin{figure}
  \centering
  \includegraphics*[height=18cm]{weight_density.eps}
  \caption[Distribution of phylogeny-weighted residue cost.]{Distribution of phylogeny-weighted residue cost. The left figure represents the costs for all alignment columns excluding those with gaps. The figure on the right breaks the distribution into alignment columns that contain the same amino acid across all species (fixed), and alignment columns where the amino acid varies between species (varying). Molecular weight is used as the estimate of amino acid cost.}
  \label{figure:mutation_density}
\end{figure}

When all ungapped alignment sites are compared distinct peaks highlight conserved use of an amino acid at the same site across all species, between the high density peaks are the regions where the amino acid used varies between species. The distribution of cost between fixed and vary sites is contrasted in the right hand plot of Figure~\ref{figure:mutation_density}. This shows that varying sites display a more Gaussian distribution phylogeny-weighted costs, though there are still visible peaks. The fixed sites, as expected, show the density costs of each twenty amino acids where amino acids with a similar cost are merged into the same peak.

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{fixation.eps}
  \caption[Comparison of amino acid fixation with molecular weight]{Comparison of amino acid fixation with molecular weight. Percentage fixation is the number of times the amino acid is used across all species in an alignment column, divided by the total number of times the amino is used in the \emph{Saccharomyces species}. Loess smoothing is used to indicate a trend line.}
  \label{figure:fixation_weight}
\end{figure}

The average percentage fixation rate across all each species in the alignment, was calculated for each residue in the \emph{S. cerevisiae} protein sequences. The percentage fixation of each amino acid is compared with molecular weight in Figure~\vref{figure:fixation_weight}. The figure indicates no clear relationship between the molecular weight of an amino acid and the level of site-fixation across species. The Spearman's Rank correlation between the two variable is 0.41, however the indicated relationship may not be statistically significant ($p = 0.07$). This suggestion more expensive amino acids do not show a trend to be retained in protein sequences compared to less expensive amino acids.

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{mutation.eps}
  \caption[Comparison of amino acid mean mutation rate with molecular weight]{Comparison of amino acid mean mutation rate with molecular weight. The mutation rate is the mean of each alignment column where the amino acid appears in the \emph{S. cerevisiae} sequence. Loess smoothing is used to indicate trend.}
  \label{figure:mutation_weight}
\end{figure}

The mean mutation rate of each amino acid in \emph{cerevisiae} across all alignment columns is compared with the molecular weight in Figure~\vref{figure:mutation_weight}. This figure suggests a clearer trend where the use of more expensive amino acids is correlated with a reduced mutation rate (Spearman R = -0.81, $p <$ 0.0001).

\begin{figure}
  \centering
  \includegraphics*[height=10cm]{mutation_vs_fixed.eps}
  \caption[Comparison of mutation rate with percentage sites fixed]{Comparison of mutation rate with percentage sites fixed. Mutation rate is the mean of each alignment column where the amino acid appears in the \emph{S. cerevisiae} sequence. Percentage fixation is the number of times the amino acid is used across all species in an alignment column, divided by the total number of times the amino is used in the \emph{Saccharomyces species}.Robust linear regression is used to indicate trend line.}
  \label{figure:mutation_vs_fixed}
\end{figure}

Figure~\vref{figure:mutation_vs_fixed} compares the average mutation for each amino acid with percentage the usage of each amino acid is fixed. This figure shows that fixation rate do not show a strong correlation (Spearman R = -0.58, $p$ = 0.008) as may be expected from intuition. Robust linear regression is used to highlight the amino acids with a large residual from the predicted trend. The amino acids cysteine, proline, and glycine have a higher mutation rate than expected given the percentage of sites at which they are fixed across the alignments: at sites where the use of the amino acid is not fixed there is a high amino acid turnover across species. The amino acid glutamine has a smaller mutation rate given the number of sites fixed: if glutamine is not fixed at position the rate of mutation is slower than would be expected on average. The large residual deviations may indicate a selection pressures associated with the functions of particular amino acids.

\clearpage

\section{Discussion}

\subsection{A small factor in yeast gene expression}

\subsubsection{Predictors of transcript and protein levels}

The multivariate analysis indicated that amino acid cost is a small factor explanatory factor in the analysed \emph{S. cerevisiae} transcript and protein data. Regardless of which cost type was used in the regression the CAI of the encoding transcript was the most important variable for predicting transcript and protein levels. The importance of codon adaptation for predicting transcript levels is not a new discovery, but this analysis compares codon adaptation with other predictors of transcript and protein levels to show that codon adaptation is a much greater predictor of transcript and protein levels than factors such as amino acid costs \cite{akashi2002}, atomic content \cite{mazel1989,baudoin2001}, or tRNA levels \cite{akashi2003}.

After codon adaptation is considered the other factors in the regression all appear to play approximately a similar role in explaining variation in protein and transcript expression. In general across each cost regression model of the transcript data cost carbon and nitrogen appear more important than tRNA or sulphur content. There is more variation in the protein data, but a similar trend across these variables is apparent. In some cases the removal of tRNA or sulphur content, and even carbon content in one instance produces a more parsimonious regression model. 

Interpreting the results of the cost regression analysis, codon adaptation is the strongest predictor of both the transcript and protein expression data. The other factors in the regression equation, while do show some effect in prediction are much less important than that of codon adaptation.

\subsubsection{Predictors of metabolite levels}

Codon adaptation is not applicable to metabolite levels, and therefore all explanatory power is based on amino acid cost, genome encoded tRNA, and carbon, nitrogen and sulphur content. The variation explained in the metabolite data was much greater than that of the transcript and protein data, more than twice as much depending on which cost measure was used. This result indicates that the combination of factors used in that analysis are strong predictors of the variation in metabolomic data, and that amino acids are maintained in the cell at levels related to cost, atomic content and genome encoded tRNA levels.

\subsection{A selection pressure in protein sequence evolution}

\subsubsection{Usage variation between species}

Comparison of the average usage of each amino acid between indicates a cost selection in a weak selection pressure in \emph{Saccharomyces} species. The high degree of variation in the relationship between cost and usage show amino acid levels are not maintained in the genome relative to biosynthetic cost. Amino acid usage in the data is instead likely to be as a result of functional requirement where each amino acid is retained according to functional at each site in the protein the amino acid is used. The limited relationship between cost and usage may however still indicate a trend to minimise the use of expensive amino acids albeit after satisfying all constraints of protein structure and function.

The comparison of amino acid genome usage and metabolic cost is not a novel approach as outlined in the introduction. Two studies have analysed cost related amino acid usage across a large range of species \cite{seligmann2003,swire2007} and found the cost minimisation pressure is variable dependent of the phylogenetic group examined. Comparison of the usage of amino acids between closely related species in the \emph{Saccharomyces} genus shows, while cost minimisation is not apparent is amino usage, the variability in amino acids shows a strong cost related trend. Costly amino acid show a reduced interspecies variance compared with cheaper amino acids. This may indicate a selection pressure to maintain existing expensive amino acids for a specific functional role, while the insertion of additional costly acids is reduced to minimise cost. Cheaper amino acids may vary more in genome usage as the cost selection pressure on their is proportionally reduced.

Examining the median deviation in amino acid use across a single genus highlights phylogenetically related trends not necessarily apparent in comparing the trends in simple median amino acid use. This results indicates a trend related the substitution rates of amino acids but to a lesser extent in the levels at which amino acids are maintained in the genome.

\subsubsection{Multivariate trends in amino acid use}

Singular value decomposition of amino acid across the \emph{Saccharomyces} genus shows a single component explains ~90\% of the variation in amino acid use across species. The remaining components explain a small fractional amount of the remaining variation. The decomposition indicates a broad trend of relative invariance in amino acid usage, which may would be expected between closely related species. At longer evolutionary time scales greater variance between amino acid usage is observed \emph{tekaia2006} delineating species based on environment and GC content.

The first two discriminatory components of the decomposition show indicate that usage of smaller and cheaper amino acid explain the variation between species where smaller amino acids have a larger absolute singular value weight than larger amino acids. This trend is however difficult to quantitatively measure and requires further analysis to determine the exact nature and significance of cost to explaining the variation between species.

Canonical correlation of amino acid characteristics and their usage between species highlights a strong trend related to biosynthesis from glucose and sulphate. Only two amino acids include a sulphur atom and therefore indicates that amino usage in the \emph{Saccharomyces} genus is related to energy expenditure in synthesis plus sulphur usage where relevant. Nitrogen content and molecular weight made no large contribution to the first canonical correlation and suggests that neither nitrogen material costs or the additional burden of synthesising large complex amino acid are a significant factor in amino acid synthesis.

The use of a glucose and sulphate integrated cost is validated in the greater amount of variation explained in \emph{S. cerevisiae} amino acid levels contrasted with the variation explained using molecular weight alone. This result suggests an integrated multi-factorial view of amino acid biosynthesis may be necessary to understand the importance of cost minimisation in shaping amino acid usage.

\subsubsection{Amino acid cost affects mutation rate}

Mutation rates in amino acid usage across the four \emph{Saccharomyces} species considered show a strong trend related to biosynthetic cost. This indicates cost selection may be a strong trend in the evolution of protein sequence. Biosynthetically expensive amino acids have a low mutation rate indicating a selection pressure against transitions to other amino acids at the same position. This may indicate that expensive amino acids are required and fixed in the genome for a specific functional role, and additional expensive amino acids are not included because of the cost burden of their synthesis. Biosynthetically cheaper amino acids may vary more because the cost burden association with random insertion into the genome is much less. The only constraint on cheaper amino acids may be their functional role in the protein structure or function.

In contrast with mutation rate, the fixation of amino acids shows no trend with biosynthetic cost, unusual given that fixation rate does. The selection pressure behind this trend is unclear, though the cause may be that certain amino acid have a higher fixation rate for specific functional roles. Glycine for example is biosynthetically cheap but has a high fixation rate, the reason for this may be that glycine has a very small mass and therefore has specific functional roles in protein structure requiring such small size.

Comparing mean mutation rate with percentage fixation highlights specific amino acids with greater or lesser than expected mutation rates given the level of fixation. In addition to glycine are the amino acids cysteine, proline, and glutamine with large difference in mutation and fixation. The causes underlying these differences may be attributed to specific functional purposes of the amino acid which therefore result in changes in selection pressures for particular amino acids.

\subsubsection{Future work to understand the role of metabolic cost}

The previous chapter showed amino acid metabolic cost varies between species and environment. A similar trend may also be inferred for the importance of metabolic cost of sequence evolution. The \emph{Saccharomyces} genus examined in this analysis have an extensive regulatory network and show distinct cellular responses to a variety of conditions \cite{castrillo2007}. Therefore metabolic cost may not be as strong a selection pressure as the ability to adapt to a variety of conditions. Future analysis could consider small oceanic species invariant to environmental conditions where metabolic cost may be a much stronger selection pressure given the genome is streamlined for a single environment \cite{peligabacter}. Such species with compact genomes and limited number of paralogs may indicate that cost minimisation is inversely proportional to adaptation to multiple environments. Gene duplication is a factor in adaptation to new environments \cite{ryan,envirologs} and the the duplication of gene expression is above the threshold of negative selection \cite{swire2007}, therefore the paralog count may be key to identifying the relationship between cost selection and environmental adaptation.

TODO: Metagenomics paper, variance in amino acid composition not related to cost. Instead may related to limitation of sparse cofactors related to synthesis.

\section{Conclusions}

Is metabolic cost a force in protein sequence evolution? Analysis of transcript and protein production, or biomass composition suggests that cost minimisation explains a limited amount of variation in the data examined here, and is therefore a weak selection pressure is shaping sequence composition or cellular expression. Detailed analysis of the mutation rates of each amino acid reveals a strong trend related to amino acid biosynthetic cost indicating that cost minimisation is indeed a selection pressure in sequence evolution.
